{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Datasets had to be updated so each set had consistent variables. Also, some sample data had been left out of the\n",
    "# metadata files, which needed to be added.\n",
    "\n",
    "## Most of the datasets required the same corrections: fixing the tax table column names, adding \"PyOM\" and \"PyOM.pH\" \n",
    "##columns, fixing spelling mistakes, fixing/correcting sample data column names.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Loading R packages\n",
    "library(reshape)\n",
    "library(ggplot2)\n",
    "library(phyloseq)\n",
    "library(plyr)\n",
    "library(dplyr)\n",
    "library(plotly)\n",
    "library(wesanderson)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Wu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ps_wu = import_biom(\"Wu_OTU_table/feature-table-metaD-tax_json.biom\" , parseFunction = parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x = data.frame(tax_table(ps_wu))\n",
    "# Making a dummy variable to store the taxonomy data\n",
    "\n",
    "colnames(x) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x$Domain = gsub(\"D_0__\", \"\", as.character(x$Domain))\n",
    "x$Phylum = gsub(\"D_1__\", \"\", as.character(x$Phylum))\n",
    "x$Class = gsub(\"D_2__\", \"\", as.character(x$Class))\n",
    "x$Order = gsub(\"D_3__\", \"\", as.character(x$Order))\n",
    "x$Family = gsub(\"D_4__\", \"\", as.character(x$Family))\n",
    "x$Genus = gsub(\"D_5__\", \"\", as.character(x$Genus))\n",
    "x$Species = gsub(\"D_6__\", \"\", as.character(x$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x=tax_table(as.matrix(x,dimnames=list(row.names(x),colnames(x))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "tax_table(ps_wu)=x\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_wu))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#no soil type was provided in the publication; we looked at the paper to obtain a rough estimate on where the soil was \n",
    "#collected- the soil type was sourced from the FAO/UNESCO Soil Map of the World\n",
    "\n",
    "##Adding the soil-type to the Wu sample data\n",
    "sample_data(ps_wu)[sample_data(ps_wu)$Author==\"Wu\",]$SoilType = \"Gleysol\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "sample_data(ps_wu)$Treatment = revalue(sample_data(ps_wu)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_wu)$SoilType = revalue(sample_data(ps_wu)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_wu)$Feedstock = revalue(sample_data(ps_wu)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_wu)$Habitat = revalue(sample_data(ps_wu)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_wu)$CurrentLandUse = revalue(sample_data(ps_wu)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_wu)$PyOM.Temp = revalue(sample_data(ps_wu)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_wu)$Depth = revalue(sample_data(ps_wu)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'%C'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'Feedstock'\n",
       "\\item '\\%C'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'RevBarcode'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'Silt'\n",
       "\\item 'Sand'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Clay'\n",
       "\\item 'Treatment'\n",
       "\\item 'BioSample'\n",
       "\\item 'Organism'\n",
       "\\item 'SoilType'\n",
       "\\item 'Habitat'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'BioProject'\n",
       "\\item 'Crop'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'TargetGene'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'Author'\n",
       "\\item 'SRA'\n",
       "\\item 'pH'\n",
       "\\item 'Region'\n",
       "\\item 'PyOM'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeographicalLocation'\n",
       "2. 'PyOM.Temp'\n",
       "3. 'Sampling_Depth_cm'\n",
       "4. 'CurrentLandUse'\n",
       "5. 'FwdBarcode'\n",
       "6. 'NucleicAcidExtraction'\n",
       "7. 'Feedstock'\n",
       "8. '%C'\n",
       "9. 'ForwardPrimerName'\n",
       "10. 'RevBarcode'\n",
       "11. 't.per.ha.PyOM'\n",
       "12. 'Silt'\n",
       "13. 'Sand'\n",
       "14. 'RevPrimerName'\n",
       "15. 'Clay'\n",
       "16. 'Treatment'\n",
       "17. 'BioSample'\n",
       "18. 'Organism'\n",
       "19. 'SoilType'\n",
       "20. 'Habitat'\n",
       "21. 'Incubation_Time_Weeks'\n",
       "22. 'BioProject'\n",
       "23. 'Crop'\n",
       "24. 'ReversePrimerSequence'\n",
       "25. 'CollectionYear'\n",
       "26. 'TargetGene'\n",
       "27. 'SequencingMethod'\n",
       "28. 'ForwardPrimerSequence'\n",
       "29. 'Author'\n",
       "30. 'SRA'\n",
       "31. 'pH'\n",
       "32. 'Region'\n",
       "33. 'PyOM'\n",
       "34. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GeographicalLocation\"  \"PyOM.Temp\"             \"Sampling_Depth_cm\"    \n",
       " [4] \"CurrentLandUse\"        \"FwdBarcode\"            \"NucleicAcidExtraction\"\n",
       " [7] \"Feedstock\"             \"%C\"                    \"ForwardPrimerName\"    \n",
       "[10] \"RevBarcode\"            \"t.per.ha.PyOM\"         \"Silt\"                 \n",
       "[13] \"Sand\"                  \"RevPrimerName\"         \"Clay\"                 \n",
       "[16] \"Treatment\"             \"BioSample\"             \"Organism\"             \n",
       "[19] \"SoilType\"              \"Habitat\"               \"Incubation_Time_Weeks\"\n",
       "[22] \"BioProject\"            \"Crop\"                  \"ReversePrimerSequence\"\n",
       "[25] \"CollectionYear\"        \"TargetGene\"            \"SequencingMethod\"     \n",
       "[28] \"ForwardPrimerSequence\" \"Author\"                \"SRA\"                  \n",
       "[31] \"pH\"                    \"Region\"                \"PyOM\"                 \n",
       "[34] \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'%C'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'Feedstock'\n",
       "\\item '\\%C'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'RevBarcode'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'Silt'\n",
       "\\item 'Sand'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Clay'\n",
       "\\item 'Treatment'\n",
       "\\item 'BioSample'\n",
       "\\item 'Organism'\n",
       "\\item 'SoilType'\n",
       "\\item 'Habitat'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'BioProject'\n",
       "\\item 'Crop'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'TargetGene'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'Author'\n",
       "\\item 'SRA'\n",
       "\\item 'pH'\n",
       "\\item 'Region'\n",
       "\\item 'PyOM'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeographicalLocation'\n",
       "2. 'PyOM.Temp'\n",
       "3. 'Sampling_Depth_cm'\n",
       "4. 'CurrentLandUse'\n",
       "5. 'FwdBarcode'\n",
       "6. 'NucleicAcidExtraction'\n",
       "7. 'Feedstock'\n",
       "8. '%C'\n",
       "9. 'ForwardPrimerName'\n",
       "10. 'RevBarcode'\n",
       "11. 't.per.ha.PyOM'\n",
       "12. 'Silt'\n",
       "13. 'Sand'\n",
       "14. 'RevPrimerName'\n",
       "15. 'Clay'\n",
       "16. 'Treatment'\n",
       "17. 'BioSample'\n",
       "18. 'Organism'\n",
       "19. 'SoilType'\n",
       "20. 'Habitat'\n",
       "21. 'Incubation_Time_Weeks'\n",
       "22. 'BioProject'\n",
       "23. 'Crop'\n",
       "24. 'ReversePrimerSequence'\n",
       "25. 'CollectionYear'\n",
       "26. 'TargetGene'\n",
       "27. 'SequencingMethod'\n",
       "28. 'ForwardPrimerSequence'\n",
       "29. 'Author'\n",
       "30. 'SRA'\n",
       "31. 'pH'\n",
       "32. 'Region'\n",
       "33. 'PyOM'\n",
       "34. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GeographicalLocation\"  \"PyOM.Temp\"             \"Sampling_Depth_cm\"    \n",
       " [4] \"CurrentLandUse\"        \"FwdBarcode\"            \"NucleicAcidExtraction\"\n",
       " [7] \"Feedstock\"             \"%C\"                    \"ForwardPrimerName\"    \n",
       "[10] \"RevBarcode\"            \"t.per.ha.PyOM\"         \"Silt\"                 \n",
       "[13] \"Sand\"                  \"RevPrimerName\"         \"Clay\"                 \n",
       "[16] \"Treatment\"             \"BioSample\"             \"Organism\"             \n",
       "[19] \"SoilType\"              \"Habitat\"               \"Incubation_Time_Weeks\"\n",
       "[22] \"BioProject\"            \"Crop\"                  \"ReversePrimerSequence\"\n",
       "[25] \"CollectionYear\"        \"TargetGene\"            \"SequencingMethod\"     \n",
       "[28] \"ForwardPrimerSequence\" \"Author\"                \"SRA\"                  \n",
       "[31] \"pH\"                    \"Region\"                \"PyOM\"                 \n",
       "[34] \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and t.per.ha.PyOM are consistent among datasets\n",
    "names(sample_data(ps_wu))[names(sample_data(ps_wu)) == \"PyOM-Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_wu))[names(sample_data(ps_wu)) == \"t.per.hakg.PyOM\"] <- \"t.per.ha.PyOM\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\")\n",
    "OMAdded = c('Composted biochar and straw + soil','Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'%C'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'Feedstock'\n",
       "\\item '\\%C'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'RevBarcode'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'Silt'\n",
       "\\item 'Sand'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Clay'\n",
       "\\item 'Treatment'\n",
       "\\item 'BioSample'\n",
       "\\item 'Organism'\n",
       "\\item 'SoilType'\n",
       "\\item 'Habitat'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'BioProject'\n",
       "\\item 'Crop'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'TargetGene'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'Author'\n",
       "\\item 'SRA'\n",
       "\\item 'pH'\n",
       "\\item 'Region'\n",
       "\\item 'PyOM'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeographicalLocation'\n",
       "2. 'PyOM.Temp'\n",
       "3. 'Sampling_Depth_cm'\n",
       "4. 'CurrentLandUse'\n",
       "5. 'FwdBarcode'\n",
       "6. 'NucleicAcidExtraction'\n",
       "7. 'Feedstock'\n",
       "8. '%C'\n",
       "9. 'ForwardPrimerName'\n",
       "10. 'RevBarcode'\n",
       "11. 't.per.ha.PyOM'\n",
       "12. 'Silt'\n",
       "13. 'Sand'\n",
       "14. 'RevPrimerName'\n",
       "15. 'Clay'\n",
       "16. 'Treatment'\n",
       "17. 'BioSample'\n",
       "18. 'Organism'\n",
       "19. 'SoilType'\n",
       "20. 'Habitat'\n",
       "21. 'Incubation_Time_Weeks'\n",
       "22. 'BioProject'\n",
       "23. 'Crop'\n",
       "24. 'ReversePrimerSequence'\n",
       "25. 'CollectionYear'\n",
       "26. 'TargetGene'\n",
       "27. 'SequencingMethod'\n",
       "28. 'ForwardPrimerSequence'\n",
       "29. 'Author'\n",
       "30. 'SRA'\n",
       "31. 'pH'\n",
       "32. 'Region'\n",
       "33. 'PyOM'\n",
       "34. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"GeographicalLocation\"  \"PyOM.Temp\"             \"Sampling_Depth_cm\"    \n",
       " [4] \"CurrentLandUse\"        \"FwdBarcode\"            \"NucleicAcidExtraction\"\n",
       " [7] \"Feedstock\"             \"%C\"                    \"ForwardPrimerName\"    \n",
       "[10] \"RevBarcode\"            \"t.per.ha.PyOM\"         \"Silt\"                 \n",
       "[13] \"Sand\"                  \"RevPrimerName\"         \"Clay\"                 \n",
       "[16] \"Treatment\"             \"BioSample\"             \"Organism\"             \n",
       "[19] \"SoilType\"              \"Habitat\"               \"Incubation_Time_Weeks\"\n",
       "[22] \"BioProject\"            \"Crop\"                  \"ReversePrimerSequence\"\n",
       "[25] \"CollectionYear\"        \"TargetGene\"            \"SequencingMethod\"     \n",
       "[28] \"ForwardPrimerSequence\" \"Author\"                \"SRA\"                  \n",
       "[31] \"pH\"                    \"Region\"                \"PyOM\"                 \n",
       "[34] \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Control'</li>\n",
       "\t<li>'OM'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'OM'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'OM'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Control'\n",
       "\\item 'OM'\n",
       "\\item 'PyOM'\n",
       "\\item 'OM'\n",
       "\\item 'PyOM'\n",
       "\\item 'OM'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Control'\n",
       "2. 'OM'\n",
       "3. 'PyOM'\n",
       "4. 'OM'\n",
       "5. 'PyOM'\n",
       "6. 'OM'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Control\" \"OM\"      \"PyOM\"    \"OM\"      \"PyOM\"    \"OM\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "sample_data(ps_wu)$PyOM = ifelse(sample_data(ps_wu)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_wu)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_wu)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>9.98</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>7.13</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 9.98\n",
       "\\item <NA>\n",
       "\\item 7.13\n",
       "\\item <NA>\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. &lt;NA&gt;\n",
       "3. 9.98\n",
       "4. &lt;NA&gt;\n",
       "5. 7.13\n",
       "6. &lt;NA&gt;\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1]   NA   NA 9.98   NA 7.13   NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "sample_data(ps_wu)$PyOM.pH = ifelse(sample_data(ps_wu)$Treatment %in% \"Biochar + soil\", 9.98 ,\n",
    "                                    ifelse(sample_data(ps_wu)$Treatment %in% \"Composted biochar + soil\", 7.13, NA))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "saveRDS(ps_wu, \"wu.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Imparato"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "ps_imparato = import_biom(\"Imparato_OTU_table/feature-table-metaD-tax_json.biom\" , parseFunction = parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "x.imparato = data.frame(tax_table(ps_imparato))\n",
    "colnames(x.imparato) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x.imparato$Domain = gsub(\"D_0__\", \"\", as.character(x.imparato$Domain))\n",
    "x.imparato$Phylum = gsub(\"D_1__\", \"\", as.character(x.imparato$Phylum))\n",
    "x.imparato$Class = gsub(\"D_2__\", \"\", as.character(x.imparato$Class))\n",
    "x.imparato$Order = gsub(\"D_3__\", \"\", as.character(x.imparato$Order))\n",
    "x.imparato$Family = gsub(\"D_4__\", \"\", as.character(x.imparato$Family))\n",
    "x.imparato$Genus = gsub(\"D_5__\", \"\", as.character(x.imparato$Genus))\n",
    "x.imparato$Species = gsub(\"D_6__\", \"\", as.character(x.imparato$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x.imparato=tax_table(as.matrix(x.imparato,dimnames=list(row.names(x.imparato),colnames(x.imparato))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_imparato)=x.imparato\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_imparato))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_imparato)$Treatment = revalue(sample_data(ps_imparato)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_imparato)$SoilType = revalue(sample_data(ps_imparato)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_imparato)$Feedstock = revalue(sample_data(ps_imparato)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_imparato)$Habitat = revalue(sample_data(ps_imparato)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_imparato)$CurrentLandUse = revalue(sample_data(ps_imparato)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_imparato)$PyOM.Temp = revalue(sample_data(ps_imparato)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_imparato)$Depth = revalue(sample_data(ps_imparato)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'t.per.hakg.PyOM'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'%C'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Depth'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Feedstock'\n",
       "\\item 'Clay'\n",
       "\\item 't.per.hakg.PyOM'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'pH'\n",
       "\\item '\\%C'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Crop'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'TargetGene'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'Region'\n",
       "\\item 'Silt'\n",
       "\\item 'BioProject'\n",
       "\\item '\\%SOM'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'Organism'\n",
       "\\item 'Author'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'BioSample'\n",
       "\\item 'SoilType'\n",
       "\\item 'Habitat'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Depth'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'Treatment'\n",
       "\\item 'Sand'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'SRA'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'PyOM'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Feedstock'\n",
       "2. 'Clay'\n",
       "3. 't.per.hakg.PyOM'\n",
       "4. 'NucleicAcidExtraction'\n",
       "5. 'pH'\n",
       "6. '%C'\n",
       "7. 'ForwardPrimerName'\n",
       "8. 'Crop'\n",
       "9. 'ForwardPrimerSequence'\n",
       "10. 'TargetGene'\n",
       "11. 'Incubation_Time_Weeks'\n",
       "12. 'Region'\n",
       "13. 'Silt'\n",
       "14. 'BioProject'\n",
       "15. '%SOM'\n",
       "16. 'CurrentLandUse'\n",
       "17. 'Organism'\n",
       "18. 'Author'\n",
       "19. 'PyOM.Temp'\n",
       "20. 'FwdBarcode'\n",
       "21. 'RevBarcode'\n",
       "22. 'BioSample'\n",
       "23. 'SoilType'\n",
       "24. 'Habitat'\n",
       "25. 'RevPrimerName'\n",
       "26. 'Depth'\n",
       "27. 'SequencingMethod'\n",
       "28. 'Treatment'\n",
       "29. 'Sand'\n",
       "30. 'GeographicalLocation'\n",
       "31. 'SRA'\n",
       "32. 'CollectionYear'\n",
       "33. 'ReversePrimerSequence'\n",
       "34. 'PyOM'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Feedstock\"             \"Clay\"                  \"t.per.hakg.PyOM\"      \n",
       " [4] \"NucleicAcidExtraction\" \"pH\"                    \"%C\"                   \n",
       " [7] \"ForwardPrimerName\"     \"Crop\"                  \"ForwardPrimerSequence\"\n",
       "[10] \"TargetGene\"            \"Incubation_Time_Weeks\" \"Region\"               \n",
       "[13] \"Silt\"                  \"BioProject\"            \"%SOM\"                 \n",
       "[16] \"CurrentLandUse\"        \"Organism\"              \"Author\"               \n",
       "[19] \"PyOM.Temp\"             \"FwdBarcode\"            \"RevBarcode\"           \n",
       "[22] \"BioSample\"             \"SoilType\"              \"Habitat\"              \n",
       "[25] \"RevPrimerName\"         \"Depth\"                 \"SequencingMethod\"     \n",
       "[28] \"Treatment\"             \"Sand\"                  \"GeographicalLocation\" \n",
       "[31] \"SRA\"                   \"CollectionYear\"        \"ReversePrimerSequence\"\n",
       "[34] \"PyOM\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'%C'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Depth'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Feedstock'\n",
       "\\item 'Clay'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'pH'\n",
       "\\item '\\%C'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Crop'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'TargetGene'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'Region'\n",
       "\\item 'Silt'\n",
       "\\item 'BioProject'\n",
       "\\item '\\%SOM'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'Organism'\n",
       "\\item 'Author'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'BioSample'\n",
       "\\item 'SoilType'\n",
       "\\item 'Habitat'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Depth'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'Treatment'\n",
       "\\item 'Sand'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'SRA'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'PyOM'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Feedstock'\n",
       "2. 'Clay'\n",
       "3. 't.per.ha.PyOM'\n",
       "4. 'NucleicAcidExtraction'\n",
       "5. 'pH'\n",
       "6. '%C'\n",
       "7. 'ForwardPrimerName'\n",
       "8. 'Crop'\n",
       "9. 'ForwardPrimerSequence'\n",
       "10. 'TargetGene'\n",
       "11. 'Incubation_Time_Weeks'\n",
       "12. 'Region'\n",
       "13. 'Silt'\n",
       "14. 'BioProject'\n",
       "15. '%SOM'\n",
       "16. 'CurrentLandUse'\n",
       "17. 'Organism'\n",
       "18. 'Author'\n",
       "19. 'PyOM.Temp'\n",
       "20. 'FwdBarcode'\n",
       "21. 'RevBarcode'\n",
       "22. 'BioSample'\n",
       "23. 'SoilType'\n",
       "24. 'Habitat'\n",
       "25. 'RevPrimerName'\n",
       "26. 'Depth'\n",
       "27. 'SequencingMethod'\n",
       "28. 'Treatment'\n",
       "29. 'Sand'\n",
       "30. 'GeographicalLocation'\n",
       "31. 'SRA'\n",
       "32. 'CollectionYear'\n",
       "33. 'ReversePrimerSequence'\n",
       "34. 'PyOM'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Feedstock\"             \"Clay\"                  \"t.per.ha.PyOM\"        \n",
       " [4] \"NucleicAcidExtraction\" \"pH\"                    \"%C\"                   \n",
       " [7] \"ForwardPrimerName\"     \"Crop\"                  \"ForwardPrimerSequence\"\n",
       "[10] \"TargetGene\"            \"Incubation_Time_Weeks\" \"Region\"               \n",
       "[13] \"Silt\"                  \"BioProject\"            \"%SOM\"                 \n",
       "[16] \"CurrentLandUse\"        \"Organism\"              \"Author\"               \n",
       "[19] \"PyOM.Temp\"             \"FwdBarcode\"            \"RevBarcode\"           \n",
       "[22] \"BioSample\"             \"SoilType\"              \"Habitat\"              \n",
       "[25] \"RevPrimerName\"         \"Depth\"                 \"SequencingMethod\"     \n",
       "[28] \"Treatment\"             \"Sand\"                  \"GeographicalLocation\" \n",
       "[31] \"SRA\"                   \"CollectionYear\"        \"ReversePrimerSequence\"\n",
       "[34] \"PyOM\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and t.per.ha.PyOM are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_imparato))[names(sample_data(ps_imparato)) == \"PyOM-Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_imparato))[names(sample_data(ps_imparato)) == \"t.per.hakg.PyOM\"] <- \"t.per.ha.PyOM\"\n",
    "colnames(sample_data(ps_imparato))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\")\n",
    "OMAdded = c('Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_imparato)$PyOM = ifelse(sample_data(ps_imparato)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_imparato)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_imparato)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>11.6</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item 11.6\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 11.6\n",
       "2. 11.6\n",
       "3. 11.6\n",
       "4. 11.6\n",
       "5. 11.6\n",
       "6. 11.6\n",
       "7. 11.6\n",
       "8. 11.6\n",
       "9. 11.6\n",
       "10. 11.6\n",
       "11. 11.6\n",
       "12. 11.6\n",
       "13. 11.6\n",
       "14. 11.6\n",
       "15. 11.6\n",
       "16. 11.6\n",
       "17. 11.6\n",
       "18. 11.6\n",
       "19. &lt;NA&gt;\n",
       "20. &lt;NA&gt;\n",
       "21. &lt;NA&gt;\n",
       "22. &lt;NA&gt;\n",
       "23. &lt;NA&gt;\n",
       "24. &lt;NA&gt;\n",
       "25. &lt;NA&gt;\n",
       "26. &lt;NA&gt;\n",
       "27. &lt;NA&gt;\n",
       "28. &lt;NA&gt;\n",
       "29. &lt;NA&gt;\n",
       "30. &lt;NA&gt;\n",
       "31. &lt;NA&gt;\n",
       "32. &lt;NA&gt;\n",
       "33. &lt;NA&gt;\n",
       "34. &lt;NA&gt;\n",
       "35. &lt;NA&gt;\n",
       "36. &lt;NA&gt;\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6 11.6\n",
       "[16] 11.6 11.6 11.6   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA   NA\n",
       "[31]   NA   NA   NA   NA   NA   NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_imparato)$PyOM.pH = ifelse(sample_data(ps_imparato)$PyOM %in% \"PyOM\", 11.6 , NA)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_imparato, \"imparato.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dai 2017"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "ps_dai2017 = import_biom(\"Dai2017_OTU_table/feature-table-metaD-tax_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "x.dai2017 = data.frame(tax_table(ps_dai2017))\n",
    "\n",
    "colnames(x.dai2017) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x.dai2017$Domain = gsub(\"D_0__\", \"\", as.character(x.dai2017$Domain))\n",
    "x.dai2017$Phylum = gsub(\"D_1__\", \"\", as.character(x.dai2017$Phylum))\n",
    "x.dai2017$Class = gsub(\"D_2__\", \"\", as.character(x.dai2017$Class))\n",
    "x.dai2017$Order = gsub(\"D_3__\", \"\", as.character(x.dai2017$Order))\n",
    "x.dai2017$Family = gsub(\"D_4__\", \"\", as.character(x.dai2017$Family))\n",
    "x.dai2017$Genus = gsub(\"D_5__\", \"\", as.character(x.dai2017$Genus))\n",
    "x.dai2017$Species = gsub(\"D_6__\", \"\", as.character(x.dai2017$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x.dai2017=tax_table(as.matrix(x.dai2017,dimnames=list(row.names(x.dai2017),colnames(x.dai2017))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_dai2017)=x.dai2017\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "(tax_table(ps_dai2017))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>SoilType</th><th scope=col>Feedstock</th><th scope=col>Treatment</th><th scope=col>PyOM</th><th scope=col>PyOM.Temp</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>SRR3681435-X</th><td>Argiustoll</td><td>Manure    </td><td>Ar300     </td><td>PyOM      </td><td>300       </td></tr>\n",
       "\t<tr><th scope=row>SRR3681436-X</th><td>Argiustoll</td><td>Manure    </td><td>Ar300     </td><td>PyOM      </td><td>300       </td></tr>\n",
       "\t<tr><th scope=row>SRR3681437-X</th><td>Argiustoll</td><td>Manure    </td><td>Ar300     </td><td>PyOM      </td><td>300       </td></tr>\n",
       "\t<tr><th scope=row>SRR3681438-X</th><td>Argiustoll</td><td>Manure    </td><td>Ar700     </td><td>PyOM      </td><td>700       </td></tr>\n",
       "\t<tr><th scope=row>SRR3681439-X</th><td>Argiustoll</td><td>Manure    </td><td>Ar700     </td><td>PyOM      </td><td>700       </td></tr>\n",
       "\t<tr><th scope=row>SRR3681440-X</th><td>Argiustoll</td><td>Manure    </td><td>Ar700     </td><td>PyOM      </td><td>700       </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllll}\n",
       "  & SoilType & Feedstock & Treatment & PyOM & PyOM.Temp\\\\\n",
       "\\hline\n",
       "\tSRR3681435-X & Argiustoll & Manure     & Ar300      & PyOM       & 300       \\\\\n",
       "\tSRR3681436-X & Argiustoll & Manure     & Ar300      & PyOM       & 300       \\\\\n",
       "\tSRR3681437-X & Argiustoll & Manure     & Ar300      & PyOM       & 300       \\\\\n",
       "\tSRR3681438-X & Argiustoll & Manure     & Ar700      & PyOM       & 700       \\\\\n",
       "\tSRR3681439-X & Argiustoll & Manure     & Ar700      & PyOM       & 700       \\\\\n",
       "\tSRR3681440-X & Argiustoll & Manure     & Ar700      & PyOM       & 700       \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | SoilType | Feedstock | Treatment | PyOM | PyOM.Temp | \n",
       "|---|---|---|---|---|---|\n",
       "| SRR3681435-X | Argiustoll | Manure     | Ar300      | PyOM       | 300        | \n",
       "| SRR3681436-X | Argiustoll | Manure     | Ar300      | PyOM       | 300        | \n",
       "| SRR3681437-X | Argiustoll | Manure     | Ar300      | PyOM       | 300        | \n",
       "| SRR3681438-X | Argiustoll | Manure     | Ar700      | PyOM       | 700        | \n",
       "| SRR3681439-X | Argiustoll | Manure     | Ar700      | PyOM       | 700        | \n",
       "| SRR3681440-X | Argiustoll | Manure     | Ar700      | PyOM       | 700        | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "             SoilType   Feedstock Treatment PyOM PyOM.Temp\n",
       "SRR3681435-X Argiustoll Manure    Ar300     PyOM 300      \n",
       "SRR3681436-X Argiustoll Manure    Ar300     PyOM 300      \n",
       "SRR3681437-X Argiustoll Manure    Ar300     PyOM 300      \n",
       "SRR3681438-X Argiustoll Manure    Ar700     PyOM 700      \n",
       "SRR3681439-X Argiustoll Manure    Ar700     PyOM 700      \n",
       "SRR3681440-X Argiustoll Manure    Ar700     PyOM 700      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Author did not include what treatment corresponded with which sample, so hypothetical matches were recorded based on NMDS \n",
    "#of Bray-Curtis and some sample ordering logic\n",
    "\n",
    "##Create a matrix to match samples with likely treatments\n",
    "\n",
    "Dai2017Matrix = c('SRR3681435-X',\"Argiustoll\",\"Manure\", \"Ar300\", \"PyOM\",300,\n",
    "'SRR3681436-X',\"Argiustoll\", \"Manure\",\"Ar300\", \"PyOM\", 300,\n",
    "'SRR3681437-X', \"Argiustoll\",\"Manure\",\"Ar300\", \"PyOM\", 300,\n",
    "'SRR3681438-X', \"Argiustoll\",\"Manure\",\"Ar700\", \"PyOM\", 700,\n",
    "'SRR3681439-X', \"Argiustoll\",\"Manure\",\"Ar700\", \"PyOM\", 700,\n",
    "'SRR3681440-X', \"Argiustoll\",\"Manure\",\"Ar700\", \"PyOM\", 700,\n",
    "'SRR3681441-X', \"Argiustoll\",NA,\"ArCK\", \"Control\", NA,\n",
    "'SRR3681442-X', \"Argiustoll\",NA,\"ArCK\", \"Control\", NA,\n",
    "'SRR3681471-X', \"Argiustoll\",NA,\"ArCK\", \"Control\", NA,\n",
    "'SRR3681472-X',\"PyOM\", \"Manure\",\"PyAr300\", NA, 300,\n",
    "'SRR3681473-X', \"PyOM\",\"Manure\",\"PyAr300\", NA, 300,\n",
    "'SRR3681474-X',\"PyOM\", \"Manure\",\"PyAr300\", NA, 300,\n",
    "'SRR3681475-X', \"PyOM\",\"Manure\",\"PyAr700\", NA, 700,\n",
    "'SRR3681476-X', \"PyOM\",\"Manure\",\"PyAr700\", NA, 700,\n",
    "'SRR3681477-X',\"PyOM\", \"Manure\",\"PyAr700\", NA, 700,\n",
    "'SRR3681478-X',\"Psammaquent\", \"Manure\",\"Ps300\", \"PyOM\", 300,\n",
    "'SRR3681479-X', \"Psammaquent\",\"Manure\",\"Ps300\", \"PyOM\", 300,\n",
    "'SRR3681480-X', \"Psammaquent\",\"Manure\",\"Ps300\", \"PyOM\", 300,\n",
    "'SRR3681481-X',\"Psammaquent\", \"Manure\",\"Ps700\", \"PyOM\", 700,\n",
    "'SRR3681482-X',\"Psammaquent\", \"Manure\",\"Ps700\", \"PyOM\", 700,\n",
    "'SRR3681483-X',\"Psammaquent\", \"Manure\",\"Ps700\", \"PyOM\", 700,\n",
    "'SRR3681484-X', \"Psammaquent\",NA,\"PsCK\", \"Control\", NA,\n",
    "'SRR3681485-X', \"Psammaquent\",NA,\"PsCK\", \"Control\", NA,\n",
    "'SRR3681486-X', \"Psammaquent\",NA,\"PsCK\", \"Control\", NA,\n",
    "'SRR3681488-X', \"PyOM\",\"Manure\",\"PyPs300\", NA, 300,\n",
    "'SRR3681490-X', \"PyOM\",\"Manure\",\"PyPs300\", NA, 300,\n",
    "'SRR3681491-X', \"PyOM\",\"Manure\",\"PyPs300\", NA, 300,\n",
    "'SRR3681492-X', \"PyOM\",\"Manure\",\"PyPs700\", NA, 700,\n",
    "'SRR3681494-X', \"PyOM\",\"Manure\",\"PyPs700\", NA, 700,\n",
    "'SRR3681496-X', \"PyOM\",\"Manure\",\"PyPs700\", NA, 700)\n",
    "\n",
    "Dai2017Matrix = t(matrix(Dai2017Matrix,nrow = 6))\n",
    "colnames(Dai2017Matrix) = c(\"SampleID\",\"SoilType\",\"Feedstock\",\"Treatment\",\"PyOM\",\"PyOM.Temp\")\n",
    "row.names(Dai2017Matrix) = data.frame(Dai2017Matrix)$SampleID\n",
    "Dai2017Matrix = data.frame(Dai2017Matrix[,2:6])\n",
    "#check for success\n",
    "head(Dai2017Matrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in `[<-.data.frame`(x3, i, , ..., value = value):\n",
      "provided 32 variables to replace 31 variables"
     ]
    }
   ],
   "source": [
    "# Fixing the Dai2017 sample data\n",
    "## Paste information from matrix created above into the ps\n",
    "sample_data(ps_dai2017)[row.names(sample_data(ps_dai2017)) %in% row.names(Dai2017Matrix),]$SoilType = paste(Dai2017Matrix$SoilType)\n",
    "sample_data(ps_dai2017)[row.names(sample_data(ps_dai2017)) %in% row.names(Dai2017Matrix),]$Feedstock = paste(Dai2017Matrix$Feedstock)\n",
    "sample_data(ps_dai2017)[row.names(sample_data(ps_dai2017)) %in% row.names(Dai2017Matrix),]$Treatment = paste(Dai2017Matrix$Treatment)\n",
    "sample_data(ps_dai2017)[row.names(sample_data(ps_dai2017)) %in% row.names(Dai2017Matrix),]$PyOM = paste(Dai2017Matrix$PyOM)\n",
    "sample_data(ps_dai2017)[row.names(sample_data(ps_dai2017)) %in% row.names(Dai2017Matrix),]$PyOM.Temp = paste(Dai2017Matrix$PyOM.Temp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_dai2017)$Treatment = revalue(sample_data(ps_dai2017)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_dai2017)$SoilType = revalue(sample_data(ps_dai2017)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_dai2017)$Feedstock = revalue(sample_data(ps_dai2017)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_dai2017)$Habitat = revalue(sample_data(ps_dai2017)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_dai2017)$CurrentLandUse = revalue(sample_data(ps_dai2017)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_dai2017)$PyOM.Temp = revalue(sample_data(ps_dai2017)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_dai2017)$Depth = revalue(sample_data(ps_dai2017)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'g.per.kg.PyOM'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'Crop'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'TargetGene'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'SRA'\n",
       "\\item 'Region'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'g.per.kg.PyOM'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'pH'\n",
       "\\item 'Treatment'\n",
       "\\item 'Author'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Clay'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'Habitat'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'Sand'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'Feedstock'\n",
       "\\item 'BioSample'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'Organism'\n",
       "\\item 'BioProject'\n",
       "\\item 'SoilType'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Silt'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'Crop'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'TargetGene'\n",
       "2. 'CurrentLandUse'\n",
       "3. 'NucleicAcidExtraction'\n",
       "4. 'SequencingMethod'\n",
       "5. 'Incubation_Time_Weeks'\n",
       "6. 'SRA'\n",
       "7. 'Region'\n",
       "8. 'ForwardPrimerName'\n",
       "9. 'g.per.kg.PyOM'\n",
       "10. 'ForwardPrimerSequence'\n",
       "11. 'pH'\n",
       "12. 'Treatment'\n",
       "13. 'Author'\n",
       "14. 'RevPrimerName'\n",
       "15. 'Clay'\n",
       "16. 'PyOM.Temp'\n",
       "17. 'FwdBarcode'\n",
       "18. 'Habitat'\n",
       "19. 'Sampling_Depth_cm'\n",
       "20. 'Sand'\n",
       "21. 'CollectionYear'\n",
       "22. 'Feedstock'\n",
       "23. 'BioSample'\n",
       "24. 'RevBarcode'\n",
       "25. 'Organism'\n",
       "26. 'BioProject'\n",
       "27. 'SoilType'\n",
       "28. 'ReversePrimerSequence'\n",
       "29. 'Silt'\n",
       "30. 'GeographicalLocation'\n",
       "31. 'Crop'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"TargetGene\"            \"CurrentLandUse\"        \"NucleicAcidExtraction\"\n",
       " [4] \"SequencingMethod\"      \"Incubation_Time_Weeks\" \"SRA\"                  \n",
       " [7] \"Region\"                \"ForwardPrimerName\"     \"g.per.kg.PyOM\"        \n",
       "[10] \"ForwardPrimerSequence\" \"pH\"                    \"Treatment\"            \n",
       "[13] \"Author\"                \"RevPrimerName\"         \"Clay\"                 \n",
       "[16] \"PyOM.Temp\"             \"FwdBarcode\"            \"Habitat\"              \n",
       "[19] \"Sampling_Depth_cm\"     \"Sand\"                  \"CollectionYear\"       \n",
       "[22] \"Feedstock\"             \"BioSample\"             \"RevBarcode\"           \n",
       "[25] \"Organism\"              \"BioProject\"            \"SoilType\"             \n",
       "[28] \"ReversePrimerSequence\" \"Silt\"                  \"GeographicalLocation\" \n",
       "[31] \"Crop\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'NA'</li>\n",
       "\t<li>'NA'</li>\n",
       "\t<li>'NA'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'NA'</li>\n",
       "\t<li>'NA'</li>\n",
       "\t<li>'NA'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'300'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "\t<li>'700'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item 'NA'\n",
       "\\item 'NA'\n",
       "\\item 'NA'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item 'NA'\n",
       "\\item 'NA'\n",
       "\\item 'NA'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '300'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\item '700'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '300'\n",
       "2. '300'\n",
       "3. '300'\n",
       "4. '700'\n",
       "5. '700'\n",
       "6. '700'\n",
       "7. 'NA'\n",
       "8. 'NA'\n",
       "9. 'NA'\n",
       "10. '300'\n",
       "11. '300'\n",
       "12. '300'\n",
       "13. '700'\n",
       "14. '700'\n",
       "15. '700'\n",
       "16. '300'\n",
       "17. '300'\n",
       "18. '300'\n",
       "19. '700'\n",
       "20. '700'\n",
       "21. '700'\n",
       "22. 'NA'\n",
       "23. 'NA'\n",
       "24. 'NA'\n",
       "25. '300'\n",
       "26. '300'\n",
       "27. '300'\n",
       "28. '700'\n",
       "29. '700'\n",
       "30. '700'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"300\" \"300\" \"300\" \"700\" \"700\" \"700\" \"NA\"  \"NA\"  \"NA\"  \"300\" \"300\" \"300\"\n",
       "[13] \"700\" \"700\" \"700\" \"300\" \"300\" \"300\" \"700\" \"700\" \"700\" \"NA\"  \"NA\"  \"NA\" \n",
       "[25] \"300\" \"300\" \"300\" \"700\" \"700\" \"700\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. &lt;NA&gt;\n",
       "3. &lt;NA&gt;\n",
       "4. &lt;NA&gt;\n",
       "5. &lt;NA&gt;\n",
       "6. &lt;NA&gt;\n",
       "7. &lt;NA&gt;\n",
       "8. &lt;NA&gt;\n",
       "9. &lt;NA&gt;\n",
       "10. &lt;NA&gt;\n",
       "11. &lt;NA&gt;\n",
       "12. &lt;NA&gt;\n",
       "13. &lt;NA&gt;\n",
       "14. &lt;NA&gt;\n",
       "15. &lt;NA&gt;\n",
       "16. &lt;NA&gt;\n",
       "17. &lt;NA&gt;\n",
       "18. &lt;NA&gt;\n",
       "19. &lt;NA&gt;\n",
       "20. &lt;NA&gt;\n",
       "21. &lt;NA&gt;\n",
       "22. &lt;NA&gt;\n",
       "23. &lt;NA&gt;\n",
       "24. &lt;NA&gt;\n",
       "25. &lt;NA&gt;\n",
       "26. &lt;NA&gt;\n",
       "27. &lt;NA&gt;\n",
       "28. &lt;NA&gt;\n",
       "29. &lt;NA&gt;\n",
       "30. &lt;NA&gt;\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA NA\n",
       "[26] NA NA NA NA NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and g.per.kg.PyOM are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_dai2017))[names(sample_data(ps_dai2017)) == \"PyOM-Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_dai2017))[names(sample_data(ps_dai2017)) == \"g-per-kg-PyOM\"] <- \"g.per.kg.PyOM\"\n",
    "\n",
    "# Changing pH to NA because pH data is the baseline soil pH, not the \"true\" pH at the time of\n",
    "# soil collection (treatment + soil)\n",
    "sample_data(ps_dai2017)$pH = NA\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM', \"Ar300\", \"Ar700\", \"Ps300\", \"Ps700\")\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\", \"ArCK\", \"PsCK\" )\n",
    "OMAdded = c('Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_dai2017)$PyOM = ifelse(sample_data(ps_dai2017)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_dai2017)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_dai2017)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "\t<li>9.6</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\item 9.6\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 7.2\n",
       "2. 7.2\n",
       "3. 7.2\n",
       "4. 9.6\n",
       "5. 9.6\n",
       "6. 9.6\n",
       "7. &lt;NA&gt;\n",
       "8. &lt;NA&gt;\n",
       "9. &lt;NA&gt;\n",
       "10. 7.2\n",
       "11. 7.2\n",
       "12. 7.2\n",
       "13. 9.6\n",
       "14. 9.6\n",
       "15. 9.6\n",
       "16. 7.2\n",
       "17. 7.2\n",
       "18. 7.2\n",
       "19. 9.6\n",
       "20. 9.6\n",
       "21. 9.6\n",
       "22. &lt;NA&gt;\n",
       "23. &lt;NA&gt;\n",
       "24. &lt;NA&gt;\n",
       "25. 7.2\n",
       "26. 7.2\n",
       "27. 7.2\n",
       "28. 9.6\n",
       "29. 9.6\n",
       "30. 9.6\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] 7.2 7.2 7.2 9.6 9.6 9.6  NA  NA  NA 7.2 7.2 7.2 9.6 9.6 9.6 7.2 7.2 7.2 9.6\n",
       "[20] 9.6 9.6  NA  NA  NA 7.2 7.2 7.2 9.6 9.6 9.6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_dai2017)$PyOM.pH = ifelse(sample_data(ps_dai2017)$Treatment %in% \"Ar300\", 7.2,\n",
    "                                         ifelse(sample_data(ps_dai2017)$Treatment %in% \"Ar700\", 9.6,\n",
    "                                                ifelse(sample_data(ps_dai2017)$Treatment %in% \"PyAr300\", 7.2,\n",
    "                                                       ifelse(sample_data(ps_dai2017)$Treatment %in% \"PyAr700\", 9.6,\n",
    "                                                              ifelse(sample_data(ps_dai2017)$Treatment %in% \"Ps300\", 7.2,\n",
    "                                                                     ifelse(sample_data(ps_dai2017)$Treatment %in% \"Ps700\", 9.6,\n",
    "                                                                            ifelse(sample_data(ps_dai2017)$Treatment %in% \"PyPs300\", 7.2,\n",
    "                                                                                   ifelse(sample_data(ps_dai2017)$Treatment %in% \"PyPs700\", 9.6, NA))))))))\n",
    "\n",
    "sample_data(ps_dai2017)$PyOM.pH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_dai2017, \"dai2017.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Dai 2016"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ps_dai2016 = import_biom(\"Dai2016_OTU_table/feature-table-metaD-tax_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>50b1f8869b00b7101fc4b6a263f1e3a2</th><td>Bacteria             </td><td>Firmicutes           </td><td>Bacilli              </td><td>Bacillales           </td><td>NA                   </td><td>NA                   </td><td>NA                   </td></tr>\n",
       "\t<tr><th scope=row>938cda0a26fe3f948909700a8bd71a61</th><td>Bacteria             </td><td>Proteobacteria       </td><td>Gammaproteobacteria  </td><td>Betaproteobacteriales</td><td>Methylophilaceae     </td><td>NA                   </td><td>NA                   </td></tr>\n",
       "\t<tr><th scope=row>270ae734802f97123fcafc2d1d2eff61</th><td>Bacteria             </td><td>Proteobacteria       </td><td>Gammaproteobacteria  </td><td>Betaproteobacteriales</td><td>Rhodocyclaceae       </td><td>NA                   </td><td>NA                   </td></tr>\n",
       "\t<tr><th scope=row>88a09b8bc83628992633f4a96ca83bc5</th><td>Bacteria             </td><td>Actinobacteria       </td><td>Actinobacteria       </td><td>Micrococcales        </td><td>Intrasporangiaceae   </td><td>Terrabacter          </td><td>Ambiguous_taxa       </td></tr>\n",
       "\t<tr><th scope=row>cf5974e3786fe60b4beb8bcae9a14957</th><td>Bacteria             </td><td>Proteobacteria       </td><td>Gammaproteobacteria  </td><td>Betaproteobacteriales</td><td>Burkholderiaceae     </td><td>Ramlibacter          </td><td>NA                   </td></tr>\n",
       "\t<tr><th scope=row>7a44aec2778c5cd208f236de47c87e5f</th><td>Bacteria             </td><td>Proteobacteria       </td><td>Gammaproteobacteria  </td><td>Betaproteobacteriales</td><td>Burkholderiaceae     </td><td>Ramlibacter          </td><td>NA                   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllll}\n",
       "  & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       "\\hline\n",
       "\t50b1f8869b00b7101fc4b6a263f1e3a2 & Bacteria              & Firmicutes            & Bacilli               & Bacillales            & NA                    & NA                    & NA                   \\\\\n",
       "\t938cda0a26fe3f948909700a8bd71a61 & Bacteria              & Proteobacteria        & Gammaproteobacteria   & Betaproteobacteriales & Methylophilaceae      & NA                    & NA                   \\\\\n",
       "\t270ae734802f97123fcafc2d1d2eff61 & Bacteria              & Proteobacteria        & Gammaproteobacteria   & Betaproteobacteriales & Rhodocyclaceae        & NA                    & NA                   \\\\\n",
       "\t88a09b8bc83628992633f4a96ca83bc5 & Bacteria              & Actinobacteria        & Actinobacteria        & Micrococcales         & Intrasporangiaceae    & Terrabacter           & Ambiguous\\_taxa     \\\\\n",
       "\tcf5974e3786fe60b4beb8bcae9a14957 & Bacteria              & Proteobacteria        & Gammaproteobacteria   & Betaproteobacteriales & Burkholderiaceae      & Ramlibacter           & NA                   \\\\\n",
       "\t7a44aec2778c5cd208f236de47c87e5f & Bacteria              & Proteobacteria        & Gammaproteobacteria   & Betaproteobacteriales & Burkholderiaceae      & Ramlibacter           & NA                   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Domain | Phylum | Class | Order | Family | Genus | Species | \n",
       "|---|---|---|---|---|---|\n",
       "| 50b1f8869b00b7101fc4b6a263f1e3a2 | Bacteria              | Firmicutes            | Bacilli               | Bacillales            | NA                    | NA                    | NA                    | \n",
       "| 938cda0a26fe3f948909700a8bd71a61 | Bacteria              | Proteobacteria        | Gammaproteobacteria   | Betaproteobacteriales | Methylophilaceae      | NA                    | NA                    | \n",
       "| 270ae734802f97123fcafc2d1d2eff61 | Bacteria              | Proteobacteria        | Gammaproteobacteria   | Betaproteobacteriales | Rhodocyclaceae        | NA                    | NA                    | \n",
       "| 88a09b8bc83628992633f4a96ca83bc5 | Bacteria              | Actinobacteria        | Actinobacteria        | Micrococcales         | Intrasporangiaceae    | Terrabacter           | Ambiguous_taxa        | \n",
       "| cf5974e3786fe60b4beb8bcae9a14957 | Bacteria              | Proteobacteria        | Gammaproteobacteria   | Betaproteobacteriales | Burkholderiaceae      | Ramlibacter           | NA                    | \n",
       "| 7a44aec2778c5cd208f236de47c87e5f | Bacteria              | Proteobacteria        | Gammaproteobacteria   | Betaproteobacteriales | Burkholderiaceae      | Ramlibacter           | NA                    | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "                                 Domain   Phylum         Class              \n",
       "50b1f8869b00b7101fc4b6a263f1e3a2 Bacteria Firmicutes     Bacilli            \n",
       "938cda0a26fe3f948909700a8bd71a61 Bacteria Proteobacteria Gammaproteobacteria\n",
       "270ae734802f97123fcafc2d1d2eff61 Bacteria Proteobacteria Gammaproteobacteria\n",
       "88a09b8bc83628992633f4a96ca83bc5 Bacteria Actinobacteria Actinobacteria     \n",
       "cf5974e3786fe60b4beb8bcae9a14957 Bacteria Proteobacteria Gammaproteobacteria\n",
       "7a44aec2778c5cd208f236de47c87e5f Bacteria Proteobacteria Gammaproteobacteria\n",
       "                                 Order                 Family            \n",
       "50b1f8869b00b7101fc4b6a263f1e3a2 Bacillales            NA                \n",
       "938cda0a26fe3f948909700a8bd71a61 Betaproteobacteriales Methylophilaceae  \n",
       "270ae734802f97123fcafc2d1d2eff61 Betaproteobacteriales Rhodocyclaceae    \n",
       "88a09b8bc83628992633f4a96ca83bc5 Micrococcales         Intrasporangiaceae\n",
       "cf5974e3786fe60b4beb8bcae9a14957 Betaproteobacteriales Burkholderiaceae  \n",
       "7a44aec2778c5cd208f236de47c87e5f Betaproteobacteriales Burkholderiaceae  \n",
       "                                 Genus       Species       \n",
       "50b1f8869b00b7101fc4b6a263f1e3a2 NA          NA            \n",
       "938cda0a26fe3f948909700a8bd71a61 NA          NA            \n",
       "270ae734802f97123fcafc2d1d2eff61 NA          NA            \n",
       "88a09b8bc83628992633f4a96ca83bc5 Terrabacter Ambiguous_taxa\n",
       "cf5974e3786fe60b4beb8bcae9a14957 Ramlibacter NA            \n",
       "7a44aec2778c5cd208f236de47c87e5f Ramlibacter NA            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x.dai2016 = data.frame(tax_table(ps_dai2016))\n",
    "\n",
    "colnames(x.dai2016) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x.dai2016$Domain = gsub(\"D_0__\", \"\", as.character(x.dai2016$Domain))\n",
    "x.dai2016$Phylum = gsub(\"D_1__\", \"\", as.character(x.dai2016$Phylum))\n",
    "x.dai2016$Class = gsub(\"D_2__\", \"\", as.character(x.dai2016$Class))\n",
    "x.dai2016$Order = gsub(\"D_3__\", \"\", as.character(x.dai2016$Order))\n",
    "x.dai2016$Family = gsub(\"D_4__\", \"\", as.character(x.dai2016$Family))\n",
    "x.dai2016$Genus = gsub(\"D_5__\", \"\", as.character(x.dai2016$Genus))\n",
    "x.dai2016$Species = gsub(\"D_6__\", \"\", as.character(x.dai2016$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x.dai2016=tax_table(as.matrix(x.dai2016,dimnames=list(row.names(x.dai2016),colnames(x.dai2016))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_dai2016)=x.dai2016\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_dai2016))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Taking out a sample that had no sequences \n",
    "ps_dai2016 = prune_samples(sample_names(ps_dai2016)!= \"SRR3452692-X\", ps_dai2016)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SoilType'\n",
       "\\item 'Feedstock'\n",
       "\\item 'Treatment'\n",
       "\\item 'PyOM'\n",
       "\\item 'pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SoilType'\n",
       "2. 'Feedstock'\n",
       "3. 'Treatment'\n",
       "4. 'PyOM'\n",
       "5. 'pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"SoilType\"  \"Feedstock\" \"Treatment\" \"PyOM\"      \"pH\"       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Author didn't include treatment for each sample, so below are the hypothetical matches, based on relative \n",
    "#abundances in samples\n",
    "# Indicators: CK soils have higher acidos and lower bacteroidetes\n",
    "# Indicators: Ma char has higher Firmicutes than St char\n",
    "# BuCK has higher BetaProteo than RhCK\n",
    "# Should be able to figure it out from that.\n",
    "# Sample IDs do seem to be in order of treatment (group in 3's sort of)\n",
    "\n",
    "Dai2016Matrix = c('SRR3466117-X', \"Inceptisol\", NA, \"BuCK\", \"Control\", 5.33,\n",
    "                  'SRR3466120-X', \"Inceptisol\", NA, \"BuCK\", \"Control\", 5.33,\n",
    "                  'SRR3466121-X', \"Inceptisol\", NA, \"BuCK\", \"Control\", 5.33,\n",
    "                  'SRR3466122-X', \"Inceptisol\", \"Manure\", \"BuMa\", \"PyOM\", 6.64,\n",
    "                  'SRR3466123-X', \"Inceptisol\", \"Manure\", \"BuMa\", \"PyOM\",  6.64,\n",
    "                  'SRR3466124-X', \"Inceptisol\", \"Manure\", \"BuMa\", \"PyOM\",  6.64,\n",
    "                  'SRR3466125-X', \"Inceptisol\", \"Stover\", \"BuSt\", \"PyOM\", 5.99,\n",
    "                  'SRR3466126-X', \"Inceptisol\", \"Stover\", \"BuSt\", \"PyOM\", 5.99,\n",
    "                  'SRR3466127-X', \"Inceptisol\", \"Stover\", \"BuSt\", \"PyOM\", 5.99,\n",
    "                  'SRR3466128-X', \"Inceptisol\", NA, \"RhCK\", \"Control\", 5.13,\n",
    "                  'SRR3466129-X', \"Inceptisol\", NA, \"RhCK\", \"Control\", 5.13,\n",
    "                  'SRR3466130-X', \"Inceptisol\", NA, \"RhCK\", \"Control\", 5.13,\n",
    "                  'SRR3466131-X', \"Inceptisol\", \"Manure\", \"RhMa\", \"PyOM\", 6.94,\n",
    "                  'SRR3466134-X', \"Inceptisol\", \"Manure\", \"RhMa\", \"PyOM\", 6.94,\n",
    "                  'SRR3466136-X', \"Inceptisol\", \"Manure\", \"RhMa\", \"PyOM\", 6.94,\n",
    "                  'SRR3466137-X', \"Inceptisol\", \"Stover\", \"RhSt\", \"PyOM\", 6.11,\n",
    "                  'SRR3466138-X', \"Inceptisol\", \"Stover\", \"RhSt\", \"PyOM\", 6.11,\n",
    "                  'SRR3466140-X', \"Inceptisol\", \"Stover\", \"RhSt\", \"PyOM\", 6.11)\n",
    "Dai2016Matrix = t(matrix(Dai2016Matrix,nrow = 6))\n",
    "colnames(Dai2016Matrix) = c(\"SampleID\",\"SoilType\",\"Feedstock\",\"Treatment\",\"PyOM\",\"pH\")\n",
    "row.names(Dai2016Matrix) = data.frame(Dai2016Matrix)$SampleID\n",
    "Dai2016Matrix = data.frame(Dai2016Matrix[,2:6])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in `[<-.data.frame`(x3, i, , ..., value = value):\n",
      "provided 31 variables to replace 30 variables"
     ]
    }
   ],
   "source": [
    "# Fixing the Dai2016 sample data\n",
    "## Paste information from matrix created above into the ps\n",
    "\n",
    "sample_data(ps_dai2016)[row.names(sample_data(ps_dai2016)) %in% row.names(Dai2016Matrix),]$SoilType = paste(Dai2016Matrix$SoilType)\n",
    "sample_data(ps_dai2016)[row.names(sample_data(ps_dai2016)) %in% row.names(Dai2016Matrix),]$Feedstock = paste(Dai2016Matrix$Feedstock)\n",
    "sample_data(ps_dai2016)[row.names(sample_data(ps_dai2016)) %in% row.names(Dai2016Matrix),]$Treatment = paste(Dai2016Matrix$Treatment)\n",
    "sample_data(ps_dai2016)[row.names(sample_data(ps_dai2016)) %in% row.names(Dai2016Matrix),]$PyOM = paste(Dai2016Matrix$PyOM)\n",
    "sample_data(ps_dai2016)[row.names(sample_data(ps_dai2016)) %in% row.names(Dai2016Matrix),]$pH = paste(Dai2016Matrix$pH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_dai2016)$Treatment = revalue(sample_data(ps_dai2016)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_dai2016)$SoilType = revalue(sample_data(ps_dai2016)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_dai2016)$Feedstock = revalue(sample_data(ps_dai2016)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_dai2016)$Habitat = revalue(sample_data(ps_dai2016)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_dai2016)$CurrentLandUse = revalue(sample_data(ps_dai2016)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_dai2016)$PyOM.Temp = revalue(sample_data(ps_dai2016)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_dai2016)$Depth = revalue(sample_data(ps_dai2016)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM', \"BuMa\", \"BuSt\", \"RhMa\", \"RhSt\")\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\", \"BuCK\", \"RhCK\")\n",
    "OMAdded = c('Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_dai2016)$PyOM = ifelse(sample_data(ps_dai2016)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_dai2016)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_dai2016)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>9.55</li>\n",
       "\t<li>9.55</li>\n",
       "\t<li>9.55</li>\n",
       "\t<li>10.09</li>\n",
       "\t<li>10.09</li>\n",
       "\t<li>10.09</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>9.55</li>\n",
       "\t<li>9.55</li>\n",
       "\t<li>9.55</li>\n",
       "\t<li>10.09</li>\n",
       "\t<li>10.09</li>\n",
       "\t<li>10.09</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 9.55\n",
       "\\item 9.55\n",
       "\\item 9.55\n",
       "\\item 10.09\n",
       "\\item 10.09\n",
       "\\item 10.09\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 9.55\n",
       "\\item 9.55\n",
       "\\item 9.55\n",
       "\\item 10.09\n",
       "\\item 10.09\n",
       "\\item 10.09\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. &lt;NA&gt;\n",
       "3. &lt;NA&gt;\n",
       "4. 9.55\n",
       "5. 9.55\n",
       "6. 9.55\n",
       "7. 10.09\n",
       "8. 10.09\n",
       "9. 10.09\n",
       "10. &lt;NA&gt;\n",
       "11. &lt;NA&gt;\n",
       "12. &lt;NA&gt;\n",
       "13. 9.55\n",
       "14. 9.55\n",
       "15. 9.55\n",
       "16. 10.09\n",
       "17. 10.09\n",
       "18. 10.09\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1]    NA    NA    NA  9.55  9.55  9.55 10.09 10.09 10.09    NA    NA    NA\n",
       "[13]  9.55  9.55  9.55 10.09 10.09 10.09"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_dai2016)$PyOM.pH = ifelse(sample_data(ps_dai2016)$Feedstock %in% \"Manure\", 9.55,\n",
    "                                         ifelse(sample_data(ps_dai2016)$Feedstock %in% \"Stover\", 10.09, NA))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "ps_dai2016 = readRDS (file = \"dai2016.ps\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'g.per.kg.PyOM'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Clay'\n",
       "\\item 'Feedstock'\n",
       "\\item 'Organism'\n",
       "\\item 'SRA'\n",
       "\\item 'Region'\n",
       "\\item 'BioProject'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'Sand'\n",
       "\\item 'Treatment'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'Author'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'TargetGene'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'Habitat'\n",
       "\\item 'g.per.kg.PyOM'\n",
       "\\item 'pH'\n",
       "\\item 'Crop'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'SoilType'\n",
       "\\item 'BioSample'\n",
       "\\item 'Silt'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'PyOM'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Clay'\n",
       "2. 'Feedstock'\n",
       "3. 'Organism'\n",
       "4. 'SRA'\n",
       "5. 'Region'\n",
       "6. 'BioProject'\n",
       "7. 'FwdBarcode'\n",
       "8. 'ReversePrimerSequence'\n",
       "9. 'CollectionYear'\n",
       "10. 'ForwardPrimerSequence'\n",
       "11. 'ForwardPrimerName'\n",
       "12. 'RevPrimerName'\n",
       "13. 'CurrentLandUse'\n",
       "14. 'Sand'\n",
       "15. 'Treatment'\n",
       "16. 'GeographicalLocation'\n",
       "17. 'Author'\n",
       "18. 'PyOM.Temp'\n",
       "19. 'TargetGene'\n",
       "20. 'RevBarcode'\n",
       "21. 'Habitat'\n",
       "22. 'g.per.kg.PyOM'\n",
       "23. 'pH'\n",
       "24. 'Crop'\n",
       "25. 'Sampling_Depth_cm'\n",
       "26. 'SequencingMethod'\n",
       "27. 'SoilType'\n",
       "28. 'BioSample'\n",
       "29. 'Silt'\n",
       "30. 'NucleicAcidExtraction'\n",
       "31. 'PyOM'\n",
       "32. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Clay\"                  \"Feedstock\"             \"Organism\"             \n",
       " [4] \"SRA\"                   \"Region\"                \"BioProject\"           \n",
       " [7] \"FwdBarcode\"            \"ReversePrimerSequence\" \"CollectionYear\"       \n",
       "[10] \"ForwardPrimerSequence\" \"ForwardPrimerName\"     \"RevPrimerName\"        \n",
       "[13] \"CurrentLandUse\"        \"Sand\"                  \"Treatment\"            \n",
       "[16] \"GeographicalLocation\"  \"Author\"                \"PyOM.Temp\"            \n",
       "[19] \"TargetGene\"            \"RevBarcode\"            \"Habitat\"              \n",
       "[22] \"g.per.kg.PyOM\"         \"pH\"                    \"Crop\"                 \n",
       "[25] \"Sampling_Depth_cm\"     \"SequencingMethod\"      \"SoilType\"             \n",
       "[28] \"BioSample\"             \"Silt\"                  \"NucleicAcidExtraction\"\n",
       "[31] \"PyOM\"                  \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "colnames(sample_data(ps_dai2016))\n",
    "sample_data(ps_dai2016)$Incubation_Time_Weeks = "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_dai2016, \"dai2016.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Yao"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ps_yao = import_biom(\"YaoBact_OTU_table/feature-table-metaD-tax_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>c2cdaf01de4b2ce2e02c78559acdff90</th><td>Bacteria                   </td><td>Actinobacteria             </td><td>Actinobacteria             </td><td>Micrococcales              </td><td>Micrococcaceae             </td><td>NA                         </td><td>NA                         </td></tr>\n",
       "\t<tr><th scope=row>9b64bd7579ea761af3631fb70b05724d</th><td>Bacteria                   </td><td>Acidobacteria              </td><td>Blastocatellia (Subgroup 4)</td><td>Pyrinomonadales            </td><td>Pyrinomonadaceae           </td><td>RB41                       </td><td>NA                         </td></tr>\n",
       "\t<tr><th scope=row>985d0b63e502e950e3d6a750178187f5</th><td>Bacteria                   </td><td>Proteobacteria             </td><td>Alphaproteobacteria        </td><td>Rhizobiales                </td><td>Xanthobacteraceae          </td><td>Bradyrhizobium             </td><td>NA                         </td></tr>\n",
       "\t<tr><th scope=row>75538196200bee33d344705bce5c6be0</th><td>Bacteria                   </td><td>Chloroflexi                </td><td>Chloroflexia               </td><td>Thermomicrobiales          </td><td>JG30-KF-CM45               </td><td>uncultured bacterium       </td><td>uncultured bacterium       </td></tr>\n",
       "\t<tr><th scope=row>6b45812af17b96798fdf1fbb314fc383</th><td>Bacteria                   </td><td>Proteobacteria             </td><td>Gammaproteobacteria        </td><td>Betaproteobacteriales      </td><td>Nitrosomonadaceae          </td><td>Ellin6067                  </td><td>uncultured bacterium       </td></tr>\n",
       "\t<tr><th scope=row>2ff7057427130406645db002539545b1</th><td>Bacteria                   </td><td>Acidobacteria              </td><td>Subgroup 6                 </td><td>NA                         </td><td>NA                         </td><td>NA                         </td><td>NA                         </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllll}\n",
       "  & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       "\\hline\n",
       "\tc2cdaf01de4b2ce2e02c78559acdff90 & Bacteria                    & Actinobacteria              & Actinobacteria              & Micrococcales               & Micrococcaceae              & NA                          & NA                         \\\\\n",
       "\t9b64bd7579ea761af3631fb70b05724d & Bacteria                    & Acidobacteria               & Blastocatellia (Subgroup 4) & Pyrinomonadales             & Pyrinomonadaceae            & RB41                        & NA                         \\\\\n",
       "\t985d0b63e502e950e3d6a750178187f5 & Bacteria                    & Proteobacteria              & Alphaproteobacteria         & Rhizobiales                 & Xanthobacteraceae           & Bradyrhizobium              & NA                         \\\\\n",
       "\t75538196200bee33d344705bce5c6be0 & Bacteria                    & Chloroflexi                 & Chloroflexia                & Thermomicrobiales           & JG30-KF-CM45                & uncultured bacterium        & uncultured bacterium       \\\\\n",
       "\t6b45812af17b96798fdf1fbb314fc383 & Bacteria                    & Proteobacteria              & Gammaproteobacteria         & Betaproteobacteriales       & Nitrosomonadaceae           & Ellin6067                   & uncultured bacterium       \\\\\n",
       "\t2ff7057427130406645db002539545b1 & Bacteria                    & Acidobacteria               & Subgroup 6                  & NA                          & NA                          & NA                          & NA                         \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Domain | Phylum | Class | Order | Family | Genus | Species | \n",
       "|---|---|---|---|---|---|\n",
       "| c2cdaf01de4b2ce2e02c78559acdff90 | Bacteria                    | Actinobacteria              | Actinobacteria              | Micrococcales               | Micrococcaceae              | NA                          | NA                          | \n",
       "| 9b64bd7579ea761af3631fb70b05724d | Bacteria                    | Acidobacteria               | Blastocatellia (Subgroup 4) | Pyrinomonadales             | Pyrinomonadaceae            | RB41                        | NA                          | \n",
       "| 985d0b63e502e950e3d6a750178187f5 | Bacteria                    | Proteobacteria              | Alphaproteobacteria         | Rhizobiales                 | Xanthobacteraceae           | Bradyrhizobium              | NA                          | \n",
       "| 75538196200bee33d344705bce5c6be0 | Bacteria                    | Chloroflexi                 | Chloroflexia                | Thermomicrobiales           | JG30-KF-CM45                | uncultured bacterium        | uncultured bacterium        | \n",
       "| 6b45812af17b96798fdf1fbb314fc383 | Bacteria                    | Proteobacteria              | Gammaproteobacteria         | Betaproteobacteriales       | Nitrosomonadaceae           | Ellin6067                   | uncultured bacterium        | \n",
       "| 2ff7057427130406645db002539545b1 | Bacteria                    | Acidobacteria               | Subgroup 6                  | NA                          | NA                          | NA                          | NA                          | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "                                 Domain   Phylum        \n",
       "c2cdaf01de4b2ce2e02c78559acdff90 Bacteria Actinobacteria\n",
       "9b64bd7579ea761af3631fb70b05724d Bacteria Acidobacteria \n",
       "985d0b63e502e950e3d6a750178187f5 Bacteria Proteobacteria\n",
       "75538196200bee33d344705bce5c6be0 Bacteria Chloroflexi   \n",
       "6b45812af17b96798fdf1fbb314fc383 Bacteria Proteobacteria\n",
       "2ff7057427130406645db002539545b1 Bacteria Acidobacteria \n",
       "                                 Class                      \n",
       "c2cdaf01de4b2ce2e02c78559acdff90 Actinobacteria             \n",
       "9b64bd7579ea761af3631fb70b05724d Blastocatellia (Subgroup 4)\n",
       "985d0b63e502e950e3d6a750178187f5 Alphaproteobacteria        \n",
       "75538196200bee33d344705bce5c6be0 Chloroflexia               \n",
       "6b45812af17b96798fdf1fbb314fc383 Gammaproteobacteria        \n",
       "2ff7057427130406645db002539545b1 Subgroup 6                 \n",
       "                                 Order                 Family           \n",
       "c2cdaf01de4b2ce2e02c78559acdff90 Micrococcales         Micrococcaceae   \n",
       "9b64bd7579ea761af3631fb70b05724d Pyrinomonadales       Pyrinomonadaceae \n",
       "985d0b63e502e950e3d6a750178187f5 Rhizobiales           Xanthobacteraceae\n",
       "75538196200bee33d344705bce5c6be0 Thermomicrobiales     JG30-KF-CM45     \n",
       "6b45812af17b96798fdf1fbb314fc383 Betaproteobacteriales Nitrosomonadaceae\n",
       "2ff7057427130406645db002539545b1 NA                    NA               \n",
       "                                 Genus                Species             \n",
       "c2cdaf01de4b2ce2e02c78559acdff90 NA                   NA                  \n",
       "9b64bd7579ea761af3631fb70b05724d RB41                 NA                  \n",
       "985d0b63e502e950e3d6a750178187f5 Bradyrhizobium       NA                  \n",
       "75538196200bee33d344705bce5c6be0 uncultured bacterium uncultured bacterium\n",
       "6b45812af17b96798fdf1fbb314fc383 Ellin6067            uncultured bacterium\n",
       "2ff7057427130406645db002539545b1 NA                   NA                  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x = data.frame(tax_table(ps_yao))\n",
    "\n",
    "colnames(x) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x$Domain = gsub(\"D_0__\", \"\", as.character(x$Domain))\n",
    "x$Phylum = gsub(\"D_1__\", \"\", as.character(x$Phylum))\n",
    "x$Class = gsub(\"D_2__\", \"\", as.character(x$Class))\n",
    "x$Order = gsub(\"D_3__\", \"\", as.character(x$Order))\n",
    "x$Family = gsub(\"D_4__\", \"\", as.character(x$Family))\n",
    "x$Genus = gsub(\"D_5__\", \"\", as.character(x$Genus))\n",
    "x$Species = gsub(\"D_6__\", \"\", as.character(x$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x=tax_table(as.matrix(x,dimnames=list(row.names(x),colnames(x))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_yao)=x\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_yao))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended, unamended\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_yao)$Treatment = revalue(sample_data(ps_yao)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_yao)$SoilType = revalue(sample_data(ps_yao)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_yao)$Feedstock = revalue(sample_data(ps_yao)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_yao)$Habitat = revalue(sample_data(ps_yao)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_yao)$CurrentLandUse = revalue(sample_data(ps_yao)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_yao)$PyOM.Temp = revalue(sample_data(ps_yao)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_yao)$Depth = revalue(sample_data(ps_yao)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'CollectionDate'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'TotalC'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'TotalN'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CollectionDate'\n",
       "\\item 'TargetGene'\n",
       "\\item 'Organism'\n",
       "\\item 'Sand'\n",
       "\\item 'Silt'\n",
       "\\item 'Clay'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'Region'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'Treatment'\n",
       "\\item 'SoilType'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'Habitat'\n",
       "\\item 'SRA'\n",
       "\\item 'TotalC'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'Feedstock'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'BioProject'\n",
       "\\item 'pH'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'Crop'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'TotalN'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Author'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'BioSample'\n",
       "\\item 'PyOM'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CollectionDate'\n",
       "2. 'TargetGene'\n",
       "3. 'Organism'\n",
       "4. 'Sand'\n",
       "5. 'Silt'\n",
       "6. 'Clay'\n",
       "7. 'FwdBarcode'\n",
       "8. 'Region'\n",
       "9. 'RevBarcode'\n",
       "10. 'Treatment'\n",
       "11. 'SoilType'\n",
       "12. 'Sampling_Depth_cm'\n",
       "13. 'Habitat'\n",
       "14. 'SRA'\n",
       "15. 'TotalC'\n",
       "16. 't.per.ha.PyOM'\n",
       "17. 'SequencingMethod'\n",
       "18. 'ForwardPrimerSequence'\n",
       "19. 'Feedstock'\n",
       "20. 'ForwardPrimerName'\n",
       "21. 'BioProject'\n",
       "22. 'pH'\n",
       "23. 'NucleicAcidExtraction'\n",
       "24. 'CurrentLandUse'\n",
       "25. 'RevPrimerName'\n",
       "26. 'PyOM.Temp'\n",
       "27. 'Crop'\n",
       "28. 'GeographicalLocation'\n",
       "29. 'TotalN'\n",
       "30. 'ReversePrimerSequence'\n",
       "31. 'Author'\n",
       "32. 'Incubation_Time_Weeks'\n",
       "33. 'BioSample'\n",
       "34. 'PyOM'\n",
       "35. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"CollectionDate\"        \"TargetGene\"            \"Organism\"             \n",
       " [4] \"Sand\"                  \"Silt\"                  \"Clay\"                 \n",
       " [7] \"FwdBarcode\"            \"Region\"                \"RevBarcode\"           \n",
       "[10] \"Treatment\"             \"SoilType\"              \"Sampling_Depth_cm\"    \n",
       "[13] \"Habitat\"               \"SRA\"                   \"TotalC\"               \n",
       "[16] \"t.per.ha.PyOM\"         \"SequencingMethod\"      \"ForwardPrimerSequence\"\n",
       "[19] \"Feedstock\"             \"ForwardPrimerName\"     \"BioProject\"           \n",
       "[22] \"pH\"                    \"NucleicAcidExtraction\" \"CurrentLandUse\"       \n",
       "[25] \"RevPrimerName\"         \"PyOM.Temp\"             \"Crop\"                 \n",
       "[28] \"GeographicalLocation\"  \"TotalN\"                \"ReversePrimerSequence\"\n",
       "[31] \"Author\"                \"Incubation_Time_Weeks\" \"BioSample\"            \n",
       "[34] \"PyOM\"                  \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'CollectionDate'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'TotalC'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'TotalN'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CollectionDate'\n",
       "\\item 'TargetGene'\n",
       "\\item 'Organism'\n",
       "\\item 'Sand'\n",
       "\\item 'Silt'\n",
       "\\item 'Clay'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'Region'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'Treatment'\n",
       "\\item 'SoilType'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'Habitat'\n",
       "\\item 'SRA'\n",
       "\\item 'TotalC'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'Feedstock'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'BioProject'\n",
       "\\item 'pH'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'Crop'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'TotalN'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Author'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'BioSample'\n",
       "\\item 'PyOM'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CollectionDate'\n",
       "2. 'TargetGene'\n",
       "3. 'Organism'\n",
       "4. 'Sand'\n",
       "5. 'Silt'\n",
       "6. 'Clay'\n",
       "7. 'FwdBarcode'\n",
       "8. 'Region'\n",
       "9. 'RevBarcode'\n",
       "10. 'Treatment'\n",
       "11. 'SoilType'\n",
       "12. 'Sampling_Depth_cm'\n",
       "13. 'Habitat'\n",
       "14. 'SRA'\n",
       "15. 'TotalC'\n",
       "16. 't.per.ha.PyOM'\n",
       "17. 'SequencingMethod'\n",
       "18. 'ForwardPrimerSequence'\n",
       "19. 'Feedstock'\n",
       "20. 'ForwardPrimerName'\n",
       "21. 'BioProject'\n",
       "22. 'pH'\n",
       "23. 'NucleicAcidExtraction'\n",
       "24. 'CurrentLandUse'\n",
       "25. 'RevPrimerName'\n",
       "26. 'PyOM.Temp'\n",
       "27. 'Crop'\n",
       "28. 'GeographicalLocation'\n",
       "29. 'TotalN'\n",
       "30. 'ReversePrimerSequence'\n",
       "31. 'Author'\n",
       "32. 'Incubation_Time_Weeks'\n",
       "33. 'BioSample'\n",
       "34. 'PyOM'\n",
       "35. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"CollectionDate\"        \"TargetGene\"            \"Organism\"             \n",
       " [4] \"Sand\"                  \"Silt\"                  \"Clay\"                 \n",
       " [7] \"FwdBarcode\"            \"Region\"                \"RevBarcode\"           \n",
       "[10] \"Treatment\"             \"SoilType\"              \"Sampling_Depth_cm\"    \n",
       "[13] \"Habitat\"               \"SRA\"                   \"TotalC\"               \n",
       "[16] \"t.per.ha.PyOM\"         \"SequencingMethod\"      \"ForwardPrimerSequence\"\n",
       "[19] \"Feedstock\"             \"ForwardPrimerName\"     \"BioProject\"           \n",
       "[22] \"pH\"                    \"NucleicAcidExtraction\" \"CurrentLandUse\"       \n",
       "[25] \"RevPrimerName\"         \"PyOM.Temp\"             \"Crop\"                 \n",
       "[28] \"GeographicalLocation\"  \"TotalN\"                \"ReversePrimerSequence\"\n",
       "[31] \"Author\"                \"Incubation_Time_Weeks\" \"BioSample\"            \n",
       "[34] \"PyOM\"                  \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and t.per.ha.PyOM are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_yao))[names(sample_data(ps_yao)) == \"PyOM_Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_yao))[names(sample_data(ps_yao)) == \"t-per-ha-PyOM\"] <- \"t.per.ha.PyOM\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\")\n",
    "OMAdded = c('Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_yao)$PyOM = ifelse(sample_data(ps_yao)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_yao)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_yao)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "\t<li>8.87</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\item 8.87\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. &lt;NA&gt;\n",
       "3. &lt;NA&gt;\n",
       "4. 8.87\n",
       "5. 8.87\n",
       "6. 8.87\n",
       "7. 8.87\n",
       "8. 8.87\n",
       "9. 8.87\n",
       "10. 8.87\n",
       "11. 8.87\n",
       "12. 8.87\n",
       "13. &lt;NA&gt;\n",
       "14. &lt;NA&gt;\n",
       "15. &lt;NA&gt;\n",
       "16. 8.87\n",
       "17. 8.87\n",
       "18. 8.87\n",
       "19. 8.87\n",
       "20. 8.87\n",
       "21. 8.87\n",
       "22. 8.87\n",
       "23. 8.87\n",
       "24. 8.87\n",
       "25. &lt;NA&gt;\n",
       "26. &lt;NA&gt;\n",
       "27. &lt;NA&gt;\n",
       "28. 8.87\n",
       "29. 8.87\n",
       "30. 8.87\n",
       "31. 8.87\n",
       "32. 8.87\n",
       "33. 8.87\n",
       "34. 8.87\n",
       "35. 8.87\n",
       "36. 8.87\n",
       "37. &lt;NA&gt;\n",
       "38. &lt;NA&gt;\n",
       "39. &lt;NA&gt;\n",
       "40. 8.87\n",
       "41. 8.87\n",
       "42. 8.87\n",
       "43. 8.87\n",
       "44. 8.87\n",
       "45. 8.87\n",
       "46. 8.87\n",
       "47. 8.87\n",
       "48. 8.87\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1]   NA   NA   NA 8.87 8.87 8.87 8.87 8.87 8.87 8.87 8.87 8.87   NA   NA   NA\n",
       "[16] 8.87 8.87 8.87 8.87 8.87 8.87 8.87 8.87 8.87   NA   NA   NA 8.87 8.87 8.87\n",
       "[31] 8.87 8.87 8.87 8.87 8.87 8.87   NA   NA   NA 8.87 8.87 8.87 8.87 8.87 8.87\n",
       "[46] 8.87 8.87 8.87"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_yao)$PyOM.pH = ifelse(sample_data(ps_yao)$Treatment %in% \"biochar\", 8.87, NA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#make pH a numeric value\n",
    "\n",
    "sample_data(ps_yao)$pH = as.numeric(sample_data(ps_yao)$pH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_yao, \"yao.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Song"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [],
   "source": [
    "ps_song = import_biom(\"Song_OTU_table/feature-table-metaD-tax-assume_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x.song = data.frame(tax_table(ps_song))\n",
    "\n",
    "colnames(x.song) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x.song$Domain = gsub(\"D_0__\", \"\", as.character(x.song$Domain))\n",
    "x.song$Phylum = gsub(\"D_1__\", \"\", as.character(x.song$Phylum))\n",
    "x.song$Class = gsub(\"D_2__\", \"\", as.character(x.song$Class))\n",
    "x.song$Order = gsub(\"D_3__\", \"\", as.character(x.song$Order))\n",
    "x.song$Family = gsub(\"D_4__\", \"\", as.character(x.song$Family))\n",
    "x.song$Genus = gsub(\"D_5__\", \"\", as.character(x.song$Genus))\n",
    "x.song$Species = gsub(\"D_6__\", \"\", as.character(x.song$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x.song=tax_table(as.matrix(x.song,dimnames=list(row.names(x.song),colnames(x.song))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_song)=x.song\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_song))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>SampleID</th><th scope=col>Incubation_Time_Weeks</th><th scope=col>pH</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>SRR3984709-X</th><td>SRR3984709-X</td><td>0           </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984710-X</th><td>SRR3984710-X</td><td>0           </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984721-X</th><td>SRR3984721-X</td><td>4           </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984743-X</th><td>SRR3984743-X</td><td>12          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984755-X</th><td>SRR3984755-X</td><td>12          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984767-X</th><td>SRR3984767-X</td><td>24          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984779-X</th><td>SRR3984779-X</td><td>24          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984792-X</th><td>SRR3984792-X</td><td>0           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984793-X</th><td>SRR3984793-X</td><td>0           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984711-X</th><td>SRR3984711-X</td><td>4           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984712-X</th><td>SRR3984712-X</td><td>4           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984713-X</th><td>SRR3984713-X</td><td>12          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984714-X</th><td>SRR3984714-X</td><td>12          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984715-X</th><td>SRR3984715-X</td><td>24          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984716-X</th><td>SRR3984716-X</td><td>24          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984717-X</th><td>SRR3984717-X</td><td>0           </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984719-X</th><td>SRR3984719-X</td><td>4           </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984720-X</th><td>SRR3984720-X</td><td>4           </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984722-X</th><td>SRR3984722-X</td><td>12          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984723-X</th><td>SRR3984723-X</td><td>12          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984725-X</th><td>SRR3984725-X</td><td>24          </td><td>7.5         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984726-X</th><td>SRR3984726-X</td><td>0           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984727-X</th><td>SRR3984727-X</td><td>0           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984728-X</th><td>SRR3984728-X</td><td>4           </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984730-X</th><td>SRR3984730-X</td><td>12          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984731-X</th><td>SRR3984731-X</td><td>12          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984733-X</th><td>SRR3984733-X</td><td>24          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984734-X</th><td>SRR3984734-X</td><td>24          </td><td>7.4         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984735-X</th><td>SRR3984735-X</td><td>0           </td><td>7.6         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984736-X</th><td>SRR3984736-X</td><td>0           </td><td>7.6         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984737-X</th><td>SRR3984737-X</td><td>4           </td><td>7.6         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984738-X</th><td>SRR3984738-X</td><td>4           </td><td>7.6         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984740-X</th><td>SRR3984740-X</td><td>12          </td><td>7.6         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984741-X</th><td>SRR3984741-X</td><td>24          </td><td>7.6         </td></tr>\n",
       "\t<tr><th scope=row>SRR3984742-X</th><td>SRR3984742-X</td><td>24          </td><td>7.6         </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lll}\n",
       "  & SampleID & Incubation\\_Time\\_Weeks & pH\\\\\n",
       "\\hline\n",
       "\tSRR3984709-X & SRR3984709-X & 0            & 7.5         \\\\\n",
       "\tSRR3984710-X & SRR3984710-X & 0            & 7.5         \\\\\n",
       "\tSRR3984721-X & SRR3984721-X & 4            & 7.5         \\\\\n",
       "\tSRR3984743-X & SRR3984743-X & 12           & 7.5         \\\\\n",
       "\tSRR3984755-X & SRR3984755-X & 12           & 7.5         \\\\\n",
       "\tSRR3984767-X & SRR3984767-X & 24           & 7.5         \\\\\n",
       "\tSRR3984779-X & SRR3984779-X & 24           & 7.5         \\\\\n",
       "\tSRR3984792-X & SRR3984792-X & 0            & 7.4         \\\\\n",
       "\tSRR3984793-X & SRR3984793-X & 0            & 7.4         \\\\\n",
       "\tSRR3984711-X & SRR3984711-X & 4            & 7.4         \\\\\n",
       "\tSRR3984712-X & SRR3984712-X & 4            & 7.4         \\\\\n",
       "\tSRR3984713-X & SRR3984713-X & 12           & 7.4         \\\\\n",
       "\tSRR3984714-X & SRR3984714-X & 12           & 7.4         \\\\\n",
       "\tSRR3984715-X & SRR3984715-X & 24           & 7.4         \\\\\n",
       "\tSRR3984716-X & SRR3984716-X & 24           & 7.4         \\\\\n",
       "\tSRR3984717-X & SRR3984717-X & 0            & 7.5         \\\\\n",
       "\tSRR3984719-X & SRR3984719-X & 4            & 7.5         \\\\\n",
       "\tSRR3984720-X & SRR3984720-X & 4            & 7.5         \\\\\n",
       "\tSRR3984722-X & SRR3984722-X & 12           & 7.5         \\\\\n",
       "\tSRR3984723-X & SRR3984723-X & 12           & 7.5         \\\\\n",
       "\tSRR3984725-X & SRR3984725-X & 24           & 7.5         \\\\\n",
       "\tSRR3984726-X & SRR3984726-X & 0            & 7.4         \\\\\n",
       "\tSRR3984727-X & SRR3984727-X & 0            & 7.4         \\\\\n",
       "\tSRR3984728-X & SRR3984728-X & 4            & 7.4         \\\\\n",
       "\tSRR3984730-X & SRR3984730-X & 12           & 7.4         \\\\\n",
       "\tSRR3984731-X & SRR3984731-X & 12           & 7.4         \\\\\n",
       "\tSRR3984733-X & SRR3984733-X & 24           & 7.4         \\\\\n",
       "\tSRR3984734-X & SRR3984734-X & 24           & 7.4         \\\\\n",
       "\tSRR3984735-X & SRR3984735-X & 0            & 7.6         \\\\\n",
       "\tSRR3984736-X & SRR3984736-X & 0            & 7.6         \\\\\n",
       "\tSRR3984737-X & SRR3984737-X & 4            & 7.6         \\\\\n",
       "\tSRR3984738-X & SRR3984738-X & 4            & 7.6         \\\\\n",
       "\tSRR3984740-X & SRR3984740-X & 12           & 7.6         \\\\\n",
       "\tSRR3984741-X & SRR3984741-X & 24           & 7.6         \\\\\n",
       "\tSRR3984742-X & SRR3984742-X & 24           & 7.6         \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | SampleID | Incubation_Time_Weeks | pH | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR3984709-X | SRR3984709-X | 0            | 7.5          | \n",
       "| SRR3984710-X | SRR3984710-X | 0            | 7.5          | \n",
       "| SRR3984721-X | SRR3984721-X | 4            | 7.5          | \n",
       "| SRR3984743-X | SRR3984743-X | 12           | 7.5          | \n",
       "| SRR3984755-X | SRR3984755-X | 12           | 7.5          | \n",
       "| SRR3984767-X | SRR3984767-X | 24           | 7.5          | \n",
       "| SRR3984779-X | SRR3984779-X | 24           | 7.5          | \n",
       "| SRR3984792-X | SRR3984792-X | 0            | 7.4          | \n",
       "| SRR3984793-X | SRR3984793-X | 0            | 7.4          | \n",
       "| SRR3984711-X | SRR3984711-X | 4            | 7.4          | \n",
       "| SRR3984712-X | SRR3984712-X | 4            | 7.4          | \n",
       "| SRR3984713-X | SRR3984713-X | 12           | 7.4          | \n",
       "| SRR3984714-X | SRR3984714-X | 12           | 7.4          | \n",
       "| SRR3984715-X | SRR3984715-X | 24           | 7.4          | \n",
       "| SRR3984716-X | SRR3984716-X | 24           | 7.4          | \n",
       "| SRR3984717-X | SRR3984717-X | 0            | 7.5          | \n",
       "| SRR3984719-X | SRR3984719-X | 4            | 7.5          | \n",
       "| SRR3984720-X | SRR3984720-X | 4            | 7.5          | \n",
       "| SRR3984722-X | SRR3984722-X | 12           | 7.5          | \n",
       "| SRR3984723-X | SRR3984723-X | 12           | 7.5          | \n",
       "| SRR3984725-X | SRR3984725-X | 24           | 7.5          | \n",
       "| SRR3984726-X | SRR3984726-X | 0            | 7.4          | \n",
       "| SRR3984727-X | SRR3984727-X | 0            | 7.4          | \n",
       "| SRR3984728-X | SRR3984728-X | 4            | 7.4          | \n",
       "| SRR3984730-X | SRR3984730-X | 12           | 7.4          | \n",
       "| SRR3984731-X | SRR3984731-X | 12           | 7.4          | \n",
       "| SRR3984733-X | SRR3984733-X | 24           | 7.4          | \n",
       "| SRR3984734-X | SRR3984734-X | 24           | 7.4          | \n",
       "| SRR3984735-X | SRR3984735-X | 0            | 7.6          | \n",
       "| SRR3984736-X | SRR3984736-X | 0            | 7.6          | \n",
       "| SRR3984737-X | SRR3984737-X | 4            | 7.6          | \n",
       "| SRR3984738-X | SRR3984738-X | 4            | 7.6          | \n",
       "| SRR3984740-X | SRR3984740-X | 12           | 7.6          | \n",
       "| SRR3984741-X | SRR3984741-X | 24           | 7.6          | \n",
       "| SRR3984742-X | SRR3984742-X | 24           | 7.6          | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "             SampleID     Incubation_Time_Weeks pH \n",
       "SRR3984709-X SRR3984709-X 0                     7.5\n",
       "SRR3984710-X SRR3984710-X 0                     7.5\n",
       "SRR3984721-X SRR3984721-X 4                     7.5\n",
       "SRR3984743-X SRR3984743-X 12                    7.5\n",
       "SRR3984755-X SRR3984755-X 12                    7.5\n",
       "SRR3984767-X SRR3984767-X 24                    7.5\n",
       "SRR3984779-X SRR3984779-X 24                    7.5\n",
       "SRR3984792-X SRR3984792-X 0                     7.4\n",
       "SRR3984793-X SRR3984793-X 0                     7.4\n",
       "SRR3984711-X SRR3984711-X 4                     7.4\n",
       "SRR3984712-X SRR3984712-X 4                     7.4\n",
       "SRR3984713-X SRR3984713-X 12                    7.4\n",
       "SRR3984714-X SRR3984714-X 12                    7.4\n",
       "SRR3984715-X SRR3984715-X 24                    7.4\n",
       "SRR3984716-X SRR3984716-X 24                    7.4\n",
       "SRR3984717-X SRR3984717-X 0                     7.5\n",
       "SRR3984719-X SRR3984719-X 4                     7.5\n",
       "SRR3984720-X SRR3984720-X 4                     7.5\n",
       "SRR3984722-X SRR3984722-X 12                    7.5\n",
       "SRR3984723-X SRR3984723-X 12                    7.5\n",
       "SRR3984725-X SRR3984725-X 24                    7.5\n",
       "SRR3984726-X SRR3984726-X 0                     7.4\n",
       "SRR3984727-X SRR3984727-X 0                     7.4\n",
       "SRR3984728-X SRR3984728-X 4                     7.4\n",
       "SRR3984730-X SRR3984730-X 12                    7.4\n",
       "SRR3984731-X SRR3984731-X 12                    7.4\n",
       "SRR3984733-X SRR3984733-X 24                    7.4\n",
       "SRR3984734-X SRR3984734-X 24                    7.4\n",
       "SRR3984735-X SRR3984735-X 0                     7.6\n",
       "SRR3984736-X SRR3984736-X 0                     7.6\n",
       "SRR3984737-X SRR3984737-X 4                     7.6\n",
       "SRR3984738-X SRR3984738-X 4                     7.6\n",
       "SRR3984740-X SRR3984740-X 12                    7.6\n",
       "SRR3984741-X SRR3984741-X 24                    7.6\n",
       "SRR3984742-X SRR3984742-X 24                    7.6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create a matrix to use to add incubation time and pH to sample IDs\n",
    "SongMatrix = c('SRR3984709-X',0, 7.5,\n",
    "'SRR3984710-X',0, 7.5,\n",
    "'SRR3984721-X',4, 7.5,\n",
    "'SRR3984743-X',12, 7.5,\n",
    "'SRR3984755-X',12, 7.5,\n",
    "'SRR3984767-X',24, 7.5,\n",
    "'SRR3984779-X',24, 7.5,\n",
    "'SRR3984792-X',0, 7.4,\n",
    "'SRR3984793-X',0, 7.4,\n",
    "'SRR3984711-X',4, 7.4,\n",
    "'SRR3984712-X',4, 7.4,\n",
    "'SRR3984713-X',12, 7.4,\n",
    "'SRR3984714-X',12, 7.4,\n",
    "'SRR3984715-X',24, 7.4,\n",
    "'SRR3984716-X',24, 7.4,\n",
    "'SRR3984717-X',0, 7.5,\n",
    "'SRR3984719-X',4, 7.5,\n",
    "'SRR3984720-X',4, 7.5,\n",
    "'SRR3984722-X',12, 7.5,\n",
    "'SRR3984723-X',12, 7.5,\n",
    "'SRR3984725-X',24, 7.5,\n",
    "'SRR3984726-X',0, 7.4, \n",
    "'SRR3984727-X',0, 7.4,\n",
    "'SRR3984728-X',4, 7.4,\n",
    "'SRR3984730-X',12, 7.4,\n",
    "'SRR3984731-X',12, 7.4,\n",
    "'SRR3984733-X',24, 7.4,\n",
    "'SRR3984734-X',24, 7.4,\n",
    "'SRR3984735-X',0, 7.6,\n",
    "'SRR3984736-X',0, 7.6,\n",
    "'SRR3984737-X',4, 7.6,\n",
    "'SRR3984738-X',4, 7.6,\n",
    "'SRR3984740-X',12, 7.6,\n",
    "'SRR3984741-X',24, 7.6,\n",
    "'SRR3984742-X',24, 7.6)\n",
    "SongMatrix = t(matrix(SongMatrix,nrow = 3))\n",
    "colnames(SongMatrix) = c(\"SampleID\",\"Incubation_Time_Weeks\", \"pH\")\n",
    "row.names(SongMatrix) = data.frame(SongMatrix)$SampleID\n",
    "SongMatrix = data.frame(SongMatrix[,])\n",
    "SongMatrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>SequencingMethod</th><th scope=col>TargetGene</th><th scope=col>Habitat</th><th scope=col>NCBISampleID</th><th scope=col>ForwardPrimerName</th><th scope=col>Sampling_Depth_cm</th><th scope=col>ForwardPrimerSequence</th><th scope=col>g.per.kg.PyOM</th><th scope=col>CollectionYear</th><th scope=col>ReversePrimerSequence</th><th scope=col></th><th scope=col>X.SOM</th><th scope=col>Region</th><th scope=col>PyOM.Temp</th><th scope=col>pH</th><th scope=col>Incubation_Time_Weeks</th><th scope=col>GeographicalLocation</th><th scope=col>Treatment</th><th scope=col>Crop</th><th scope=col>Conc_PAH_ng_per_g</th><th scope=col>Author</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>SRR3984709-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A01                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984710-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A02                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984711-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B11                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984712-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B12                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984713-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B31                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984714-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B32                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984715-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B61                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984716-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B62                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984717-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>C01                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>1%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984719-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>C11                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>1%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984720-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>C12                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>1%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984721-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A11                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984722-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>C31                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>1%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984723-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>C32                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>1%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984725-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>C62                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>1%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984726-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D01                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984727-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D02                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984728-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D11                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984730-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D31                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984731-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D32                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984733-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D61                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984734-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>D62                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>2%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984735-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E01                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984736-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E02                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984737-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E11                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984738-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E12                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984740-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E32                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984741-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E61                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984742-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>E62                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>20.4                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>600                   </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>2%BC600               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984743-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A31                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>0                     </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984755-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A32                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984767-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A61                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>4                     </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984779-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>A62                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>0                     </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>0                     </td><td>7.4                   </td><td>12                    </td><td>China                 </td><td>control               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984792-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B01                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "\t<tr><th scope=row>SRR3984793-X</th><td>Illumina MiSeq        </td><td>16S rRNA              </td><td>PAH-contaminated field</td><td>B02                   </td><td>338F                  </td><td>20                    </td><td>ACTCCTACGGGAGGCAGCA   </td><td>10.1                  </td><td>                      </td><td>GGACTACHVGGGTWTCTAAT  </td><td>                     </td><td>2                     </td><td>V4-V5                 </td><td>300                   </td><td>7.4                   </td><td>24                    </td><td>China                 </td><td>1%BC300               </td><td>                      </td><td>7581.61               </td><td>Song                  </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllll}\n",
       "  & SequencingMethod & TargetGene & Habitat & NCBISampleID & ForwardPrimerName & Sampling\\_Depth\\_cm & ForwardPrimerSequence & g.per.kg.PyOM & CollectionYear & ReversePrimerSequence &  & X.SOM & Region & PyOM.Temp & pH & Incubation\\_Time\\_Weeks & GeographicalLocation & Treatment & Crop & Conc\\_PAH\\_ng\\_per\\_g & Author\\\\\n",
       "\\hline\n",
       "\tSRR3984709-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A01                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 0                      & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984710-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A02                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 0                      & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984711-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B11                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 4                      & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984712-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B12                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 12                     & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984713-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B31                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 12                     & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984714-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B32                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 24                     & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984715-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B61                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 24                     & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984716-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B62                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 0                      & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984717-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & C01                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 0                      & China                  & 1\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984719-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & C11                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 4                      & China                  & 1\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984720-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & C12                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 4                      & China                  & 1\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984721-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A11                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 12                     & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984722-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & C31                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 12                     & China                  & 1\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984723-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & C32                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 24                     & China                  & 1\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984725-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & C62                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 24                     & China                  & 1\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984726-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D01                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 0                      & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984727-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D02                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 4                      & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984728-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D11                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 4                      & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984730-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D31                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 12                     & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984731-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D32                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 12                     & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984733-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D61                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 24                     & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984734-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & D62                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 0                      & China                  & 2\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984735-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E01                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 0                      & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984736-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E02                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 4                      & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984737-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E11                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 12                     & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984738-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E12                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 12                     & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984740-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E32                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 24                     & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984741-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E61                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 24                     & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984742-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & E62                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 20.4                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 600                    & 7.4                    & 0                      & China                  & 2\\%BC600              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984743-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A31                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 0                      & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984755-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A32                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 4                      & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984767-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A61                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 4                      & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984779-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & A62                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 0                      &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 0                      & 7.4                    & 12                     & China                  & control                &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984792-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B01                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 24                     & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\tSRR3984793-X & Illumina MiSeq         & 16S rRNA               & PAH-contaminated field & B02                    & 338F                   & 20                     & ACTCCTACGGGAGGCAGCA    & 10.1                   &                        & GGACTACHVGGGTWTCTAAT   &                       & 2                      & V4-V5                  & 300                    & 7.4                    & 24                     & China                  & 1\\%BC300              &                        & 7581.61                & Song                  \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | SequencingMethod | TargetGene | Habitat | NCBISampleID | ForwardPrimerName | Sampling_Depth_cm | ForwardPrimerSequence | g.per.kg.PyOM | CollectionYear | ReversePrimerSequence |  | X.SOM | Region | PyOM.Temp | pH | Incubation_Time_Weeks | GeographicalLocation | Treatment | Crop | Conc_PAH_ng_per_g | Author | \n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| SRR3984709-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A01                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 0                      | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984710-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A02                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 0                      | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984711-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B11                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 4                      | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984712-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B12                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 12                     | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984713-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B31                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 12                     | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984714-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B32                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 24                     | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984715-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B61                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 24                     | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984716-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B62                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 0                      | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984717-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | C01                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 0                      | China                  | 1%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984719-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | C11                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 4                      | China                  | 1%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984720-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | C12                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 4                      | China                  | 1%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984721-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A11                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 12                     | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984722-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | C31                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 12                     | China                  | 1%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984723-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | C32                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 24                     | China                  | 1%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984725-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | C62                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 24                     | China                  | 1%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984726-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D01                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 0                      | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984727-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D02                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 4                      | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984728-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D11                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 4                      | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984730-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D31                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 12                     | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984731-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D32                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 12                     | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984733-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D61                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 24                     | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984734-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | D62                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 0                      | China                  | 2%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984735-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E01                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 0                      | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984736-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E02                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 4                      | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984737-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E11                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 12                     | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984738-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E12                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 12                     | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984740-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E32                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 24                     | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984741-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E61                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 24                     | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984742-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | E62                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 20.4                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 600                    | 7.4                    | 0                      | China                  | 2%BC600                |                        | 7581.61                | Song                   | \n",
       "| SRR3984743-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A31                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 0                      | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984755-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A32                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 4                      | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984767-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A61                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 4                      | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984779-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | A62                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 0                      |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 0                      | 7.4                    | 12                     | China                  | control                |                        | 7581.61                | Song                   | \n",
       "| SRR3984792-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B01                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 24                     | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "| SRR3984793-X | Illumina MiSeq         | 16S rRNA               | PAH-contaminated field | B02                    | 338F                   | 20                     | ACTCCTACGGGAGGCAGCA    | 10.1                   |                        | GGACTACHVGGGTWTCTAAT   |                       | 2                      | V4-V5                  | 300                    | 7.4                    | 24                     | China                  | 1%BC300                |                        | 7581.61                | Song                   | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "             SequencingMethod TargetGene Habitat                NCBISampleID\n",
       "SRR3984709-X Illumina MiSeq   16S rRNA   PAH-contaminated field A01         \n",
       "SRR3984710-X Illumina MiSeq   16S rRNA   PAH-contaminated field A02         \n",
       "SRR3984711-X Illumina MiSeq   16S rRNA   PAH-contaminated field B11         \n",
       "SRR3984712-X Illumina MiSeq   16S rRNA   PAH-contaminated field B12         \n",
       "SRR3984713-X Illumina MiSeq   16S rRNA   PAH-contaminated field B31         \n",
       "SRR3984714-X Illumina MiSeq   16S rRNA   PAH-contaminated field B32         \n",
       "SRR3984715-X Illumina MiSeq   16S rRNA   PAH-contaminated field B61         \n",
       "SRR3984716-X Illumina MiSeq   16S rRNA   PAH-contaminated field B62         \n",
       "SRR3984717-X Illumina MiSeq   16S rRNA   PAH-contaminated field C01         \n",
       "SRR3984719-X Illumina MiSeq   16S rRNA   PAH-contaminated field C11         \n",
       "SRR3984720-X Illumina MiSeq   16S rRNA   PAH-contaminated field C12         \n",
       "SRR3984721-X Illumina MiSeq   16S rRNA   PAH-contaminated field A11         \n",
       "SRR3984722-X Illumina MiSeq   16S rRNA   PAH-contaminated field C31         \n",
       "SRR3984723-X Illumina MiSeq   16S rRNA   PAH-contaminated field C32         \n",
       "SRR3984725-X Illumina MiSeq   16S rRNA   PAH-contaminated field C62         \n",
       "SRR3984726-X Illumina MiSeq   16S rRNA   PAH-contaminated field D01         \n",
       "SRR3984727-X Illumina MiSeq   16S rRNA   PAH-contaminated field D02         \n",
       "SRR3984728-X Illumina MiSeq   16S rRNA   PAH-contaminated field D11         \n",
       "SRR3984730-X Illumina MiSeq   16S rRNA   PAH-contaminated field D31         \n",
       "SRR3984731-X Illumina MiSeq   16S rRNA   PAH-contaminated field D32         \n",
       "SRR3984733-X Illumina MiSeq   16S rRNA   PAH-contaminated field D61         \n",
       "SRR3984734-X Illumina MiSeq   16S rRNA   PAH-contaminated field D62         \n",
       "SRR3984735-X Illumina MiSeq   16S rRNA   PAH-contaminated field E01         \n",
       "SRR3984736-X Illumina MiSeq   16S rRNA   PAH-contaminated field E02         \n",
       "SRR3984737-X Illumina MiSeq   16S rRNA   PAH-contaminated field E11         \n",
       "SRR3984738-X Illumina MiSeq   16S rRNA   PAH-contaminated field E12         \n",
       "SRR3984740-X Illumina MiSeq   16S rRNA   PAH-contaminated field E32         \n",
       "SRR3984741-X Illumina MiSeq   16S rRNA   PAH-contaminated field E61         \n",
       "SRR3984742-X Illumina MiSeq   16S rRNA   PAH-contaminated field E62         \n",
       "SRR3984743-X Illumina MiSeq   16S rRNA   PAH-contaminated field A31         \n",
       "SRR3984755-X Illumina MiSeq   16S rRNA   PAH-contaminated field A32         \n",
       "SRR3984767-X Illumina MiSeq   16S rRNA   PAH-contaminated field A61         \n",
       "SRR3984779-X Illumina MiSeq   16S rRNA   PAH-contaminated field A62         \n",
       "SRR3984792-X Illumina MiSeq   16S rRNA   PAH-contaminated field B01         \n",
       "SRR3984793-X Illumina MiSeq   16S rRNA   PAH-contaminated field B02         \n",
       "             ForwardPrimerName Sampling_Depth_cm ForwardPrimerSequence\n",
       "SRR3984709-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984710-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984711-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984712-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984713-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984714-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984715-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984716-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984717-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984719-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984720-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984721-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984722-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984723-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984725-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984726-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984727-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984728-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984730-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984731-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984733-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984734-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984735-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984736-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984737-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984738-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984740-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984741-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984742-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984743-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984755-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984767-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984779-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984792-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "SRR3984793-X 338F              20                ACTCCTACGGGAGGCAGCA  \n",
       "             g.per.kg.PyOM CollectionYear ReversePrimerSequence  X.SOM Region\n",
       "SRR3984709-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984710-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984711-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984712-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984713-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984714-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984715-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984716-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984717-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984719-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984720-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984721-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984722-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984723-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984725-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984726-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984727-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984728-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984730-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984731-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984733-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984734-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984735-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984736-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984737-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984738-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984740-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984741-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984742-X 20.4                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984743-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984755-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984767-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984779-X 0                            GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984792-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "SRR3984793-X 10.1                         GGACTACHVGGGTWTCTAAT   2     V4-V5 \n",
       "             PyOM.Temp pH  Incubation_Time_Weeks GeographicalLocation Treatment\n",
       "SRR3984709-X 0         7.4 0                     China                control  \n",
       "SRR3984710-X 0         7.4 0                     China                control  \n",
       "SRR3984711-X 300       7.4 4                     China                1%BC300  \n",
       "SRR3984712-X 300       7.4 12                    China                1%BC300  \n",
       "SRR3984713-X 300       7.4 12                    China                1%BC300  \n",
       "SRR3984714-X 300       7.4 24                    China                1%BC300  \n",
       "SRR3984715-X 300       7.4 24                    China                1%BC300  \n",
       "SRR3984716-X 300       7.4 0                     China                1%BC300  \n",
       "SRR3984717-X 600       7.4 0                     China                1%BC600  \n",
       "SRR3984719-X 600       7.4 4                     China                1%BC600  \n",
       "SRR3984720-X 600       7.4 4                     China                1%BC600  \n",
       "SRR3984721-X 0         7.4 12                    China                control  \n",
       "SRR3984722-X 600       7.4 12                    China                1%BC600  \n",
       "SRR3984723-X 600       7.4 24                    China                1%BC600  \n",
       "SRR3984725-X 600       7.4 24                    China                1%BC600  \n",
       "SRR3984726-X 300       7.4 0                     China                2%BC300  \n",
       "SRR3984727-X 300       7.4 4                     China                2%BC300  \n",
       "SRR3984728-X 300       7.4 4                     China                2%BC300  \n",
       "SRR3984730-X 300       7.4 12                    China                2%BC300  \n",
       "SRR3984731-X 300       7.4 12                    China                2%BC300  \n",
       "SRR3984733-X 300       7.4 24                    China                2%BC300  \n",
       "SRR3984734-X 300       7.4 0                     China                2%BC300  \n",
       "SRR3984735-X 600       7.4 0                     China                2%BC600  \n",
       "SRR3984736-X 600       7.4 4                     China                2%BC600  \n",
       "SRR3984737-X 600       7.4 12                    China                2%BC600  \n",
       "SRR3984738-X 600       7.4 12                    China                2%BC600  \n",
       "SRR3984740-X 600       7.4 24                    China                2%BC600  \n",
       "SRR3984741-X 600       7.4 24                    China                2%BC600  \n",
       "SRR3984742-X 600       7.4 0                     China                2%BC600  \n",
       "SRR3984743-X 0         7.4 0                     China                control  \n",
       "SRR3984755-X 0         7.4 4                     China                control  \n",
       "SRR3984767-X 0         7.4 4                     China                control  \n",
       "SRR3984779-X 0         7.4 12                    China                control  \n",
       "SRR3984792-X 300       7.4 24                    China                1%BC300  \n",
       "SRR3984793-X 300       7.4 24                    China                1%BC300  \n",
       "             Crop Conc_PAH_ng_per_g Author\n",
       "SRR3984709-X      7581.61           Song  \n",
       "SRR3984710-X      7581.61           Song  \n",
       "SRR3984711-X      7581.61           Song  \n",
       "SRR3984712-X      7581.61           Song  \n",
       "SRR3984713-X      7581.61           Song  \n",
       "SRR3984714-X      7581.61           Song  \n",
       "SRR3984715-X      7581.61           Song  \n",
       "SRR3984716-X      7581.61           Song  \n",
       "SRR3984717-X      7581.61           Song  \n",
       "SRR3984719-X      7581.61           Song  \n",
       "SRR3984720-X      7581.61           Song  \n",
       "SRR3984721-X      7581.61           Song  \n",
       "SRR3984722-X      7581.61           Song  \n",
       "SRR3984723-X      7581.61           Song  \n",
       "SRR3984725-X      7581.61           Song  \n",
       "SRR3984726-X      7581.61           Song  \n",
       "SRR3984727-X      7581.61           Song  \n",
       "SRR3984728-X      7581.61           Song  \n",
       "SRR3984730-X      7581.61           Song  \n",
       "SRR3984731-X      7581.61           Song  \n",
       "SRR3984733-X      7581.61           Song  \n",
       "SRR3984734-X      7581.61           Song  \n",
       "SRR3984735-X      7581.61           Song  \n",
       "SRR3984736-X      7581.61           Song  \n",
       "SRR3984737-X      7581.61           Song  \n",
       "SRR3984738-X      7581.61           Song  \n",
       "SRR3984740-X      7581.61           Song  \n",
       "SRR3984741-X      7581.61           Song  \n",
       "SRR3984742-X      7581.61           Song  \n",
       "SRR3984743-X      7581.61           Song  \n",
       "SRR3984755-X      7581.61           Song  \n",
       "SRR3984767-X      7581.61           Song  \n",
       "SRR3984779-X      7581.61           Song  \n",
       "SRR3984792-X      7581.61           Song  \n",
       "SRR3984793-X      7581.61           Song  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Paste incubation time from matrix created above into the ps\n",
    "sample_data(ps_song)[row.names(sample_data(ps_song)) %in% row.names(SongMatrix),]$Incubation_Time_Weeks = paste(SongMatrix$Incubation_Time_Weeks)\n",
    "sample_data(ps_song)$pH = sample_data(ps_song)[row.names(sample_data(ps_song)) %in% row.names(SongMatrix),]$pH = paste(SongMatrix$pH)\n",
    "\n",
    "sample_data(ps_song)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in `[<-.data.frame`(x3, i, , ..., value = value):\n",
      "provided 34 variables to replace 33 variables"
     ]
    }
   ],
   "source": [
    "#no soil type was provided in the publication; we looked at the paper to obtain a rough estimate on where the soil was \n",
    "#collected- the soil type was sourced from the FAO/UNESCO Soil Map of the World\n",
    "sample_data(ps_song)$SoilType = sample_data(ps_song)[sample_data(ps_song)$Author==\"Song\",]$SoilType = \"Gleysol\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: unammended, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field\n",
      "The following `from` values were not present in `x`: N/A\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_song)$Treatment = revalue(sample_data(ps_song)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_song)$SoilType = revalue(sample_data(ps_song)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_song)$Feedstock = revalue(sample_data(ps_song)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_song)$Habitat = revalue(sample_data(ps_song)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_song)$CurrentLandUse = revalue(sample_data(ps_song)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_song)$PyOM.Temp = revalue(sample_data(ps_song)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_song)$Depth = revalue(sample_data(ps_song)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'NCBISampleID'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'g.per.kg.PyOM'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'Conc_PAH_ng_per_g'</li>\n",
       "\t<li>'Author'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'TargetGene'\n",
       "\\item 'Habitat'\n",
       "\\item 'NCBISampleID'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'g.per.kg.PyOM'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Silt'\n",
       "\\item 'BioSample'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Feedstock'\n",
       "\\item 'Sand'\n",
       "\\item 'Organism'\n",
       "\\item 'BioProject'\n",
       "\\item 'Clay'\n",
       "\\item 'SRA'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item '\\%SOM'\n",
       "\\item 'Region'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'pH'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'Treatment'\n",
       "\\item 'Crop'\n",
       "\\item 'Conc\\_PAH\\_ng\\_per\\_g'\n",
       "\\item 'Author'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SequencingMethod'\n",
       "2. 'TargetGene'\n",
       "3. 'Habitat'\n",
       "4. 'NCBISampleID'\n",
       "5. 'ForwardPrimerName'\n",
       "6. 'Sampling_Depth_cm'\n",
       "7. 'ForwardPrimerSequence'\n",
       "8. 'g.per.kg.PyOM'\n",
       "9. 'CollectionYear'\n",
       "10. 'ReversePrimerSequence'\n",
       "11. 'Silt'\n",
       "12. 'BioSample'\n",
       "13. 'RevBarcode'\n",
       "14. 'RevPrimerName'\n",
       "15. 'Feedstock'\n",
       "16. 'Sand'\n",
       "17. 'Organism'\n",
       "18. 'BioProject'\n",
       "19. 'Clay'\n",
       "20. 'SRA'\n",
       "21. 'FwdBarcode'\n",
       "22. 'NucleicAcidExtraction'\n",
       "23. 'CurrentLandUse'\n",
       "24. '%SOM'\n",
       "25. 'Region'\n",
       "26. 'PyOM.Temp'\n",
       "27. 'pH'\n",
       "28. 'Incubation_Time_Weeks'\n",
       "29. 'GeographicalLocation'\n",
       "30. 'Treatment'\n",
       "31. 'Crop'\n",
       "32. 'Conc_PAH_ng_per_g'\n",
       "33. 'Author'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SequencingMethod\"      \"TargetGene\"            \"Habitat\"              \n",
       " [4] \"NCBISampleID\"          \"ForwardPrimerName\"     \"Sampling_Depth_cm\"    \n",
       " [7] \"ForwardPrimerSequence\" \"g.per.kg.PyOM\"         \"CollectionYear\"       \n",
       "[10] \"ReversePrimerSequence\" \"Silt\"                  \"BioSample\"            \n",
       "[13] \"RevBarcode\"            \"RevPrimerName\"         \"Feedstock\"            \n",
       "[16] \"Sand\"                  \"Organism\"              \"BioProject\"           \n",
       "[19] \"Clay\"                  \"SRA\"                   \"FwdBarcode\"           \n",
       "[22] \"NucleicAcidExtraction\" \"CurrentLandUse\"        \"%SOM\"                 \n",
       "[25] \"Region\"                \"PyOM.Temp\"             \"pH\"                   \n",
       "[28] \"Incubation_Time_Weeks\" \"GeographicalLocation\"  \"Treatment\"            \n",
       "[31] \"Crop\"                  \"Conc_PAH_ng_per_g\"     \"Author\"               "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "\t<li>'24'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\item '24'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '24'\n",
       "2. '24'\n",
       "3. '24'\n",
       "4. '24'\n",
       "5. '24'\n",
       "6. '24'\n",
       "7. '24'\n",
       "8. '24'\n",
       "9. '24'\n",
       "10. '24'\n",
       "11. '24'\n",
       "12. '24'\n",
       "13. '24'\n",
       "14. '24'\n",
       "15. '24'\n",
       "16. '24'\n",
       "17. '24'\n",
       "18. '24'\n",
       "19. '24'\n",
       "20. '24'\n",
       "21. '24'\n",
       "22. '24'\n",
       "23. '24'\n",
       "24. '24'\n",
       "25. '24'\n",
       "26. '24'\n",
       "27. '24'\n",
       "28. '24'\n",
       "29. '24'\n",
       "30. '24'\n",
       "31. '24'\n",
       "32. '24'\n",
       "33. '24'\n",
       "34. '24'\n",
       "35. '24'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\"\n",
       "[16] \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\" \"24\"\n",
       "[31] \"24\" \"24\" \"24\" \"24\" \"24\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and t.per.ha.PyOM are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_song))[names(sample_data(ps_song)) == \"PyOM-Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_song))[names(sample_data(ps_song)) == \"g-per-kg-PyOM\"] <- \"g.per.kg.PyOM\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\")\n",
    "OMAdded = c('Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 139,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_song)$PyOM = ifelse(sample_data(ps_song)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_song)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_song)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>10.8</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>7.2</li>\n",
       "\t<li>7.2</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item <NA>\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item 10.8\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 7.2\n",
       "\\item 7.2\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. &lt;NA&gt;\n",
       "3. 7.2\n",
       "4. 7.2\n",
       "5. 7.2\n",
       "6. 7.2\n",
       "7. 7.2\n",
       "8. 7.2\n",
       "9. 10.8\n",
       "10. 10.8\n",
       "11. 10.8\n",
       "12. &lt;NA&gt;\n",
       "13. 10.8\n",
       "14. 10.8\n",
       "15. 10.8\n",
       "16. 7.2\n",
       "17. 7.2\n",
       "18. 7.2\n",
       "19. 7.2\n",
       "20. 7.2\n",
       "21. 7.2\n",
       "22. 7.2\n",
       "23. 10.8\n",
       "24. 10.8\n",
       "25. 10.8\n",
       "26. 10.8\n",
       "27. 10.8\n",
       "28. 10.8\n",
       "29. 10.8\n",
       "30. &lt;NA&gt;\n",
       "31. &lt;NA&gt;\n",
       "32. &lt;NA&gt;\n",
       "33. &lt;NA&gt;\n",
       "34. 7.2\n",
       "35. 7.2\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1]   NA   NA  7.2  7.2  7.2  7.2  7.2  7.2 10.8 10.8 10.8   NA 10.8 10.8 10.8\n",
       "[16]  7.2  7.2  7.2  7.2  7.2  7.2  7.2 10.8 10.8 10.8 10.8 10.8 10.8 10.8   NA\n",
       "[31]   NA   NA   NA  7.2  7.2"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_song)$PyOM.pH = ifelse(sample_data(ps_song)$PyOM.Temp %in% \"300\", 7.2,\n",
    "                                        ifelse(sample_data(ps_song)$PyOM.Temp %in% \"600\", 10.8 , NA))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_song, \"song.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Whitman"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ps_whitman = import_biom(\"Whitman_OTU_table/feature-table-metaD-tax_json-ncbiID.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Prune out sample that didn't have any reads\n",
    "ps_whitman= prune_samples(sample_names(ps_whitman)!= \"SRR3305959-X\", ps_whitman)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercionWarning message in asMethod(object):\n",
      "NAs introduced by coercion"
     ]
    }
   ],
   "source": [
    "#some samples had duplicate sequencing files - the next few lines will be merging the duplicated based on sample name\n",
    "ps.merged = merge_samples(ps_whitman,\"NCBISampleID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'X.N'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'NCBISampleID'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'X.C'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'SoilType'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'Feedstock'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'pH'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'TargetGene'\n",
       "\\item 'X.N'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'Crop'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'Author'\n",
       "\\item 'Region'\n",
       "\\item 'Organism'\n",
       "\\item 'Habitat'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'Silt'\n",
       "\\item 'Sand'\n",
       "\\item 'NCBISampleID'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'SRA'\n",
       "\\item 'Clay'\n",
       "\\item 'Treatment'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'BioProject'\n",
       "\\item 'X.C'\n",
       "\\item 'BioSample'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'ForwardPrimerName'\n",
       "2. 'SoilType'\n",
       "3. 'RevBarcode'\n",
       "4. 'NucleicAcidExtraction'\n",
       "5. 'Feedstock'\n",
       "6. 'GeographicalLocation'\n",
       "7. 'ReversePrimerSequence'\n",
       "8. 'pH'\n",
       "9. 'RevPrimerName'\n",
       "10. 'TargetGene'\n",
       "11. 'X.N'\n",
       "12. 'ForwardPrimerSequence'\n",
       "13. 'Crop'\n",
       "14. 'SequencingMethod'\n",
       "15. 't.per.ha.PyOM'\n",
       "16. 'Author'\n",
       "17. 'Region'\n",
       "18. 'Organism'\n",
       "19. 'Habitat'\n",
       "20. 'CurrentLandUse'\n",
       "21. 'Silt'\n",
       "22. 'Sand'\n",
       "23. 'NCBISampleID'\n",
       "24. 'FwdBarcode'\n",
       "25. 'PyOM.Temp'\n",
       "26. 'Incubation_Time_Weeks'\n",
       "27. 'Sampling_Depth_cm'\n",
       "28. 'SRA'\n",
       "29. 'Clay'\n",
       "30. 'Treatment'\n",
       "31. 'CollectionYear'\n",
       "32. 'BioProject'\n",
       "33. 'X.C'\n",
       "34. 'BioSample'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"ForwardPrimerName\"     \"SoilType\"              \"RevBarcode\"           \n",
       " [4] \"NucleicAcidExtraction\" \"Feedstock\"             \"GeographicalLocation\" \n",
       " [7] \"ReversePrimerSequence\" \"pH\"                    \"RevPrimerName\"        \n",
       "[10] \"TargetGene\"            \"X.N\"                   \"ForwardPrimerSequence\"\n",
       "[13] \"Crop\"                  \"SequencingMethod\"      \"t.per.ha.PyOM\"        \n",
       "[16] \"Author\"                \"Region\"                \"Organism\"             \n",
       "[19] \"Habitat\"               \"CurrentLandUse\"        \"Silt\"                 \n",
       "[22] \"Sand\"                  \"NCBISampleID\"          \"FwdBarcode\"           \n",
       "[25] \"PyOM.Temp\"             \"Incubation_Time_Weeks\" \"Sampling_Depth_cm\"    \n",
       "[28] \"SRA\"                   \"Clay\"                  \"Treatment\"            \n",
       "[31] \"CollectionYear\"        \"BioProject\"            \"X.C\"                  \n",
       "[34] \"BioSample\"            "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#create variable with all the column names\n",
    "names=colnames(sample_data(ps_whitman))\n",
    "names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "notcommonnames = c('RevBarcode','FwdBarcode','SRA')\n",
    "# Create list of sample data items that are not necessarily the same within the merged pairs\n",
    "commonnames = names[(names %in% notcommonnames)]\n",
    "# Pull out only the sample data categories that are the same\n",
    "common_sample_data = sample_data(ps_whitman)[ , -which(names(sample_data(ps_whitman)) %in% commonnames)]\n",
    "# Get the sample data for those common categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "sample_data(ps_whitman)$pH = as.numeric(sample_data(ps_whitman)$pH)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "phyloseq-class experiment-level object\n",
       "otu_table()   OTU Table:         [ 8628 taxa and 119 samples ]\n",
       "sample_data() Sample Data:       [ 119 samples by 31 sample variables ]\n",
       "tax_table()   Taxonomy Table:    [ 8628 taxa by 7 taxonomic ranks ]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "commonrownames=row.names(sample_data(ps.merged))\n",
    "# Grab the rownames of the merged sample data - i.e., the new sample IDs\n",
    "common_sample_data2 = common_sample_data[which(common_sample_data$NCBISampleID %in% commonrownames),]\n",
    "# Pull out only one row for each sample ID\n",
    "common_sample_data2 = common_sample_data2[!duplicated(common_sample_data2$NCBISampleID), ]\n",
    "# Cut out all duplicates (from merging)\n",
    "row.names(common_sample_data2) = common_sample_data2$NCBISampleID\n",
    "# Rename rows with the new NCBI common sample ID\n",
    "sample_data(ps.merged)=common_sample_data2\n",
    "# Reassign sample data to the ps.merged object\n",
    "\n",
    "# Renaming it all\n",
    "ps_whitman=ps.merged\n",
    "ps_whitman\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>224c444fe97e6aa6bc29426d54fe494b</th><td>Bacteria                              </td><td>Proteobacteria                        </td><td>Alphaproteobacteria                   </td><td>Rhizobiales                           </td><td>Xanthobacteraceae                     </td><td>Bradyrhizobium                        </td><td>NA                                    </td></tr>\n",
       "\t<tr><th scope=row>3d171376b6712ac4e9f7639c4bd5324f</th><td>Bacteria                              </td><td>Proteobacteria                        </td><td>Alphaproteobacteria                   </td><td>Sphingomonadales                      </td><td>Sphingomonadaceae                     </td><td>Sphingomonas                          </td><td>uncultured Sphingomonadaceae bacterium</td></tr>\n",
       "\t<tr><th scope=row>532bc584b044143342bf078fd5bfd49b</th><td>Bacteria                              </td><td>Proteobacteria                        </td><td>Alphaproteobacteria                   </td><td>Sphingomonadales                      </td><td>Sphingomonadaceae                     </td><td>Sphingomonas                          </td><td>uncultured Sphingomonadaceae bacterium</td></tr>\n",
       "\t<tr><th scope=row>d0bfc877c648962e3b7c31828dab4797</th><td>Bacteria                              </td><td>Actinobacteria                        </td><td>Actinobacteria                        </td><td>Micrococcales                         </td><td>Micrococcaceae                        </td><td>NA                                    </td><td>NA                                    </td></tr>\n",
       "\t<tr><th scope=row>a8e90563afa8945d6bc7e9f008106ee3</th><td>Bacteria                              </td><td>Proteobacteria                        </td><td>Gammaproteobacteria                   </td><td>Betaproteobacteriales                 </td><td>Burkholderiaceae                      </td><td>Massilia                              </td><td>Ambiguous_taxa                        </td></tr>\n",
       "\t<tr><th scope=row>be23fcd748d2621432e14c0487c93e05</th><td>Bacteria                              </td><td>Proteobacteria                        </td><td>Alphaproteobacteria                   </td><td>Rhizobiales                           </td><td>Xanthobacteraceae                     </td><td>Bradyrhizobium                        </td><td>NA                                    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllll}\n",
       "  & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       "\\hline\n",
       "\t224c444fe97e6aa6bc29426d54fe494b & Bacteria                               & Proteobacteria                         & Alphaproteobacteria                    & Rhizobiales                            & Xanthobacteraceae                      & Bradyrhizobium                         & NA                                    \\\\\n",
       "\t3d171376b6712ac4e9f7639c4bd5324f & Bacteria                               & Proteobacteria                         & Alphaproteobacteria                    & Sphingomonadales                       & Sphingomonadaceae                      & Sphingomonas                           & uncultured Sphingomonadaceae bacterium\\\\\n",
       "\t532bc584b044143342bf078fd5bfd49b & Bacteria                               & Proteobacteria                         & Alphaproteobacteria                    & Sphingomonadales                       & Sphingomonadaceae                      & Sphingomonas                           & uncultured Sphingomonadaceae bacterium\\\\\n",
       "\td0bfc877c648962e3b7c31828dab4797 & Bacteria                               & Actinobacteria                         & Actinobacteria                         & Micrococcales                          & Micrococcaceae                         & NA                                     & NA                                    \\\\\n",
       "\ta8e90563afa8945d6bc7e9f008106ee3 & Bacteria                                 & Proteobacteria                           & Gammaproteobacteria                      & Betaproteobacteriales                    & Burkholderiaceae                         & Massilia                                 & Ambiguous\\_taxa                        \\\\\n",
       "\tbe23fcd748d2621432e14c0487c93e05 & Bacteria                               & Proteobacteria                         & Alphaproteobacteria                    & Rhizobiales                            & Xanthobacteraceae                      & Bradyrhizobium                         & NA                                    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Domain | Phylum | Class | Order | Family | Genus | Species | \n",
       "|---|---|---|---|---|---|\n",
       "| 224c444fe97e6aa6bc29426d54fe494b | Bacteria                               | Proteobacteria                         | Alphaproteobacteria                    | Rhizobiales                            | Xanthobacteraceae                      | Bradyrhizobium                         | NA                                     | \n",
       "| 3d171376b6712ac4e9f7639c4bd5324f | Bacteria                               | Proteobacteria                         | Alphaproteobacteria                    | Sphingomonadales                       | Sphingomonadaceae                      | Sphingomonas                           | uncultured Sphingomonadaceae bacterium | \n",
       "| 532bc584b044143342bf078fd5bfd49b | Bacteria                               | Proteobacteria                         | Alphaproteobacteria                    | Sphingomonadales                       | Sphingomonadaceae                      | Sphingomonas                           | uncultured Sphingomonadaceae bacterium | \n",
       "| d0bfc877c648962e3b7c31828dab4797 | Bacteria                               | Actinobacteria                         | Actinobacteria                         | Micrococcales                          | Micrococcaceae                         | NA                                     | NA                                     | \n",
       "| a8e90563afa8945d6bc7e9f008106ee3 | Bacteria                               | Proteobacteria                         | Gammaproteobacteria                    | Betaproteobacteriales                  | Burkholderiaceae                       | Massilia                               | Ambiguous_taxa                         | \n",
       "| be23fcd748d2621432e14c0487c93e05 | Bacteria                               | Proteobacteria                         | Alphaproteobacteria                    | Rhizobiales                            | Xanthobacteraceae                      | Bradyrhizobium                         | NA                                     | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "                                 Domain   Phylum         Class              \n",
       "224c444fe97e6aa6bc29426d54fe494b Bacteria Proteobacteria Alphaproteobacteria\n",
       "3d171376b6712ac4e9f7639c4bd5324f Bacteria Proteobacteria Alphaproteobacteria\n",
       "532bc584b044143342bf078fd5bfd49b Bacteria Proteobacteria Alphaproteobacteria\n",
       "d0bfc877c648962e3b7c31828dab4797 Bacteria Actinobacteria Actinobacteria     \n",
       "a8e90563afa8945d6bc7e9f008106ee3 Bacteria Proteobacteria Gammaproteobacteria\n",
       "be23fcd748d2621432e14c0487c93e05 Bacteria Proteobacteria Alphaproteobacteria\n",
       "                                 Order                 Family           \n",
       "224c444fe97e6aa6bc29426d54fe494b Rhizobiales           Xanthobacteraceae\n",
       "3d171376b6712ac4e9f7639c4bd5324f Sphingomonadales      Sphingomonadaceae\n",
       "532bc584b044143342bf078fd5bfd49b Sphingomonadales      Sphingomonadaceae\n",
       "d0bfc877c648962e3b7c31828dab4797 Micrococcales         Micrococcaceae   \n",
       "a8e90563afa8945d6bc7e9f008106ee3 Betaproteobacteriales Burkholderiaceae \n",
       "be23fcd748d2621432e14c0487c93e05 Rhizobiales           Xanthobacteraceae\n",
       "                                 Genus         \n",
       "224c444fe97e6aa6bc29426d54fe494b Bradyrhizobium\n",
       "3d171376b6712ac4e9f7639c4bd5324f Sphingomonas  \n",
       "532bc584b044143342bf078fd5bfd49b Sphingomonas  \n",
       "d0bfc877c648962e3b7c31828dab4797 NA            \n",
       "a8e90563afa8945d6bc7e9f008106ee3 Massilia      \n",
       "be23fcd748d2621432e14c0487c93e05 Bradyrhizobium\n",
       "                                 Species                               \n",
       "224c444fe97e6aa6bc29426d54fe494b NA                                    \n",
       "3d171376b6712ac4e9f7639c4bd5324f uncultured Sphingomonadaceae bacterium\n",
       "532bc584b044143342bf078fd5bfd49b uncultured Sphingomonadaceae bacterium\n",
       "d0bfc877c648962e3b7c31828dab4797 NA                                    \n",
       "a8e90563afa8945d6bc7e9f008106ee3 Ambiguous_taxa                        \n",
       "be23fcd748d2621432e14c0487c93e05 NA                                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x.whitman = data.frame(tax_table(ps_whitman))\n",
    "\n",
    "colnames(x.whitman) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "\n",
    "x.whitman$Domain = gsub(\"D_0__\", \"\", as.character(x.whitman$Domain))\n",
    "x.whitman$Phylum = gsub(\"D_1__\", \"\", as.character(x.whitman$Phylum))\n",
    "x.whitman$Class = gsub(\"D_2__\", \"\", as.character(x.whitman$Class))\n",
    "x.whitman$Order = gsub(\"D_3__\", \"\", as.character(x.whitman$Order))\n",
    "x.whitman$Family = gsub(\"D_4__\", \"\", as.character(x.whitman$Family))\n",
    "x.whitman$Genus = gsub(\"D_5__\", \"\", as.character(x.whitman$Genus))\n",
    "x.whitman$Species = gsub(\"D_6__\", \"\", as.character(x.whitman$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x.whitman=tax_table(as.matrix(x.whitman,dimnames=list(row.names(x.whitman),colnames(x.whitman))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_whitman)=x.whitman\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_whitman))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "#Note: in SoilType we changed \"Mardin\" to \"Fragiudept\" \n",
    "\n",
    "sample_data(ps_whitman)$Treatment = revalue(sample_data(ps_whitman)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_whitman)$SoilType = revalue(sample_data(ps_whitman)$SoilType, c(\"mollisol\"=\"Mollisol\" ,\"Mardin\" = \"Fragiudept\" ))\n",
    "sample_data(ps_whitman)$Feedstock = revalue(sample_data(ps_whitman)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_whitman)$Habitat = revalue(sample_data(ps_whitman)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_whitman)$CurrentLandUse = revalue(sample_data(ps_whitman)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_whitman)$PyOM.Temp = revalue(sample_data(ps_whitman)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_whitman)$Depth = revalue(sample_data(ps_whitman)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\")\n",
    "OMAdded = c('Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_whitman)$PyOM = ifelse(sample_data(ps_whitman)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_whitman)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_whitman)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>10</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item 10\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. 10\n",
       "3. &lt;NA&gt;\n",
       "4. 10\n",
       "5. &lt;NA&gt;\n",
       "6. 10\n",
       "7. &lt;NA&gt;\n",
       "8. 10\n",
       "9. &lt;NA&gt;\n",
       "10. 10\n",
       "11. &lt;NA&gt;\n",
       "12. 10\n",
       "13. &lt;NA&gt;\n",
       "14. 10\n",
       "15. &lt;NA&gt;\n",
       "16. 10\n",
       "17. &lt;NA&gt;\n",
       "18. 10\n",
       "19. &lt;NA&gt;\n",
       "20. 10\n",
       "21. &lt;NA&gt;\n",
       "22. 10\n",
       "23. &lt;NA&gt;\n",
       "24. 10\n",
       "25. &lt;NA&gt;\n",
       "26. 10\n",
       "27. &lt;NA&gt;\n",
       "28. 10\n",
       "29. &lt;NA&gt;\n",
       "30. 10\n",
       "31. &lt;NA&gt;\n",
       "32. 10\n",
       "33. &lt;NA&gt;\n",
       "34. &lt;NA&gt;\n",
       "35. &lt;NA&gt;\n",
       "36. &lt;NA&gt;\n",
       "37. &lt;NA&gt;\n",
       "38. &lt;NA&gt;\n",
       "39. &lt;NA&gt;\n",
       "40. &lt;NA&gt;\n",
       "41. &lt;NA&gt;\n",
       "42. 10\n",
       "43. &lt;NA&gt;\n",
       "44. 10\n",
       "45. &lt;NA&gt;\n",
       "46. 10\n",
       "47. &lt;NA&gt;\n",
       "48. 10\n",
       "49. &lt;NA&gt;\n",
       "50. 10\n",
       "51. &lt;NA&gt;\n",
       "52. 10\n",
       "53. &lt;NA&gt;\n",
       "54. 10\n",
       "55. &lt;NA&gt;\n",
       "56. 10\n",
       "57. &lt;NA&gt;\n",
       "58. 10\n",
       "59. &lt;NA&gt;\n",
       "60. 10\n",
       "61. &lt;NA&gt;\n",
       "62. 10\n",
       "63. &lt;NA&gt;\n",
       "64. 10\n",
       "65. &lt;NA&gt;\n",
       "66. 10\n",
       "67. &lt;NA&gt;\n",
       "68. 10\n",
       "69. &lt;NA&gt;\n",
       "70. 10\n",
       "71. &lt;NA&gt;\n",
       "72. 10\n",
       "73. &lt;NA&gt;\n",
       "74. &lt;NA&gt;\n",
       "75. &lt;NA&gt;\n",
       "76. &lt;NA&gt;\n",
       "77. &lt;NA&gt;\n",
       "78. &lt;NA&gt;\n",
       "79. &lt;NA&gt;\n",
       "80. &lt;NA&gt;\n",
       "81. &lt;NA&gt;\n",
       "82. 10\n",
       "83. &lt;NA&gt;\n",
       "84. 10\n",
       "85. &lt;NA&gt;\n",
       "86. 10\n",
       "87. &lt;NA&gt;\n",
       "88. 10\n",
       "89. 10\n",
       "90. &lt;NA&gt;\n",
       "91. 10\n",
       "92. &lt;NA&gt;\n",
       "93. 10\n",
       "94. &lt;NA&gt;\n",
       "95. 10\n",
       "96. &lt;NA&gt;\n",
       "97. 10\n",
       "98. &lt;NA&gt;\n",
       "99. 10\n",
       "100. &lt;NA&gt;\n",
       "101. 10\n",
       "102. &lt;NA&gt;\n",
       "103. 10\n",
       "104. &lt;NA&gt;\n",
       "105. 10\n",
       "106. &lt;NA&gt;\n",
       "107. 10\n",
       "108. &lt;NA&gt;\n",
       "109. 10\n",
       "110. &lt;NA&gt;\n",
       "111. 10\n",
       "112. &lt;NA&gt;\n",
       "113. &lt;NA&gt;\n",
       "114. &lt;NA&gt;\n",
       "115. &lt;NA&gt;\n",
       "116. &lt;NA&gt;\n",
       "117. &lt;NA&gt;\n",
       "118. &lt;NA&gt;\n",
       "119. &lt;NA&gt;\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA\n",
       " [26] 10 NA 10 NA 10 NA 10 NA NA NA NA NA NA NA NA NA 10 NA 10 NA 10 NA 10 NA 10\n",
       " [51] NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA NA NA\n",
       " [76] NA NA NA NA NA NA 10 NA 10 NA 10 NA 10 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA\n",
       "[101] 10 NA 10 NA 10 NA 10 NA 10 NA 10 NA NA NA NA NA NA NA NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6.75</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "\t<li>6</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6.75\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\item 6\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6\n",
       "2. 6.75\n",
       "3. 6\n",
       "4. 6.75\n",
       "5. 6\n",
       "6. 6.75\n",
       "7. 6\n",
       "8. 6.75\n",
       "9. 6\n",
       "10. 6.75\n",
       "11. 6\n",
       "12. 6.75\n",
       "13. 6\n",
       "14. 6.75\n",
       "15. 6\n",
       "16. 6.75\n",
       "17. 6\n",
       "18. 6.75\n",
       "19. 6\n",
       "20. 6.75\n",
       "21. 6\n",
       "22. 6.75\n",
       "23. 6\n",
       "24. 6.75\n",
       "25. 6\n",
       "26. 6.75\n",
       "27. 6\n",
       "28. 6.75\n",
       "29. 6\n",
       "30. 6.75\n",
       "31. 6\n",
       "32. 6.75\n",
       "33. 6\n",
       "34. 6\n",
       "35. 6\n",
       "36. 6\n",
       "37. 6\n",
       "38. 6\n",
       "39. 6\n",
       "40. 6\n",
       "41. 6\n",
       "42. 6.75\n",
       "43. 6\n",
       "44. 6.75\n",
       "45. 6\n",
       "46. 6.75\n",
       "47. 6\n",
       "48. 6.75\n",
       "49. 6\n",
       "50. 6.75\n",
       "51. 6\n",
       "52. 6.75\n",
       "53. 6\n",
       "54. 6.75\n",
       "55. 6\n",
       "56. 6.75\n",
       "57. 6\n",
       "58. 6.75\n",
       "59. 6\n",
       "60. 6.75\n",
       "61. 6\n",
       "62. 6.75\n",
       "63. 6\n",
       "64. 6.75\n",
       "65. 6\n",
       "66. 6.75\n",
       "67. 6\n",
       "68. 6.75\n",
       "69. 6\n",
       "70. 6.75\n",
       "71. 6\n",
       "72. 6.75\n",
       "73. 6\n",
       "74. 6\n",
       "75. 6\n",
       "76. 6\n",
       "77. 6\n",
       "78. 6\n",
       "79. 6\n",
       "80. 6\n",
       "81. 6\n",
       "82. 6.75\n",
       "83. 6\n",
       "84. 6.75\n",
       "85. 6\n",
       "86. 6.75\n",
       "87. 6\n",
       "88. 6.75\n",
       "89. 6.75\n",
       "90. 6\n",
       "91. 6.75\n",
       "92. 6\n",
       "93. 6.75\n",
       "94. 6\n",
       "95. 6.75\n",
       "96. 6\n",
       "97. 6.75\n",
       "98. 6\n",
       "99. 6.75\n",
       "100. 6\n",
       "101. 6.75\n",
       "102. 6\n",
       "103. 6.75\n",
       "104. 6\n",
       "105. 6.75\n",
       "106. 6\n",
       "107. 6.75\n",
       "108. 6\n",
       "109. 6.75\n",
       "110. 6\n",
       "111. 6.75\n",
       "112. 6\n",
       "113. 6\n",
       "114. 6\n",
       "115. 6\n",
       "116. 6\n",
       "117. 6\n",
       "118. 6\n",
       "119. 6\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00\n",
       " [16] 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75\n",
       " [31] 6.00 6.75 6.00 6.00 6.00 6.00 6.00 6.00 6.00 6.00 6.00 6.75 6.00 6.75 6.00\n",
       " [46] 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75\n",
       " [61] 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.00 6.00\n",
       " [76] 6.00 6.00 6.00 6.00 6.00 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.75 6.00\n",
       " [91] 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.75\n",
       "[106] 6.00 6.75 6.00 6.75 6.00 6.75 6.00 6.00 6.00 6.00 6.00 6.00 6.00 6.00"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to identify the pH of the PyOM addition\n",
    "#Fix pH from baseline soil pH to \"true\" pH after treatment additions\n",
    "sample_data(ps_whitman)$PyOM.pH = ifelse(sample_data(ps_whitman)$PyOM %in% \"PyOM\", 10.0, NA)\n",
    "sample_data(ps_whitman)$pH = ifelse(sample_data(ps_whitman)$PyOM %in% \"PyOM\", 6.75, 6)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_whitman, \"whitman.ps\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Ye"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ps_ye = import_biom(\"Ye_OTU_table-v3/feature-table-metaD-tax_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x = data.frame(tax_table(ps_ye))\n",
    "\n",
    "colnames(x) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "x$Domain = gsub(\"D_0__\", \"\", as.character(x$Domain))\n",
    "x$Phylum = gsub(\"D_1__\", \"\", as.character(x$Phylum))\n",
    "x$Class = gsub(\"D_2__\", \"\", as.character(x$Class))\n",
    "x$Order = gsub(\"D_3__\", \"\", as.character(x$Order))\n",
    "x$Family = gsub(\"D_4__\", \"\", as.character(x$Family))\n",
    "x$Genus = gsub(\"D_5__\", \"\", as.character(x$Genus))\n",
    "x$Species = gsub(\"D_6__\", \"\", as.character(x$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x=tax_table(as.matrix(x,dimnames=list(row.names(x),colnames(x))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_ye)=x\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "tax_table(ps_ye)\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_ye)$Treatment = revalue(sample_data(ps_ye)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_ye)$SoilType = revalue(sample_data(ps_ye)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_ye)$Feedstock = revalue(sample_data(ps_ye)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_ye)$Habitat = revalue(sample_data(ps_ye)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_ye)$CurrentLandUse = revalue(sample_data(ps_ye)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_ye)$PyOM.Temp = revalue(sample_data(ps_ye)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_ye)$Depth = revalue(sample_data(ps_ye)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'Avail_K'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'TotalN'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'TotalC'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'Org_C'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Nitrate'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'NCBISampleID'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Avail_P'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'g.per.kg.PyOM'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'TSN'</li>\n",
       "\t<li>'Ammonium'</li>\n",
       "\t<li>'EC'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Author'\n",
       "\\item 'pH'\n",
       "\\item 'Region'\n",
       "\\item 'BioSample'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'Avail\\_K'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'TotalN'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'TotalC'\n",
       "\\item 'Organism'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'Habitat'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'Org\\_C'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'Clay'\n",
       "\\item 'Feedstock'\n",
       "\\item 'BioProject'\n",
       "\\item 'Nitrate'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Sand'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'TargetGene'\n",
       "\\item '\\%SOM'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'SRA'\n",
       "\\item 'NCBISampleID'\n",
       "\\item 'Silt'\n",
       "\\item 'Avail\\_P'\n",
       "\\item 'Treatment'\n",
       "\\item 'g.per.kg.PyOM'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Crop'\n",
       "\\item 'TSN'\n",
       "\\item 'Ammonium'\n",
       "\\item 'EC'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'PyOM'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'RevPrimerName'\n",
       "2. 'Author'\n",
       "3. 'pH'\n",
       "4. 'Region'\n",
       "5. 'BioSample'\n",
       "6. 'Sampling_Depth_cm'\n",
       "7. 'Avail_K'\n",
       "8. 'ForwardPrimerSequence'\n",
       "9. 'TotalN'\n",
       "10. 'CurrentLandUse'\n",
       "11. 'TotalC'\n",
       "12. 'Organism'\n",
       "13. 'SequencingMethod'\n",
       "14. 'Habitat'\n",
       "15. 'Incubation_Time_Weeks'\n",
       "16. 'Org_C'\n",
       "17. 'GeographicalLocation'\n",
       "18. 'Clay'\n",
       "19. 'Feedstock'\n",
       "20. 'BioProject'\n",
       "21. 'Nitrate'\n",
       "22. 'ReversePrimerSequence'\n",
       "23. 'Sand'\n",
       "24. 'CollectionYear'\n",
       "25. 'TargetGene'\n",
       "26. '%SOM'\n",
       "27. 'PyOM.Temp'\n",
       "28. 'NucleicAcidExtraction'\n",
       "29. 'SRA'\n",
       "30. 'NCBISampleID'\n",
       "31. 'Silt'\n",
       "32. 'Avail_P'\n",
       "33. 'Treatment'\n",
       "34. 'g.per.kg.PyOM'\n",
       "35. 'FwdBarcode'\n",
       "36. 'ForwardPrimerName'\n",
       "37. 'Crop'\n",
       "38. 'TSN'\n",
       "39. 'Ammonium'\n",
       "40. 'EC'\n",
       "41. 'RevBarcode'\n",
       "42. 'PyOM'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"RevPrimerName\"         \"Author\"                \"pH\"                   \n",
       " [4] \"Region\"                \"BioSample\"             \"Sampling_Depth_cm\"    \n",
       " [7] \"Avail_K\"               \"ForwardPrimerSequence\" \"TotalN\"               \n",
       "[10] \"CurrentLandUse\"        \"TotalC\"                \"Organism\"             \n",
       "[13] \"SequencingMethod\"      \"Habitat\"               \"Incubation_Time_Weeks\"\n",
       "[16] \"Org_C\"                 \"GeographicalLocation\"  \"Clay\"                 \n",
       "[19] \"Feedstock\"             \"BioProject\"            \"Nitrate\"              \n",
       "[22] \"ReversePrimerSequence\" \"Sand\"                  \"CollectionYear\"       \n",
       "[25] \"TargetGene\"            \"%SOM\"                  \"PyOM.Temp\"            \n",
       "[28] \"NucleicAcidExtraction\" \"SRA\"                   \"NCBISampleID\"         \n",
       "[31] \"Silt\"                  \"Avail_P\"               \"Treatment\"            \n",
       "[34] \"g.per.kg.PyOM\"         \"FwdBarcode\"            \"ForwardPrimerName\"    \n",
       "[37] \"Crop\"                  \"TSN\"                   \"Ammonium\"             \n",
       "[40] \"EC\"                    \"RevBarcode\"            \"PyOM\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'Avail_K'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'TotalN'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'TotalC'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'Org_C'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Nitrate'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'NCBISampleID'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Avail_P'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'TSN'</li>\n",
       "\t<li>'Ammonium'</li>\n",
       "\t<li>'EC'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Author'\n",
       "\\item 'pH'\n",
       "\\item 'Region'\n",
       "\\item 'BioSample'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'Avail\\_K'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'TotalN'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'TotalC'\n",
       "\\item 'Organism'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'Habitat'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'Org\\_C'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'Clay'\n",
       "\\item 'Feedstock'\n",
       "\\item 'BioProject'\n",
       "\\item 'Nitrate'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Sand'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'TargetGene'\n",
       "\\item '\\%SOM'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'SRA'\n",
       "\\item 'NCBISampleID'\n",
       "\\item 'Silt'\n",
       "\\item 'Avail\\_P'\n",
       "\\item 'Treatment'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Crop'\n",
       "\\item 'TSN'\n",
       "\\item 'Ammonium'\n",
       "\\item 'EC'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'PyOM'\n",
       "\\item 'SoilType'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'RevPrimerName'\n",
       "2. 'Author'\n",
       "3. 'pH'\n",
       "4. 'Region'\n",
       "5. 'BioSample'\n",
       "6. 'Sampling_Depth_cm'\n",
       "7. 'Avail_K'\n",
       "8. 'ForwardPrimerSequence'\n",
       "9. 'TotalN'\n",
       "10. 'CurrentLandUse'\n",
       "11. 'TotalC'\n",
       "12. 'Organism'\n",
       "13. 'SequencingMethod'\n",
       "14. 'Habitat'\n",
       "15. 'Incubation_Time_Weeks'\n",
       "16. 'Org_C'\n",
       "17. 'GeographicalLocation'\n",
       "18. 'Clay'\n",
       "19. 'Feedstock'\n",
       "20. 'BioProject'\n",
       "21. 'Nitrate'\n",
       "22. 'ReversePrimerSequence'\n",
       "23. 'Sand'\n",
       "24. 'CollectionYear'\n",
       "25. 'TargetGene'\n",
       "26. '%SOM'\n",
       "27. 'PyOM.Temp'\n",
       "28. 'NucleicAcidExtraction'\n",
       "29. 'SRA'\n",
       "30. 'NCBISampleID'\n",
       "31. 'Silt'\n",
       "32. 'Avail_P'\n",
       "33. 'Treatment'\n",
       "34. 't.per.ha.PyOM'\n",
       "35. 'FwdBarcode'\n",
       "36. 'ForwardPrimerName'\n",
       "37. 'Crop'\n",
       "38. 'TSN'\n",
       "39. 'Ammonium'\n",
       "40. 'EC'\n",
       "41. 'RevBarcode'\n",
       "42. 'PyOM'\n",
       "43. 'SoilType'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"RevPrimerName\"         \"Author\"                \"pH\"                   \n",
       " [4] \"Region\"                \"BioSample\"             \"Sampling_Depth_cm\"    \n",
       " [7] \"Avail_K\"               \"ForwardPrimerSequence\" \"TotalN\"               \n",
       "[10] \"CurrentLandUse\"        \"TotalC\"                \"Organism\"             \n",
       "[13] \"SequencingMethod\"      \"Habitat\"               \"Incubation_Time_Weeks\"\n",
       "[16] \"Org_C\"                 \"GeographicalLocation\"  \"Clay\"                 \n",
       "[19] \"Feedstock\"             \"BioProject\"            \"Nitrate\"              \n",
       "[22] \"ReversePrimerSequence\" \"Sand\"                  \"CollectionYear\"       \n",
       "[25] \"TargetGene\"            \"%SOM\"                  \"PyOM.Temp\"            \n",
       "[28] \"NucleicAcidExtraction\" \"SRA\"                   \"NCBISampleID\"         \n",
       "[31] \"Silt\"                  \"Avail_P\"               \"Treatment\"            \n",
       "[34] \"t.per.ha.PyOM\"         \"FwdBarcode\"            \"ForwardPrimerName\"    \n",
       "[37] \"Crop\"                  \"TSN\"                   \"Ammonium\"             \n",
       "[40] \"EC\"                    \"RevBarcode\"            \"PyOM\"                 \n",
       "[43] \"SoilType\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "\t<li>'Fluvisol'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\item 'Fluvisol'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Fluvisol'\n",
       "2. 'Fluvisol'\n",
       "3. 'Fluvisol'\n",
       "4. 'Fluvisol'\n",
       "5. 'Fluvisol'\n",
       "6. 'Fluvisol'\n",
       "7. 'Fluvisol'\n",
       "8. 'Fluvisol'\n",
       "9. 'Fluvisol'\n",
       "10. 'Fluvisol'\n",
       "11. 'Fluvisol'\n",
       "12. 'Fluvisol'\n",
       "13. 'Fluvisol'\n",
       "14. 'Fluvisol'\n",
       "15. 'Fluvisol'\n",
       "16. 'Fluvisol'\n",
       "17. 'Fluvisol'\n",
       "18. 'Fluvisol'\n",
       "19. 'Fluvisol'\n",
       "20. 'Fluvisol'\n",
       "21. 'Fluvisol'\n",
       "22. 'Fluvisol'\n",
       "23. 'Fluvisol'\n",
       "24. 'Fluvisol'\n",
       "25. 'Fluvisol'\n",
       "26. 'Fluvisol'\n",
       "27. 'Fluvisol'\n",
       "28. 'Fluvisol'\n",
       "29. 'Fluvisol'\n",
       "30. 'Fluvisol'\n",
       "31. 'Fluvisol'\n",
       "32. 'Fluvisol'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\"\n",
       " [7] \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\"\n",
       "[13] \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\"\n",
       "[19] \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\"\n",
       "[25] \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\" \"Fluvisol\"\n",
       "[31] \"Fluvisol\" \"Fluvisol\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and t.per.ha.PyOM are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_ye))[names(sample_data(ps_ye)) == \"PyOM-Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_ye))[names(sample_data(ps_ye)) == \"g.per.kg.PyOM\"] <- \"t.per.ha.PyOM\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('soil with biochar-mineral complex', '1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\", 'soil')\n",
    "OMAdded = c('soil with organic fertilizer and biochar-mineral complex', 'soil with organic fertilizer','Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_ye)$PyOM = ifelse(sample_data(ps_ye)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_ye)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_ye)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>6.98</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 6.98\n",
       "\\item 6.98\n",
       "\\item 6.98\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item 6.98\n",
       "\\item 6.98\n",
       "\\item 6.98\n",
       "\\item 6.98\n",
       "\\item 6.98\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. &lt;NA&gt;\n",
       "2. &lt;NA&gt;\n",
       "3. &lt;NA&gt;\n",
       "4. &lt;NA&gt;\n",
       "5. &lt;NA&gt;\n",
       "6. &lt;NA&gt;\n",
       "7. 6.98\n",
       "8. 6.98\n",
       "9. 6.98\n",
       "10. &lt;NA&gt;\n",
       "11. &lt;NA&gt;\n",
       "12. &lt;NA&gt;\n",
       "13. &lt;NA&gt;\n",
       "14. &lt;NA&gt;\n",
       "15. &lt;NA&gt;\n",
       "16. &lt;NA&gt;\n",
       "17. &lt;NA&gt;\n",
       "18. &lt;NA&gt;\n",
       "19. &lt;NA&gt;\n",
       "20. &lt;NA&gt;\n",
       "21. &lt;NA&gt;\n",
       "22. &lt;NA&gt;\n",
       "23. 6.98\n",
       "24. 6.98\n",
       "25. 6.98\n",
       "26. 6.98\n",
       "27. 6.98\n",
       "28. &lt;NA&gt;\n",
       "29. &lt;NA&gt;\n",
       "30. &lt;NA&gt;\n",
       "31. &lt;NA&gt;\n",
       "32. &lt;NA&gt;\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1]   NA   NA   NA   NA   NA   NA 6.98 6.98 6.98   NA   NA   NA   NA   NA   NA\n",
       "[16]   NA   NA   NA   NA   NA   NA   NA 6.98 6.98 6.98 6.98 6.98   NA   NA   NA\n",
       "[31]   NA   NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'Avail_K'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'TotalN'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'TotalC'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'Org_C'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'Nitrate'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'NCBISampleID'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Avail_P'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'TSN'</li>\n",
       "\t<li>'Ammonium'</li>\n",
       "\t<li>'EC'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'PyOM.pH'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'Author'\n",
       "\\item 'pH'\n",
       "\\item 'Region'\n",
       "\\item 'BioSample'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'Avail\\_K'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'TotalN'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'TotalC'\n",
       "\\item 'Organism'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'Habitat'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'Org\\_C'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'Clay'\n",
       "\\item 'Feedstock'\n",
       "\\item 'BioProject'\n",
       "\\item 'Nitrate'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'Sand'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'TargetGene'\n",
       "\\item '\\%SOM'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'SRA'\n",
       "\\item 'NCBISampleID'\n",
       "\\item 'Silt'\n",
       "\\item 'Avail\\_P'\n",
       "\\item 'Treatment'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Crop'\n",
       "\\item 'TSN'\n",
       "\\item 'Ammonium'\n",
       "\\item 'EC'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'PyOM'\n",
       "\\item 'SoilType'\n",
       "\\item 'PyOM.pH'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'RevPrimerName'\n",
       "2. 'Author'\n",
       "3. 'pH'\n",
       "4. 'Region'\n",
       "5. 'BioSample'\n",
       "6. 'Sampling_Depth_cm'\n",
       "7. 'Avail_K'\n",
       "8. 'ForwardPrimerSequence'\n",
       "9. 'TotalN'\n",
       "10. 'CurrentLandUse'\n",
       "11. 'TotalC'\n",
       "12. 'Organism'\n",
       "13. 'SequencingMethod'\n",
       "14. 'Habitat'\n",
       "15. 'Incubation_Time_Weeks'\n",
       "16. 'Org_C'\n",
       "17. 'GeographicalLocation'\n",
       "18. 'Clay'\n",
       "19. 'Feedstock'\n",
       "20. 'BioProject'\n",
       "21. 'Nitrate'\n",
       "22. 'ReversePrimerSequence'\n",
       "23. 'Sand'\n",
       "24. 'CollectionYear'\n",
       "25. 'TargetGene'\n",
       "26. '%SOM'\n",
       "27. 'PyOM.Temp'\n",
       "28. 'NucleicAcidExtraction'\n",
       "29. 'SRA'\n",
       "30. 'NCBISampleID'\n",
       "31. 'Silt'\n",
       "32. 'Avail_P'\n",
       "33. 'Treatment'\n",
       "34. 't.per.ha.PyOM'\n",
       "35. 'FwdBarcode'\n",
       "36. 'ForwardPrimerName'\n",
       "37. 'Crop'\n",
       "38. 'TSN'\n",
       "39. 'Ammonium'\n",
       "40. 'EC'\n",
       "41. 'RevBarcode'\n",
       "42. 'PyOM'\n",
       "43. 'SoilType'\n",
       "44. 'PyOM.pH'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"RevPrimerName\"         \"Author\"                \"pH\"                   \n",
       " [4] \"Region\"                \"BioSample\"             \"Sampling_Depth_cm\"    \n",
       " [7] \"Avail_K\"               \"ForwardPrimerSequence\" \"TotalN\"               \n",
       "[10] \"CurrentLandUse\"        \"TotalC\"                \"Organism\"             \n",
       "[13] \"SequencingMethod\"      \"Habitat\"               \"Incubation_Time_Weeks\"\n",
       "[16] \"Org_C\"                 \"GeographicalLocation\"  \"Clay\"                 \n",
       "[19] \"Feedstock\"             \"BioProject\"            \"Nitrate\"              \n",
       "[22] \"ReversePrimerSequence\" \"Sand\"                  \"CollectionYear\"       \n",
       "[25] \"TargetGene\"            \"%SOM\"                  \"PyOM.Temp\"            \n",
       "[28] \"NucleicAcidExtraction\" \"SRA\"                   \"NCBISampleID\"         \n",
       "[31] \"Silt\"                  \"Avail_P\"               \"Treatment\"            \n",
       "[34] \"t.per.ha.PyOM\"         \"FwdBarcode\"            \"ForwardPrimerName\"    \n",
       "[37] \"Crop\"                  \"TSN\"                   \"Ammonium\"             \n",
       "[40] \"EC\"                    \"RevBarcode\"            \"PyOM\"                 \n",
       "[43] \"SoilType\"              \"PyOM.pH\"              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_ye)$PyOM.pH = ifelse(sample_data(ps_ye)$PyOM %in% \"PyOM\", 6.98, NA)\n",
    "\n",
    "#Add soil type as \"Fluvisol\"\n",
    "\n",
    "sample_data(ps_ye)$SoilType = sample_data(ps_ye)$SoilType = \"Fluvisol\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_ye, \"ps.ye\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Nielsen 2014"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "ps_nielsen2014 = import_biom(\"Nielsen2014_OTU_table/feature-table-metaD-tax_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th></th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>79ae2c2f8c966f38184481500c6b8a58</th><td>Bacteria              </td><td>Firmicutes            </td><td>Bacilli               </td><td>Bacillales            </td><td>NA                    </td><td>NA                    </td><td>NA                    </td></tr>\n",
       "\t<tr><th scope=row>5386969700ee27766c23c43d036a3add</th><td>Bacteria              </td><td>Verrucomicrobia       </td><td>Verrucomicrobiae      </td><td>Chthoniobacterales    </td><td>Chthoniobacteraceae   </td><td>Candidatus Udaeobacter</td><td>NA                    </td></tr>\n",
       "\t<tr><th scope=row>27d37bfd668a29fa70eac0c8e621a132</th><td>Bacteria              </td><td>Verrucomicrobia       </td><td>Verrucomicrobiae      </td><td>Chthoniobacterales    </td><td>Chthoniobacteraceae   </td><td>Candidatus Udaeobacter</td><td>NA                    </td></tr>\n",
       "\t<tr><th scope=row>86eff6cc9a77cf0f0e31668d0a2e5618</th><td>Bacteria              </td><td>Actinobacteria        </td><td>Thermoleophilia       </td><td>Gaiellales            </td><td>uncultured            </td><td>NA                    </td><td>NA                    </td></tr>\n",
       "\t<tr><th scope=row>bc84a4db4cfa2319a4b89eb0029cf982</th><td>Bacteria              </td><td>Acidobacteria         </td><td>Subgroup 6            </td><td>NA                    </td><td>NA                    </td><td>NA                    </td><td>NA                    </td></tr>\n",
       "\t<tr><th scope=row>fd0fb74123bd8f51e6eaa34f69036a03</th><td>Bacteria              </td><td>Actinobacteria        </td><td>Actinobacteria        </td><td>Frankiales            </td><td>NA                    </td><td>NA                    </td><td>NA                    </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllll}\n",
       "  & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       "\\hline\n",
       "\t79ae2c2f8c966f38184481500c6b8a58 & Bacteria               & Firmicutes             & Bacilli                & Bacillales             & NA                     & NA                     & NA                    \\\\\n",
       "\t5386969700ee27766c23c43d036a3add & Bacteria               & Verrucomicrobia        & Verrucomicrobiae       & Chthoniobacterales     & Chthoniobacteraceae    & Candidatus Udaeobacter & NA                    \\\\\n",
       "\t27d37bfd668a29fa70eac0c8e621a132 & Bacteria               & Verrucomicrobia        & Verrucomicrobiae       & Chthoniobacterales     & Chthoniobacteraceae    & Candidatus Udaeobacter & NA                    \\\\\n",
       "\t86eff6cc9a77cf0f0e31668d0a2e5618 & Bacteria               & Actinobacteria         & Thermoleophilia        & Gaiellales             & uncultured             & NA                     & NA                    \\\\\n",
       "\tbc84a4db4cfa2319a4b89eb0029cf982 & Bacteria               & Acidobacteria          & Subgroup 6             & NA                     & NA                     & NA                     & NA                    \\\\\n",
       "\tfd0fb74123bd8f51e6eaa34f69036a03 & Bacteria               & Actinobacteria         & Actinobacteria         & Frankiales             & NA                     & NA                     & NA                    \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "| <!--/--> | Domain | Phylum | Class | Order | Family | Genus | Species | \n",
       "|---|---|---|---|---|---|\n",
       "| 79ae2c2f8c966f38184481500c6b8a58 | Bacteria               | Firmicutes             | Bacilli                | Bacillales             | NA                     | NA                     | NA                     | \n",
       "| 5386969700ee27766c23c43d036a3add | Bacteria               | Verrucomicrobia        | Verrucomicrobiae       | Chthoniobacterales     | Chthoniobacteraceae    | Candidatus Udaeobacter | NA                     | \n",
       "| 27d37bfd668a29fa70eac0c8e621a132 | Bacteria               | Verrucomicrobia        | Verrucomicrobiae       | Chthoniobacterales     | Chthoniobacteraceae    | Candidatus Udaeobacter | NA                     | \n",
       "| 86eff6cc9a77cf0f0e31668d0a2e5618 | Bacteria               | Actinobacteria         | Thermoleophilia        | Gaiellales             | uncultured             | NA                     | NA                     | \n",
       "| bc84a4db4cfa2319a4b89eb0029cf982 | Bacteria               | Acidobacteria          | Subgroup 6             | NA                     | NA                     | NA                     | NA                     | \n",
       "| fd0fb74123bd8f51e6eaa34f69036a03 | Bacteria               | Actinobacteria         | Actinobacteria         | Frankiales             | NA                     | NA                     | NA                     | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "                                 Domain   Phylum          Class           \n",
       "79ae2c2f8c966f38184481500c6b8a58 Bacteria Firmicutes      Bacilli         \n",
       "5386969700ee27766c23c43d036a3add Bacteria Verrucomicrobia Verrucomicrobiae\n",
       "27d37bfd668a29fa70eac0c8e621a132 Bacteria Verrucomicrobia Verrucomicrobiae\n",
       "86eff6cc9a77cf0f0e31668d0a2e5618 Bacteria Actinobacteria  Thermoleophilia \n",
       "bc84a4db4cfa2319a4b89eb0029cf982 Bacteria Acidobacteria   Subgroup 6      \n",
       "fd0fb74123bd8f51e6eaa34f69036a03 Bacteria Actinobacteria  Actinobacteria  \n",
       "                                 Order              Family             \n",
       "79ae2c2f8c966f38184481500c6b8a58 Bacillales         NA                 \n",
       "5386969700ee27766c23c43d036a3add Chthoniobacterales Chthoniobacteraceae\n",
       "27d37bfd668a29fa70eac0c8e621a132 Chthoniobacterales Chthoniobacteraceae\n",
       "86eff6cc9a77cf0f0e31668d0a2e5618 Gaiellales         uncultured         \n",
       "bc84a4db4cfa2319a4b89eb0029cf982 NA                 NA                 \n",
       "fd0fb74123bd8f51e6eaa34f69036a03 Frankiales         NA                 \n",
       "                                 Genus                  Species\n",
       "79ae2c2f8c966f38184481500c6b8a58 NA                     NA     \n",
       "5386969700ee27766c23c43d036a3add Candidatus Udaeobacter NA     \n",
       "27d37bfd668a29fa70eac0c8e621a132 Candidatus Udaeobacter NA     \n",
       "86eff6cc9a77cf0f0e31668d0a2e5618 NA                     NA     \n",
       "bc84a4db4cfa2319a4b89eb0029cf982 NA                     NA     \n",
       "fd0fb74123bd8f51e6eaa34f69036a03 NA                     NA     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "##Fixing the tax_table object to \"clean-up\" the column names\n",
    "\n",
    "x = data.frame(tax_table(ps_nielsen2014))\n",
    "\n",
    "colnames(x) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "x$Domain = gsub(\"D_0__\", \"\", as.character(x$Domain))\n",
    "x$Phylum = gsub(\"D_1__\", \"\", as.character(x$Phylum))\n",
    "x$Class = gsub(\"D_2__\", \"\", as.character(x$Class))\n",
    "x$Order = gsub(\"D_3__\", \"\", as.character(x$Order))\n",
    "x$Family = gsub(\"D_4__\", \"\", as.character(x$Family))\n",
    "x$Genus = gsub(\"D_5__\", \"\", as.character(x$Genus))\n",
    "x$Species = gsub(\"D_6__\", \"\", as.character(x$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x=tax_table(as.matrix(x,dimnames=list(row.names(x),colnames(x))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_nielsen2014)=x\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_nielsen2014))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "The following `from` values were not present in `x`: control, unammended, unamended\n",
      "The following `from` values were not present in `x`: mollisol\n",
      "The following `from` values were not present in `x`: none\n",
      "The following `from` values were not present in `x`: agriculture\n",
      "The following `from` values were not present in `x`: agriculture, minimim tillage agriculture field, field\n",
      "The following `from` values were not present in `x`: N/A, 0\n",
      "The following `from` values were not present in `x`: 15cm\n"
     ]
    }
   ],
   "source": [
    "#making sample data variables consistent ; fixing spelling, consitent capital letters, changing different versions \n",
    "#of Agriculture to \"Agriculture\"\n",
    "\n",
    "sample_data(ps_nielsen2014)$Treatment = revalue(sample_data(ps_nielsen2014)$Treatment, c(\"control\"=\"Control\", \"unammended\"=\"Unamended\",\"unamended\"=\"Unamended\"))\n",
    "sample_data(ps_nielsen2014)$SoilType = revalue(sample_data(ps_nielsen2014)$SoilType, c(\"mollisol\"=\"Mollisol\"))\n",
    "sample_data(ps_nielsen2014)$Feedstock = revalue(sample_data(ps_nielsen2014)$Feedstock, c(\"none\"=\"NA\"))\n",
    "sample_data(ps_nielsen2014)$Habitat = revalue(sample_data(ps_nielsen2014)$Habitat, c(\"agriculture\"=\"Agriculture\"))\n",
    "sample_data(ps_nielsen2014)$CurrentLandUse = revalue(sample_data(ps_nielsen2014)$CurrentLandUse, c(\"agriculture\"=\"Agriculture\", \"minimim tillage agriculture field\" = \"Agriculture\", \"field\" = \"Agriculture\"))\n",
    "sample_data(ps_nielsen2014)$PyOM.Temp = revalue(sample_data(ps_nielsen2014)$PyOM.Temp, c(\"N/A\"=\"NA\", \"0\" = \"NA\"))\n",
    "sample_data(ps_nielsen2014)$Depth = revalue(sample_data(ps_nielsen2014)$Depth, c(\"15cm\"=\"15\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>'BioProject'</li>\n",
       "\t<li>'TargetGene'</li>\n",
       "\t<li>'EMBISampleID'</li>\n",
       "\t<li>'Author'</li>\n",
       "\t<li>'TotalC'</li>\n",
       "\t<li>'Treatment'</li>\n",
       "\t<li>'SRA'</li>\n",
       "\t<li>'CollectionYear'</li>\n",
       "\t<li>'PyOM.Temp'</li>\n",
       "\t<li>'pH'</li>\n",
       "\t<li>'ReversePrimerSequence'</li>\n",
       "\t<li>'ForwardPrimerSequence'</li>\n",
       "\t<li>'RevPrimerName'</li>\n",
       "\t<li>'ForwardPrimerName'</li>\n",
       "\t<li>'Silt'</li>\n",
       "\t<li>'Region'</li>\n",
       "\t<li>'Crop'</li>\n",
       "\t<li>'SoilType'</li>\n",
       "\t<li>'t.per.ha.PyOM'</li>\n",
       "\t<li>'Clay'</li>\n",
       "\t<li>'SequencingMethod'</li>\n",
       "\t<li>'BioSample'</li>\n",
       "\t<li>'Cowell_P_mg_kg'</li>\n",
       "\t<li>'Habitat'</li>\n",
       "\t<li>'Sampling_Depth_cm'</li>\n",
       "\t<li>'FwdBarcode'</li>\n",
       "\t<li>'EC'</li>\n",
       "\t<li>'%SOM'</li>\n",
       "\t<li>'Sand'</li>\n",
       "\t<li>'Organism'</li>\n",
       "\t<li>'Ammonium'</li>\n",
       "\t<li>'TotalN'</li>\n",
       "\t<li>'Bray_P_mg_kg'</li>\n",
       "\t<li>'CurrentLandUse'</li>\n",
       "\t<li>'Nitrate'</li>\n",
       "\t<li>'Incubation_Time_Weeks'</li>\n",
       "\t<li>'RevBarcode'</li>\n",
       "\t<li>'Feedstock'</li>\n",
       "\t<li>'GeographicalLocation'</li>\n",
       "\t<li>'NucleicAcidExtraction'</li>\n",
       "\t<li>'PyOM'</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'BioProject'\n",
       "\\item 'TargetGene'\n",
       "\\item 'EMBISampleID'\n",
       "\\item 'Author'\n",
       "\\item 'TotalC'\n",
       "\\item 'Treatment'\n",
       "\\item 'SRA'\n",
       "\\item 'CollectionYear'\n",
       "\\item 'PyOM.Temp'\n",
       "\\item 'pH'\n",
       "\\item 'ReversePrimerSequence'\n",
       "\\item 'ForwardPrimerSequence'\n",
       "\\item 'RevPrimerName'\n",
       "\\item 'ForwardPrimerName'\n",
       "\\item 'Silt'\n",
       "\\item 'Region'\n",
       "\\item 'Crop'\n",
       "\\item 'SoilType'\n",
       "\\item 't.per.ha.PyOM'\n",
       "\\item 'Clay'\n",
       "\\item 'SequencingMethod'\n",
       "\\item 'BioSample'\n",
       "\\item 'Cowell\\_P\\_mg\\_kg'\n",
       "\\item 'Habitat'\n",
       "\\item 'Sampling\\_Depth\\_cm'\n",
       "\\item 'FwdBarcode'\n",
       "\\item 'EC'\n",
       "\\item '\\%SOM'\n",
       "\\item 'Sand'\n",
       "\\item 'Organism'\n",
       "\\item 'Ammonium'\n",
       "\\item 'TotalN'\n",
       "\\item 'Bray\\_P\\_mg\\_kg'\n",
       "\\item 'CurrentLandUse'\n",
       "\\item 'Nitrate'\n",
       "\\item 'Incubation\\_Time\\_Weeks'\n",
       "\\item 'RevBarcode'\n",
       "\\item 'Feedstock'\n",
       "\\item 'GeographicalLocation'\n",
       "\\item 'NucleicAcidExtraction'\n",
       "\\item 'PyOM'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'BioProject'\n",
       "2. 'TargetGene'\n",
       "3. 'EMBISampleID'\n",
       "4. 'Author'\n",
       "5. 'TotalC'\n",
       "6. 'Treatment'\n",
       "7. 'SRA'\n",
       "8. 'CollectionYear'\n",
       "9. 'PyOM.Temp'\n",
       "10. 'pH'\n",
       "11. 'ReversePrimerSequence'\n",
       "12. 'ForwardPrimerSequence'\n",
       "13. 'RevPrimerName'\n",
       "14. 'ForwardPrimerName'\n",
       "15. 'Silt'\n",
       "16. 'Region'\n",
       "17. 'Crop'\n",
       "18. 'SoilType'\n",
       "19. 't.per.ha.PyOM'\n",
       "20. 'Clay'\n",
       "21. 'SequencingMethod'\n",
       "22. 'BioSample'\n",
       "23. 'Cowell_P_mg_kg'\n",
       "24. 'Habitat'\n",
       "25. 'Sampling_Depth_cm'\n",
       "26. 'FwdBarcode'\n",
       "27. 'EC'\n",
       "28. '%SOM'\n",
       "29. 'Sand'\n",
       "30. 'Organism'\n",
       "31. 'Ammonium'\n",
       "32. 'TotalN'\n",
       "33. 'Bray_P_mg_kg'\n",
       "34. 'CurrentLandUse'\n",
       "35. 'Nitrate'\n",
       "36. 'Incubation_Time_Weeks'\n",
       "37. 'RevBarcode'\n",
       "38. 'Feedstock'\n",
       "39. 'GeographicalLocation'\n",
       "40. 'NucleicAcidExtraction'\n",
       "41. 'PyOM'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"BioProject\"            \"TargetGene\"            \"EMBISampleID\"         \n",
       " [4] \"Author\"                \"TotalC\"                \"Treatment\"            \n",
       " [7] \"SRA\"                   \"CollectionYear\"        \"PyOM.Temp\"            \n",
       "[10] \"pH\"                    \"ReversePrimerSequence\" \"ForwardPrimerSequence\"\n",
       "[13] \"RevPrimerName\"         \"ForwardPrimerName\"     \"Silt\"                 \n",
       "[16] \"Region\"                \"Crop\"                  \"SoilType\"             \n",
       "[19] \"t.per.ha.PyOM\"         \"Clay\"                  \"SequencingMethod\"     \n",
       "[22] \"BioSample\"             \"Cowell_P_mg_kg\"        \"Habitat\"              \n",
       "[25] \"Sampling_Depth_cm\"     \"FwdBarcode\"            \"EC\"                   \n",
       "[28] \"%SOM\"                  \"Sand\"                  \"Organism\"             \n",
       "[31] \"Ammonium\"              \"TotalN\"                \"Bray_P_mg_kg\"         \n",
       "[34] \"CurrentLandUse\"        \"Nitrate\"               \"Incubation_Time_Weeks\"\n",
       "[37] \"RevBarcode\"            \"Feedstock\"             \"GeographicalLocation\" \n",
       "[40] \"NucleicAcidExtraction\" \"PyOM\"                 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Fixing the titles so PyOM.Temp and t.per.ha.PyOM are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_nielsen2014))[names(sample_data(ps_nielsen2014)) == \"t_per_ha_PyOM\"] <- \"t.per.ha.PyOM\"\n",
    "names(sample_data(ps_nielsen2014))[names(sample_data(ps_nielsen2014)) == \"PyOM_Temp\"] <- \"PyOM.Temp\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('Low biochar', 'High biochar', 'soil with biochar-mineral complex', '1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\", 'soil','Fertilizer')\n",
    "OMAdded = c('soil with organic fertilizer and biochar-mineral complex', 'soil with organic fertilizer','Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_nielsen2014)$PyOM = ifelse(sample_data(ps_nielsen2014)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_nielsen2014)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_nielsen2014)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<ol class=list-inline>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>6.8</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "\t<li>&lt;NA&gt;</li>\n",
       "</ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item 6.8\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\item <NA>\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 6.8\n",
       "2. 6.8\n",
       "3. 6.8\n",
       "4. 6.8\n",
       "5. 6.8\n",
       "6. 6.8\n",
       "7. 6.8\n",
       "8. 6.8\n",
       "9. 6.8\n",
       "10. 6.8\n",
       "11. 6.8\n",
       "12. 6.8\n",
       "13. 6.8\n",
       "14. 6.8\n",
       "15. 6.8\n",
       "16. 6.8\n",
       "17. 6.8\n",
       "18. 6.8\n",
       "19. 6.8\n",
       "20. 6.8\n",
       "21. 6.8\n",
       "22. 6.8\n",
       "23. 6.8\n",
       "24. 6.8\n",
       "25. 6.8\n",
       "26. 6.8\n",
       "27. 6.8\n",
       "28. 6.8\n",
       "29. 6.8\n",
       "30. 6.8\n",
       "31. 6.8\n",
       "32. 6.8\n",
       "33. 6.8\n",
       "34. &lt;NA&gt;\n",
       "35. &lt;NA&gt;\n",
       "36. &lt;NA&gt;\n",
       "37. &lt;NA&gt;\n",
       "38. &lt;NA&gt;\n",
       "39. &lt;NA&gt;\n",
       "40. &lt;NA&gt;\n",
       "41. &lt;NA&gt;\n",
       "42. &lt;NA&gt;\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8\n",
       "[20] 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8 6.8  NA  NA  NA  NA  NA\n",
       "[39]  NA  NA  NA  NA"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Create new column \"PyOM.pH\" to add the PyOM pH to the sample data\n",
    "\n",
    "sample_data(ps_nielsen2014)$PyOM.pH = ifelse(sample_data(ps_nielsen2014)$PyOM %in% \"PyOM\", 6.8, NA)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_nielsen2014, \"ps.nielsen2014\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Xu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "ps_Xu = import_biom(\"Xu_OTU_table/feature-table-metaD-tax_json.biom\", parseFunction=parse_taxonomy_default)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A taxonomyTable: 6  7 of type chr</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Domain</th><th scope=col>Phylum</th><th scope=col>Class</th><th scope=col>Order</th><th scope=col>Family</th><th scope=col>Genus</th><th scope=col>Species</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>a2f64386bf9b31f4174e4a9e338afe6a</th><td>Bacteria</td><td>Proteobacteria</td><td>Gammaproteobacteria</td><td>Betaproteobacteriales</td><td>Nitrosomonadaceae</td><td>MND1      </td><td>NA                             </td></tr>\n",
       "\t<tr><th scope=row>ed67895944faf08e227f05155ee07889</th><td>Bacteria</td><td>Proteobacteria</td><td>Alphaproteobacteria</td><td>Sphingomonadales     </td><td>Sphingomonadaceae</td><td>NA        </td><td>NA                             </td></tr>\n",
       "\t<tr><th scope=row>1f5b5b8cda6ae40b783f5b1e8b37007b</th><td>Bacteria</td><td>Proteobacteria</td><td>Gammaproteobacteria</td><td>Betaproteobacteriales</td><td>TRA3-20          </td><td>NA        </td><td>NA                             </td></tr>\n",
       "\t<tr><th scope=row>e5505c99dafc3b62ffa8c60b463b09bf</th><td>Bacteria</td><td>Proteobacteria</td><td>Alphaproteobacteria</td><td>Dongiales            </td><td>Dongiaceae       </td><td>Dongia    </td><td>NA                             </td></tr>\n",
       "\t<tr><th scope=row>e972c196dc49da1d1a67150771f2be42</th><td>Bacteria</td><td>Proteobacteria</td><td>Deltaproteobacteria</td><td>NA                   </td><td>NA               </td><td>NA        </td><td>NA                             </td></tr>\n",
       "\t<tr><th scope=row>e6536447b654c8f6d13315c1eeeaa516</th><td>Bacteria</td><td>Proteobacteria</td><td>Gammaproteobacteria</td><td>Betaproteobacteriales</td><td>Burkholderiaceae </td><td>uncultured</td><td>uncultured beta proteobacterium</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A taxonomyTable: 6  7 of type chr\n",
       "\\begin{tabular}{r|lllllll}\n",
       "  & Domain & Phylum & Class & Order & Family & Genus & Species\\\\\n",
       "\\hline\n",
       "\ta2f64386bf9b31f4174e4a9e338afe6a & Bacteria & Proteobacteria & Gammaproteobacteria & Betaproteobacteriales & Nitrosomonadaceae & MND1       & NA                             \\\\\n",
       "\ted67895944faf08e227f05155ee07889 & Bacteria & Proteobacteria & Alphaproteobacteria & Sphingomonadales      & Sphingomonadaceae & NA         & NA                             \\\\\n",
       "\t1f5b5b8cda6ae40b783f5b1e8b37007b & Bacteria & Proteobacteria & Gammaproteobacteria & Betaproteobacteriales & TRA3-20           & NA         & NA                             \\\\\n",
       "\te5505c99dafc3b62ffa8c60b463b09bf & Bacteria & Proteobacteria & Alphaproteobacteria & Dongiales             & Dongiaceae        & Dongia     & NA                             \\\\\n",
       "\te972c196dc49da1d1a67150771f2be42 & Bacteria & Proteobacteria & Deltaproteobacteria & NA                    & NA                & NA         & NA                             \\\\\n",
       "\te6536447b654c8f6d13315c1eeeaa516 & Bacteria & Proteobacteria & Gammaproteobacteria & Betaproteobacteriales & Burkholderiaceae  & uncultured & uncultured beta proteobacterium\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A taxonomyTable: 6  7 of type chr\n",
       "\n",
       "| <!--/--> | Domain | Phylum | Class | Order | Family | Genus | Species |\n",
       "|---|---|---|---|---|---|---|---|\n",
       "| a2f64386bf9b31f4174e4a9e338afe6a | Bacteria | Proteobacteria | Gammaproteobacteria | Betaproteobacteriales | Nitrosomonadaceae | MND1       | NA                              |\n",
       "| ed67895944faf08e227f05155ee07889 | Bacteria | Proteobacteria | Alphaproteobacteria | Sphingomonadales      | Sphingomonadaceae | NA         | NA                              |\n",
       "| 1f5b5b8cda6ae40b783f5b1e8b37007b | Bacteria | Proteobacteria | Gammaproteobacteria | Betaproteobacteriales | TRA3-20           | NA         | NA                              |\n",
       "| e5505c99dafc3b62ffa8c60b463b09bf | Bacteria | Proteobacteria | Alphaproteobacteria | Dongiales             | Dongiaceae        | Dongia     | NA                              |\n",
       "| e972c196dc49da1d1a67150771f2be42 | Bacteria | Proteobacteria | Deltaproteobacteria | NA                    | NA                | NA         | NA                              |\n",
       "| e6536447b654c8f6d13315c1eeeaa516 | Bacteria | Proteobacteria | Gammaproteobacteria | Betaproteobacteriales | Burkholderiaceae  | uncultured | uncultured beta proteobacterium |\n",
       "\n"
      ],
      "text/plain": [
       "                                 Domain   Phylum         Class              \n",
       "a2f64386bf9b31f4174e4a9e338afe6a Bacteria Proteobacteria Gammaproteobacteria\n",
       "ed67895944faf08e227f05155ee07889 Bacteria Proteobacteria Alphaproteobacteria\n",
       "1f5b5b8cda6ae40b783f5b1e8b37007b Bacteria Proteobacteria Gammaproteobacteria\n",
       "e5505c99dafc3b62ffa8c60b463b09bf Bacteria Proteobacteria Alphaproteobacteria\n",
       "e972c196dc49da1d1a67150771f2be42 Bacteria Proteobacteria Deltaproteobacteria\n",
       "e6536447b654c8f6d13315c1eeeaa516 Bacteria Proteobacteria Gammaproteobacteria\n",
       "                                 Order                 Family           \n",
       "a2f64386bf9b31f4174e4a9e338afe6a Betaproteobacteriales Nitrosomonadaceae\n",
       "ed67895944faf08e227f05155ee07889 Sphingomonadales      Sphingomonadaceae\n",
       "1f5b5b8cda6ae40b783f5b1e8b37007b Betaproteobacteriales TRA3-20          \n",
       "e5505c99dafc3b62ffa8c60b463b09bf Dongiales             Dongiaceae       \n",
       "e972c196dc49da1d1a67150771f2be42 NA                    NA               \n",
       "e6536447b654c8f6d13315c1eeeaa516 Betaproteobacteriales Burkholderiaceae \n",
       "                                 Genus      Species                        \n",
       "a2f64386bf9b31f4174e4a9e338afe6a MND1       NA                             \n",
       "ed67895944faf08e227f05155ee07889 NA         NA                             \n",
       "1f5b5b8cda6ae40b783f5b1e8b37007b NA         NA                             \n",
       "e5505c99dafc3b62ffa8c60b463b09bf Dongia     NA                             \n",
       "e972c196dc49da1d1a67150771f2be42 NA         NA                             \n",
       "e6536447b654c8f6d13315c1eeeaa516 uncultured uncultured beta proteobacterium"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "\n",
    "x = data.frame(tax_table(ps_Xu))\n",
    "\n",
    "colnames(x) = c(\"Domain\", \"Phylum\", \"Class\", \"Order\", \"Family\", \"Genus\", \"Species\")\n",
    "# Assigning the proper column names instead of SILVA ranks\n",
    "x$Domain = gsub(\"D_0__\", \"\", as.character(x$Domain))\n",
    "x$Phylum = gsub(\"D_1__\", \"\", as.character(x$Phylum))\n",
    "x$Class = gsub(\"D_2__\", \"\", as.character(x$Class))\n",
    "x$Order = gsub(\"D_3__\", \"\", as.character(x$Order))\n",
    "x$Family = gsub(\"D_4__\", \"\", as.character(x$Family))\n",
    "x$Genus = gsub(\"D_5__\", \"\", as.character(x$Genus))\n",
    "x$Species = gsub(\"D_6__\", \"\", as.character(x$Species))\n",
    "# Substituting the characters we don't want with nothing in the taxonomy\n",
    "\n",
    "x=tax_table(as.matrix(x,dimnames=list(row.names(x),colnames(x))))\n",
    "# Turning it into a taxonomy table, while saving the rownames and column names\n",
    "\n",
    "tax_table(ps_Xu)=x\n",
    "# Reassigning the taxonomy table in ps_xxx to the new modified one\n",
    "\n",
    "head(tax_table(ps_Xu))\n",
    "# Check for success"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<caption>A sample_data: 5  43</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>X.SOM</th><th scope=col>TotalN</th><th scope=col>t.per.ha.PyOM</th><th scope=col>BioSample</th><th scope=col>ForwardPrimerName</th><th scope=col>RevBarcode</th><th scope=col>Crop</th><th scope=col>Avail.P</th><th scope=col>Incubation.Time.Weeks</th><th scope=col>TSN</th><th scope=col></th><th scope=col>GeographicalLocation</th><th scope=col>Avail.K</th><th scope=col>FwdBarcode</th><th scope=col>NCBISampleID</th><th scope=col>CurrentLandUse</th><th scope=col>PyOM.Temp</th><th scope=col>SequencingMethod</th><th scope=col>Org.C</th><th scope=col>Nitrate</th><th scope=col>PyOM</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>B0-X</th><td></td><td></td><td>0  </td><td>NA</td><td>338F</td><td></td><td></td><td></td><td>20</td><td></td><td></td><td>China</td><td></td><td></td><td>NA</td><td>Agriculture</td><td>NA </td><td>Illumina HiSeq</td><td>16.4</td><td></td><td>Control</td></tr>\n",
       "\t<tr><th scope=row>B1-X</th><td></td><td></td><td>20 </td><td>NA</td><td>338F</td><td></td><td></td><td></td><td>20</td><td></td><td></td><td>China</td><td></td><td></td><td>NA</td><td>Agriculture</td><td>500</td><td>Illumina HiSeq</td><td>16.4</td><td></td><td>PyOM   </td></tr>\n",
       "\t<tr><th scope=row>B2-X</th><td></td><td></td><td>40 </td><td>NA</td><td>338F</td><td></td><td></td><td></td><td>20</td><td></td><td></td><td>China</td><td></td><td></td><td>NA</td><td>Agriculture</td><td>500</td><td>Illumina HiSeq</td><td>16.4</td><td></td><td>PyOM   </td></tr>\n",
       "\t<tr><th scope=row>B4-X</th><td></td><td></td><td>80 </td><td>NA</td><td>338F</td><td></td><td></td><td></td><td>20</td><td></td><td></td><td>China</td><td></td><td></td><td>NA</td><td>Agriculture</td><td>500</td><td>Illumina HiSeq</td><td>16.4</td><td></td><td>PyOM   </td></tr>\n",
       "\t<tr><th scope=row>B8-X</th><td></td><td></td><td>160</td><td>NA</td><td>338F</td><td></td><td></td><td></td><td>20</td><td></td><td></td><td>China</td><td></td><td></td><td>NA</td><td>Agriculture</td><td>500</td><td>Illumina HiSeq</td><td>16.4</td><td></td><td>PyOM   </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A sample_data: 5  43\n",
       "\\begin{tabular}{r|lllllllllllllllllllllllllllllllllllllllllll}\n",
       "  & X.SOM & TotalN & t.per.ha.PyOM & BioSample & ForwardPrimerName & RevBarcode & Crop & Avail.P & Incubation.Time.Weeks & TSN & Region & Sampling.Depth.cm & Sand & ReversePrimerSequence & Organism & RevPrimerName & Treatment & TotalC & CollectionYear & SRA & EC & Author & ForwardPrimerSequence & Soil.Type & pH & NucleicAcidExtraction & Ammonium & TargetGene & Habitat & BioProject & Clay & Silt & Feedstock & GeographicalLocation & Avail.K & FwdBarcode & NCBISampleID & CurrentLandUse & PyOM.Temp & SequencingMethod & Org.C & Nitrate & PyOM\\\\\n",
       "  & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\tB0-X &  &  & 0   & NA & 338F &  &  &  & 20 &  & V3-V4 &  &  & GGACTACHVGGGTWTCTAAT & Soil & 806R & NA   &  &  & NA & 141.4 & Xu & ACTCCTACGGGAGGCAGCAG & Cambisol & 8.1 & MoBio PowerMax Soil DNA Isolation Kit & 0.8 & 16S &  & NA &  &  & NA         & China &  &  & NA & Agriculture & NA  & Illumina HiSeq & 16.4 &  & Control\\\\\n",
       "\tB1-X &  &  & 20  & NA & 338F &  &  &  & 20 &  & V3-V4 &  &  & GGACTACHVGGGTWTCTAAT & Soil & 806R & PyOM &  &  & NA & 141.4 & Xu & ACTCCTACGGGAGGCAGCAG & Cambisol & 8.1 & MoBio PowerMax Soil DNA Isolation Kit & 0.8 & 16S &  & NA &  &  & Corn Straw & China &  &  & NA & Agriculture & 500 & Illumina HiSeq & 16.4 &  & PyOM   \\\\\n",
       "\tB2-X &  &  & 40  & NA & 338F &  &  &  & 20 &  & V3-V5 &  &  & GGACTACHVGGGTWTCTAAT & Soil & 806R & PyOM &  &  & NA & 141.4 & Xu & ACTCCTACGGGAGGCAGCAG & Cambisol & 8.1 & MoBio PowerMax Soil DNA Isolation Kit & 0.8 & 16S &  & NA &  &  & Corn Straw & China &  &  & NA & Agriculture & 500 & Illumina HiSeq & 16.4 &  & PyOM   \\\\\n",
       "\tB4-X &  &  & 80  & NA & 338F &  &  &  & 20 &  & V3-V6 &  &  & GGACTACHVGGGTWTCTAAT & Soil & 806R & PyOM &  &  & NA & 141.4 & Xu & ACTCCTACGGGAGGCAGCAG & Cambisol & 8.1 & MoBio PowerMax Soil DNA Isolation Kit & 0.8 & 16S &  & NA &  &  & Corn Straw & China &  &  & NA & Agriculture & 500 & Illumina HiSeq & 16.4 &  & PyOM   \\\\\n",
       "\tB8-X &  &  & 160 & NA & 338F &  &  &  & 20 &  & V3-V7 &  &  & GGACTACHVGGGTWTCTAAT & Soil & 806R & PyOM &  &  & NA & 141.4 & Xu & ACTCCTACGGGAGGCAGCAG & Cambisol & 8.1 & MoBio PowerMax Soil DNA Isolation Kit & 0.8 & 16S &  & NA &  &  & Corn Straw & China &  &  & NA & Agriculture & 500 & Illumina HiSeq & 16.4 &  & PyOM   \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A sample_data: 5  43\n",
       "\n",
       "| <!--/--> | X.SOM &lt;chr&gt; | TotalN &lt;chr&gt; | t.per.ha.PyOM &lt;chr&gt; | BioSample &lt;chr&gt; | ForwardPrimerName &lt;chr&gt; | RevBarcode &lt;chr&gt; | Crop &lt;chr&gt; | Avail.P &lt;chr&gt; | Incubation.Time.Weeks &lt;chr&gt; | TSN &lt;chr&gt; |   | GeographicalLocation &lt;chr&gt; | Avail.K &lt;chr&gt; | FwdBarcode &lt;chr&gt; | NCBISampleID &lt;chr&gt; | CurrentLandUse &lt;chr&gt; | PyOM.Temp &lt;chr&gt; | SequencingMethod &lt;chr&gt; | Org.C &lt;chr&gt; | Nitrate &lt;chr&gt; | PyOM &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| B0-X | <!----> | <!----> | 0   | NA | 338F | <!----> | <!----> | <!----> | 20 | <!----> |  | China | <!----> | <!----> | NA | Agriculture | NA  | Illumina HiSeq | 16.4 | <!----> | Control |\n",
       "| B1-X | <!----> | <!----> | 20  | NA | 338F | <!----> | <!----> | <!----> | 20 | <!----> |  | China | <!----> | <!----> | NA | Agriculture | 500 | Illumina HiSeq | 16.4 | <!----> | PyOM    |\n",
       "| B2-X | <!----> | <!----> | 40  | NA | 338F | <!----> | <!----> | <!----> | 20 | <!----> |  | China | <!----> | <!----> | NA | Agriculture | 500 | Illumina HiSeq | 16.4 | <!----> | PyOM    |\n",
       "| B4-X | <!----> | <!----> | 80  | NA | 338F | <!----> | <!----> | <!----> | 20 | <!----> |  | China | <!----> | <!----> | NA | Agriculture | 500 | Illumina HiSeq | 16.4 | <!----> | PyOM    |\n",
       "| B8-X | <!----> | <!----> | 160 | NA | 338F | <!----> | <!----> | <!----> | 20 | <!----> |  | China | <!----> | <!----> | NA | Agriculture | 500 | Illumina HiSeq | 16.4 | <!----> | PyOM    |\n",
       "\n"
      ],
      "text/plain": [
       "     X.SOM TotalN t.per.ha.PyOM BioSample ForwardPrimerName RevBarcode Crop\n",
       "B0-X              0             NA        338F                             \n",
       "B1-X              20            NA        338F                             \n",
       "B2-X              40            NA        338F                             \n",
       "B4-X              80            NA        338F                             \n",
       "B8-X              160           NA        338F                             \n",
       "     Avail.P Incubation.Time.Weeks TSN  GeographicalLocation Avail.K\n",
       "B0-X         20                         China                       \n",
       "B1-X         20                         China                       \n",
       "B2-X         20                         China                       \n",
       "B4-X         20                         China                       \n",
       "B8-X         20                         China                       \n",
       "     FwdBarcode NCBISampleID CurrentLandUse PyOM.Temp SequencingMethod Org.C\n",
       "B0-X            NA           Agriculture    NA        Illumina HiSeq   16.4 \n",
       "B1-X            NA           Agriculture    500       Illumina HiSeq   16.4 \n",
       "B2-X            NA           Agriculture    500       Illumina HiSeq   16.4 \n",
       "B4-X            NA           Agriculture    500       Illumina HiSeq   16.4 \n",
       "B8-X            NA           Agriculture    500       Illumina HiSeq   16.4 \n",
       "     Nitrate PyOM   \n",
       "B0-X         Control\n",
       "B1-X         PyOM   \n",
       "B2-X         PyOM   \n",
       "B4-X         PyOM   \n",
       "B8-X         PyOM   "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(sample_data(ps_Xu))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Fixing the titles so PyOM.Temp, t.per.ha.PyOM, and SoilType are consistent among datasets\n",
    "\n",
    "names(sample_data(ps_Xu))[names(sample_data(ps_Xu)) == \"t-per-ha-PyOM\"] <- \"t.per.ha.PyOM\"\n",
    "names(sample_data(ps_Xu))[names(sample_data(ps_Xu)) == \"PyOM-Temp\"] <- \"PyOM.Temp\"\n",
    "names(sample_data(ps_Xu))[names(sample_data(ps_Xu)) == \"Soil_Type\"] <- \"SoilType\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Add soil type\n",
    "\n",
    "sample_data(ps_Xu)$SoilType = \"Cambisol\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Add treatment variable\n",
    "\n",
    "sample_data(ps_Xu)$Treatment = ifelse(sample_data(ps_Xu)$PyOM.Temp =='500', \"PyOM\", \"NA\")\n",
    "                                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new variables to identify whether char, OM, or nothing was added to the sample\n",
    "\n",
    "CharOnlyAdded = c('Low biochar', 'High biochar', 'soil with biochar-mineral complex', '1%BC300' ,'1%BC600','2%BC300', '2%BC600', 'biochar', 'Biochar + soil', 'Composted biochar + soil', 'High GBC' ,'Low GBC', 'PyOM',  'Soil Incubation with PyOM')\n",
    "NothingAdded = c(\"Control\",\"Unamended\",\"NA\", 'soil','Fertilizer')\n",
    "OMAdded = c('soil with organic fertilizer and biochar-mineral complex', 'soil with organic fertilizer','Compost + biochar + soil', 'Compost + soil','Compost + biochar + soil' ,'Compost + soil','stover' ,'Straw')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Create new column \"PyOM\" to identify whether char was added or not using the above variables\n",
    "\n",
    "sample_data(ps_Xu)$PyOM = ifelse(sample_data(ps_Xu)$Treatment %in% CharOnlyAdded, \"PyOM\",\n",
    "                                        ifelse(sample_data(ps_Xu)$Treatment %in% NothingAdded,\"Control\",\n",
    "                                               ifelse(sample_data(ps_Xu)$Treatment %in% OMAdded, \"OM\", NA)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Save as an R object so that the phyloseq object can be called in future processing\n",
    "\n",
    "saveRDS(ps_Xu, \"ps.xu\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
