{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Qiime2 Pipeline (Version 2018.2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Sequences and create quality-score plot visualizations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Song\n",
    "!qiime tools import \\\n",
    "    --type 'SampleData[PairedEndSequencesWithQuality]' \\\n",
    "    --source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "    --input-path SongBact-seqs/ \\\n",
    "    --output-path SongBact.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data SongBact.qza \\\n",
    "--o-visualization SongBact.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Yao\n",
    "!qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "--source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "--input-path YaoBact-seqs/ \\\n",
    "--output-path yaobact.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data yaobact.qza \\\n",
    "--o-visualization yaobact.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Dai2017\n",
    "!qiime tools import --type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "    --source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "    --input-path Dai2017-seqs/ \\\n",
    "    --output-path Dai2017.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "    --i-data Dai2017.qza \\\n",
    "    --o-visualization Dai2017.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Imparato\n",
    "!qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "--source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "--input-path imparato-seqs/ \\\n",
    "--output-path imparato.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data imparato.qza \\\n",
    "--o-visualization imparato.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Dai2016\n",
    "!qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "--source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "--input-path Dai2016-seqs/ \\\n",
    "--output-path Dai2016.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data Dai2016.qza \\\n",
    "--o-visualization Dai2016.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Wu\n",
    "!qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "--source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "--input-path wu-seqs/ \\\n",
    "--output-path wu.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data wu.qza \\\n",
    "--o-visualization wu.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Whitman\n",
    "!qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "--source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "--input-path whitman-seqs/ \\\n",
    "--output-path whitman.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data whitman.qza \\\n",
    "--o-visualization whitman.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Ye\n",
    "!qiime tools import \\\n",
    "--type 'SampleData[PairedEndSequencesWithQuality]'  \\\n",
    "--source-format CasavaOneEightSingleLanePerSampleDirFmt \\\n",
    "--input-path Ye-seqs/ \\\n",
    "--output-path Ye.qza\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data Ye.qza \\\n",
    "--o-visualization Ye.qzv\n",
    "\n",
    "\n",
    "\n",
    "#Nielsen *single-end sequences*\n",
    "!qiime tools import \\\n",
    "  --type 'SampleData[SequencesWithQuality]' \\\n",
    "  --input-path Nielsen2014-manifest.csv \\\n",
    "  --output-path Nielsen2014-demux.qza \\\n",
    "  --source-format SingleEndFastqManifestPhred33\n",
    "\n",
    "!qiime demux summarize \\\n",
    "--i-data Nielsen2014-demux.qza \\\n",
    "--o-visualization Nielsen2014.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## View visualizations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## View one at a time\n",
    "\n",
    "!qiime tools view SongBact.qzv\n",
    "\n",
    "#!qiime tools view yaobact.qzv\n",
    "\n",
    "#!qiime tools view Dai2017.qzv\n",
    "\n",
    "#!qiime tools view imparato.qzv\n",
    "\n",
    "#!qiime tools view Dai2016.qzv\n",
    "\n",
    "#!qiime tools view wu.qzv\n",
    "\n",
    "#!qiime tools view whitman.qzv\n",
    "\n",
    "#!qiime tools view Ye.qzv\n",
    "\n",
    "#!qiime tools view Nielsen2014.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quality control using DADA2 and generate OTU table\n",
    "#### Each dataset went through trimming and truncating optimization to allow for best sequence retention\n",
    "#### Noisy sequences were trimmed off the beginning and most datasets were truncated where sequences consistantly dropped under a qc-score of 35; Whitman and Ye were exceptions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Song\n",
    "\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs SongBact.qza \\\n",
    "  --o-table Song-bact-table \\\n",
    "  --o-representative-sequences song-bact-rep-seqs \\\n",
    "  --p-n-threads 0 \\\n",
    "  --p-trim-left-f 25 \\\n",
    "  --p-trim-left-r 26 \\\n",
    "  --p-trunc-len-f 297 \\\n",
    "  --p-trunc-len-r 210 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table Song-bact-table.qza \\\n",
    "--o-visualization Song-bact-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data song-bact-rep-seqs.qza \\\n",
    "--o-visualization song-bact-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences \n",
    "!qiime tools view Song-bact-table.qzv\n",
    "!qiime tools view song-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Yao\n",
    "\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs yaobact.qza \\\n",
    "  --o-table yaobact-table \\\n",
    "  --o-representative-sequences yaobact-rep-seqs \\\n",
    "  --p-trim-left-f 22 \\\n",
    "  --p-trim-left-r 26 \\\n",
    "  --p-trunc-len-f 268 \\\n",
    "  --p-trunc-len-r 204 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table yaobact-table.qza \\\n",
    "--o-visualization yaobact-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data yaobact-rep-seqs.qza \\\n",
    "--o-visualization yaobact-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view yaobact-table.qzv\n",
    "!qiime tools view yaobact-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Dai2017\n",
    "\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs dai2017.qza \\\n",
    "  --o-table dai2017-table \\\n",
    "  --o-representative-sequences dai2017-rep-seqs \\\n",
    "  --p-trim-left-f 25 \\\n",
    "  --p-trim-left-r 26 \\\n",
    "  --p-trunc-len-f 286 \\\n",
    "  --p-trunc-len-r 200 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table dai2017-table.qza \\\n",
    "--o-visualization dai2017-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data dai2017-rep-seqs.qza \\\n",
    "--o-visualization dai2017-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view dai2017-table.qzv\n",
    "!qiime tools view dai2017-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Imparato\n",
    "\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs imparato.qza \\\n",
    "  --o-table imparato-table \\\n",
    "  --o-representative-sequences imparato-rep-seqs \\\n",
    "  --p-trim-left-f 25 \\\n",
    "  --p-trim-left-r 25 \\\n",
    "  --p-trunc-len-f 243 \\\n",
    "  --p-trunc-len-r 228 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table imparato-table.qza \\\n",
    "--o-visualization imparato-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data imparato-rep-seqs.qza \\\n",
    "--o-visualization imparato-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view imparato-table.qzv\n",
    "!qiime tools view imparato-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Dai2016\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs Dai2016.qza \\\n",
    "  --o-table Dai2016-table \\\n",
    "  --o-representative-sequences Dai2016-rep-seqs \\\n",
    "  --p-trim-left-f 22 \\\n",
    "  --p-trim-left-r 26 \\\n",
    "  --p-trunc-len-f 200 \\\n",
    "  --p-trunc-len-r 210 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table  Dai2016-table.qza \\\n",
    "--o-visualization Dai2016-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data Dai2016-rep-seqs.qza \\\n",
    "--o-visualization Dai2016-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view Dai2016-table.qzv\n",
    "!qiime tools view Dai2016-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Wu\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs wu.qza \\\n",
    "  --o-table wu-table \\\n",
    "  --o-representative-sequences wu-rep-seqs \\\n",
    "  --p-trim-left-f 26 \\\n",
    "  --p-trim-left-r 26 \\\n",
    "  --p-trunc-len-f 270 \\\n",
    "  --p-trunc-len-r 230 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table wu-table.qza \\\n",
    "--o-visualization wu-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data wu-rep-seqs.qza \\\n",
    "--o-visualization wu-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view wu-table.qzv\n",
    "!qiime tools view wu-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Whitman\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs whitman.qza \\\n",
    "  --o-table whitman-table \\\n",
    "  --o-representative-sequences whitman-rep-seqs \\\n",
    "  --p-trim-left-f 5 \\\n",
    "  --p-trim-left-r 5 \\\n",
    "  --p-trunc-len-f 275 \\\n",
    "  --p-trunc-len-r 200 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table  whitman-table.qza \\\n",
    "--o-visualization whitman-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data whitman-rep-seqs.qza \\\n",
    "--o-visualization whitman-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view whitman-table.qzv\n",
    "!qiime tools view whitman-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Ye\n",
    "!qiime dada2 denoise-paired --verbose \\\n",
    "  --i-demultiplexed-seqs Ye.qza \\\n",
    "  --o-table Ye-table-v3 \\\n",
    "  --o-representative-sequences Ye-rep-seqs-v3 \\\n",
    "  --p-trim-left-f 6 \\\n",
    "  --p-trim-left-r 6 \\\n",
    "  --p-trunc-len-f 290 \\\n",
    "  --p-trunc-len-r 269 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table Ye-table-v3.qza \\\n",
    "--o-visualization Ye-table-v3.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data Ye-rep-seqs-v3.qza \\\n",
    "--o-visualization Ye-rep-seqs-v3.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view Ye-table-v3.qzv\n",
    "!qiime tools view Ye-rep-seqs-v3.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Nielsen\n",
    "!qiime dada2 denoise-single --verbose \\\n",
    "  --i-demultiplexed-seqs Nielsen2014-demux.qza \\\n",
    "  --o-table Nielsen2014-table \\\n",
    "  --o-representative-sequences Nielsen2014-rep-seqs \\\n",
    "  --p-trim-left 9 \\\n",
    "  --p-trunc-len 70 \n",
    "\n",
    "# Create visualizations for OTU table and representative sequences\n",
    "!qiime feature-table summarize \\\n",
    "--i-table  Nielsen2014-table.qza \\\n",
    "--o-visualization Nielsen2014-table.qzv \n",
    "\n",
    "!qiime feature-table tabulate-seqs \\\n",
    "--i-data Nielsen2014-rep-seqs.qza \\\n",
    "--o-visualization Nielsen2014-rep-seqs.qzv\n",
    "\n",
    "# View visualizations for OTU table and representative sequences\n",
    "!qiime tools view Nielsen2014-table.qzv\n",
    "!qiime tools view Nielsen2014-rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Taxonomy assignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Download Silva_132 zip file ( https://www.arb-silva.de/download/archive/qiime ) and train the classifer to primers used in samples. \n",
    "\n",
    "Classifier is trained on each different primer combinations (for primers modified, use most ambiguous one (i.e. for 515f/907R use Whitman primer combinations) \n",
    "\n",
    "Dai2016- 515F GTGCCAGCMGCCGCGGTAA ; 806R GGACTACHVGGGTWTCTAAT  \n",
    "\n",
    "Dai2017- 338F ACTCCTACGGGAGGCAGCA ; 806R GGACTACHVGGGTWTCTAAT\n",
    "\n",
    "Song- 338F ACTCCTACGGGAGGCAGCA ; 806R GGACTACHVGGGTWTCTAAT\n",
    "\n",
    "Wu- 338F ACTCCTACGGGAGGCAGCAG ; 806R GGACTACHVGGGTWTCTAAT \n",
    "\n",
    "Imparato- 341F CCTAYGGGRBGCASCAG ; 806R GGACTACNNGGGTATCTAAT \n",
    "\n",
    "Whitman- 515F GTGYCAGCMGCMGCGGTRA ; 907R CCGYCCAATTYMTTTRAGTTT\n",
    "\n",
    "Yao- 515F GTGCCAGCMGCCGCGG ; 907R CCGTCAATTCMTTTRAGTTT\n",
    "\n",
    "Ye - 27F AGAGTTTGATCMTGGCTCAG ; 519R GWATTACCGCGGCKGCTG\n",
    "\n",
    "Nielsen - 515F GTGCCAGCMGCCGCGGTAA ; 806R GGACTACHVGGGTWTCTAAT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import Silva132 99% sequence file and taxonomy as a Qiime artifact - we used the majority taxonomy 7 levels\n",
    "\n",
    "!qiime tools import \\\n",
    "  --type 'FeatureData[Sequence]' \\\n",
    "  --input-path silva132_99.fna \\\n",
    "  --output-path silva132_99.qza\n",
    "\n",
    "!qiime tools import \\\n",
    "  --type 'FeatureData[Taxonomy]' \\\n",
    "  --source-format HeaderlessTSVTaxonomyFormat \\\n",
    "  --input-path majority_taxonomy_7_levels.txt \\\n",
    "  --output-path ref-taxonomy.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Train Silva132 classifier to each primer pair"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## This step sometimes freezes up, so it is best to run each one at a time\n",
    "\n",
    "# 515f and 806r\n",
    "!qiime feature-classifier extract-reads \\\n",
    "  --i-sequences silva132_99.qza \\\n",
    "  --p-f-primer GTGCCAGCMGCCGCGGTAA \\\n",
    "  --p-r-primer GGACTACHVGGGTWTCTAAT \\\n",
    "  --o-reads 515f-806r-ref-seqs.qza\n",
    "\n",
    "!qiime feature-classifier fit-classifier-naive-bayes --verbose\\\n",
    "  --i-reference-reads 515f-806r-ref-seqs.qza \\\n",
    "  --i-reference-taxonomy ref-taxonomy.qza \\\n",
    "  --o-classifier 515f-806r-classifier.qza\n",
    "\n",
    "# 338f and 806r\n",
    "!qiime feature-classifier extract-reads \\\n",
    "  --i-sequences silva132_99.qza \\\n",
    "  --p-f-primer ACTCCTACGGGAGGCAGCA \\\n",
    "  --p-r-primer GGACTACHVGGGTWTCTAAT \\\n",
    "  --o-reads 338f-806r-ref-seqs.qza\n",
    "\n",
    "!qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads 338f-806r-ref-seqs.qza \\\n",
    "  --i-reference-taxonomy ref-taxonomy.qza \\\n",
    "  --o-classifier 338f-806r-classifier.qza\n",
    "\n",
    "# 341f and 806r\n",
    "!qiime feature-classifier extract-reads \\\n",
    "  --i-sequences silva132_99.qza \\\n",
    "  --p-f-primer CCTAYGGGRBGCASCAG \\\n",
    "  --p-r-primer GGACTACNNGGGTATCTAAT \\\n",
    "  --o-reads 341f-806r-ref-seqs.qza\n",
    "\n",
    "!qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads 341f-806r-ref-seqs.qza \\\n",
    "  --i-reference-taxonomy ref-taxonomy.qza \\\n",
    "  --o-classifier 341f-806r-classifier.qza\n",
    "\n",
    "# 515f and 907r\n",
    "!qiime feature-classifier extract-reads \\\n",
    "  --i-sequences silva132_99.qza \\\n",
    "  --p-f-primer GTGYCAGCMGCMGCGGTRA \\\n",
    "  --87`6p-r-primer CCGYCCAATTYMTTTRAGTTT \\\n",
    "  --o-reads 515f-907r-ref-seqs.qza\n",
    "\n",
    "!qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads 515f-907r-ref-seqs.qza \\\n",
    "  --i-reference-taxonomy ref-taxonomy.qza \\\n",
    "  --o-classifier 515f-907r-classifier.qza\n",
    "\n",
    "# 27f and 519r\n",
    "!qiime feature-classifier extract-reads \\\n",
    "  --i-sequences silva132_99.qza \\\n",
    "  --p-f-primer AGAGTTTGATCMTGGCTCAG \\\n",
    "  --p-r-primer GWATTACCGCGGCKGCTG \\\n",
    "  --o-reads 27f-519r-ref-seqs.qza\n",
    "\n",
    "!qiime feature-classifier fit-classifier-naive-bayes \\\n",
    "  --i-reference-reads 27f-519r-ref-seqs.qza \\\n",
    "  --i-reference-taxonomy ref-taxonomy.qza \\\n",
    "  --o-classifier 27f-519r-classifier.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Classify representative sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## This step sometimes freezes up, so it is best to run each one at a time\n",
    "\n",
    "\n",
    "#Dai2016\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "  --i-classifier 515f-806r-classifier.qza \\\n",
    "  --i-reads Dai2016-rep-seqs.qza \\\n",
    "  --o-classification Dai2016-taxonomy.qza \n",
    "\n",
    "#Dai2017\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 338f-806r-classifier.qza \\\n",
    "--i-reads dai2017-rep-seqs.qza \\\n",
    "--o-classification dai2017-taxonomy.qza\n",
    "\n",
    "#Song\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 338f-806r-classifier.qza \\\n",
    "--i-reads song-bact-rep-seqs.qza \\\n",
    "--o-classification song-bact-taxonomy.qza\n",
    "\n",
    "#Wu\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 338f-806r-classifier.qza \\\n",
    "--i-reads wu-rep-seqs.qza \\\n",
    "--o-classification wu-taxonomy.qza\n",
    "\n",
    "#Imparato\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 341f-806r-classifier.qza \\\n",
    "--i-reads imparato-rep-seqs.qza \\\n",
    "--o-classification imparato-taxonomy.qza\n",
    "\n",
    "#Whitman\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 515f-907r-classifier.qza \\\n",
    "--i-reads whitman-rep-seqs.qza \\\n",
    "--o-classification whitman-taxonomy.qza\n",
    "\n",
    "#Yao\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 515f-907r-classifier.qza \\\n",
    "--i-reads yaobact-rep-seqs.qza \\\n",
    "--o-classification yao-taxonomy.qza\n",
    "\n",
    "#Ye\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 27f-519r-classifier.qza \\\n",
    "--i-reads Ye-rep-seqs-v3.qza \\\n",
    "--o-classification ye-taxonomy-v3.qza\n",
    "\n",
    "#Nielsen\n",
    "!qiime feature-classifier classify-sklearn \\\n",
    "--i-classifier 515f-806r-classifier.qza \\\n",
    "--i-reads Nielsen2014-rep-seqs.qza \\\n",
    "--o-classification Nielsen2014-taxonomy.qza"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Export representative sequences, taxonomy, OTU table, and metadata to create a .biom file to use for data analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Dai2016\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir dai2016_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)  \n",
    "!qiime tools export Dai2016-table.qza --output-dir Dai2016_OTU_table\n",
    "!qiime tools export Dai2016-rep-seqs.qza --output-dir Dai2016_OTU_table\n",
    "!qiime tools export Dai2016-taxonomy.qza --output-dir Dai2016_OTU_table\n",
    "!cp dai2016-metadata.txt Dai2016_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls Dai2016_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Dai2016_OTU_table/feature-table.biom \\\n",
    "-o Dai2016_OTU_table/feature-table-metaD.biom \\\n",
    "-m Dai2016_OTU_table/dai2016-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Dai2016_OTU_table/feature-table-metaD.biom \\\n",
    "-o Dai2016_OTU_table/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp Dai2016_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Dai2016_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Dai2016_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 Dai2016_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Dai2016_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Dai2016_OTU_table/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Dai2017\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Dai2017_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export dai2017-table.qza --output-dir Dai2017_OTU_table\n",
    "!qiime tools export dai2017-rep-seqs.qza --output-dir Dai2017_OTU_table\n",
    "!qiime tools export dai2017-taxonomy.qza --output-dir Dai2017_OTU_table\n",
    "!cp dai2017-metadata.txt Dai2017_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "ls Dai2017_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Dai2017_OTU_table/feature-table.biom \\\n",
    "-o Dai2017_OTU_table/feature-table-metaD.biom \\\n",
    "-m Dai2017_OTU_table/dai2017-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Dai2017_OTU_table/feature-table-metaD.biom \\\n",
    "-o Dai2017_OTU_table/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp Dai2017_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Dai2017_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Dai2017_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 Dai2017_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Dai2017_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Dai2017_OTU_table/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Song\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Song_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export Song-bact-table.qza --output-dir Song_OTU_table\n",
    "!qiime tools export song-bact-rep-seqs.qza --output-dir Song_OTU_table\n",
    "!qiime tools export song-bact-taxonomy.qza --output-dir Song_OTU_table\n",
    "\n",
    "!cp song-meta-assume.txt Song_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls Song_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Song_OTU_table/feature-table.biom \\\n",
    "-o Song_OTU_table/feature-table-metaD-assume.biom \\\n",
    "-m Song_OTU_table/song-meta-assume.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Song_OTU_table/feature-table-metaD-assume.biom \\\n",
    "-o Song_OTU_table/feature-table-metaD-tax-assume.biom \\\n",
    "--observation-metadata-fp Song_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Song_OTU_table/feature-table-metaD-tax-assume.biom \\\n",
    "-o Song_OTU_table/feature-table-metaD-tax-summary-assume.txt\n",
    "\n",
    "!head -20 Song_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Song_OTU_table/feature-table-metaD-tax-assume.biom \\\n",
    "-o Song_OTU_table/feature-table-metaD-tax-assume_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Wu\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Wu_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export wu-table.qza --output-dir Wu_OTU_table\n",
    "!qiime tools export wu-rep-seqs.qza --output-dir Wu_OTU_table\n",
    "!qiime tools export wu-taxonomy.qza --output-dir Wu_OTU_table\n",
    "\n",
    "!cp wu-metadata.txt Wu_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Wu_OTU_table/feature-table.biom \\\n",
    "-o Wu_OTU_table/feature-table-metaD.biom \\\n",
    "-m Wu_OTU_table/wu-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Wu_OTU_table/feature-table-metaD.biom \\\n",
    "-o Wu_OTU_table/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp Wu_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Wu_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Wu_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 Wu_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "\n",
    "!biom convert \\\n",
    "-i Wu_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Wu_OTU_table/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Imparato\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Imparato_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export imparato-table.qza --output-dir Imparato_OTU_table\n",
    "!qiime tools export imparato-rep-seqs.qza --output-dir Imparato_OTU_table\n",
    "!qiime tools export imparato-taxonomy.qza --output-dir Imparato_OTU_table\n",
    "\n",
    "!cp imparato-metadata.txt Imparato_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls Imparato_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Imparato_OTU_table/feature-table.biom \\\n",
    "-o Imparato_OTU_table/feature-table-metaD.biom \\\n",
    "-m Imparato_OTU_table/imparato-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Imparato_OTU_table/feature-table-metaD.biom \\\n",
    "-o Imparato_OTU_table/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp Imparato_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Imparato_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Imparato_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 Imparato_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Imparato_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Imparato_OTU_table/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Whitman\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Whitman_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export whitman-table.qza --output-dir Whitman_OTU_table\n",
    "!qiime tools export whitman-rep-seqs.qza --output-dir Whitman_OTU_table\n",
    "!qiime tools export whitman-taxonomy.qza --output-dir Whitman_OTU_table\n",
    "\n",
    "!cp whitman-metadata.txt Whitman_OTU_table/\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls Whitman_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Whitman_OTU_table/feature-table.biom \\\n",
    "-o Whitman_OTU_table/feature-table-metaD-ncbiID.biom \\\n",
    "-m Whitman_OTU_table/whitman-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Whitman_OTU_table/feature-table-metaD-ncbiID.biom \\\n",
    "-o Whitman_OTU_table/feature-table-metaD-tax-ncbiID.biom \\\n",
    "--observation-metadata-fp Whitman_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Whitman_OTU_table/feature-table-metaD-tax-ncbiID.biom \\\n",
    "-o Whitman_OTU_table/feature-table-metaD-tax-summary-ncbiID.txt\n",
    "\n",
    "!head -20 Whitman_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Whitman_OTU_table/feature-table-metaD-tax-ncbiID.biom \\\n",
    "-o Whitman_OTU_table/feature-table-metaD-tax_json-ncbiID.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Yao\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir YaoBact_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export yaobact-table.qza --output-dir YaoBact_OTU_table\n",
    "!qiime tools export yaobact-rep-seqs.qza --output-dir YaoBact_OTU_table\n",
    "!qiime tools export yao-taxonomy.qza --output-dir YaoBact_OTU_table\n",
    "\n",
    "!cp Yao-metadata.txt YaoBact_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls YaoBact_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i YaoBact_OTU_table/feature-table.biom \\\n",
    "-o YaoBact_OTU_table/feature-table-metaD.biom \\\n",
    "-m YaoBact_OTU_table/Yao-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i YaoBact_OTU_table/feature-table-metaD.biom \\\n",
    "-o YaoBact_OTU_table/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp YaoBact_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i YaoBact_OTU_table/feature-table-metaD.biom \\\n",
    "-o YaoBact_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 YaoBact_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i YaoBact_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o YaoBact_OTU_table/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Ye\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Ye_OTU_table-v3\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export Ye-table-v3.qza --output-dir Ye_OTU_table-v3\n",
    "!qiime tools export Ye-rep-seqs-v3.qza --output-dir Ye_OTU_table-v3\n",
    "!qiime tools export ye-taxonomy-v3.qza --output-dir Ye_OTU_table-v3\n",
    "\n",
    "!cp Ye-metadata.txt Ye_OTU_table-v3/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls Ye_OTU_table-v3/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Ye_OTU_table-v3/feature-table.biom \\\n",
    "-o Ye_OTU_table-v3/feature-table-metaD.biom \\\n",
    "-m Ye_OTU_table-v3/Ye-metadata.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Ye_OTU_table-v3/feature-table-metaD.biom \\\n",
    "-o Ye_OTU_table-v3/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp Ye_OTU_table-v3/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Ye_OTU_table-v3/feature-table-metaD.biom \\\n",
    "-o Ye_OTU_table-v3/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 Ye_OTU_table-v3/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Ye_OTU_table-v3/feature-table-metaD-tax.biom \\\n",
    "-o Ye_OTU_table-v3/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Nielsen\n",
    "\n",
    "# Make a new directory to export the files to\n",
    "!mkdir Nielsen2014_OTU_table\n",
    "\n",
    "# Export OTU table, representative sequences, and taxonomy; copy metadata (.txt)\n",
    "!qiime tools export Nielsen2014-table.qza --output-dir Nielsen2014_OTU_table\n",
    "!qiime tools export Nielsen2014-rep-seqs.qza --output-dir Nielsen2014_OTU_table\n",
    "!qiime tools export Nielsen2014-taxonomy.qza --output-dir Nielsen2014_OTU_table\n",
    "\n",
    "!cp Nielsen2014-metadata2.txt Nielsen2014_OTU_table/\n",
    "\n",
    "# Check out files in directory; should have dna-sequences.fastq, taxonomy.tsv, feature-table.biom, and the metadata.txt\n",
    "!ls Nielsen2014_OTU_table/\n",
    "\n",
    "# Add information from metadata.txt to the feature-table.biom\n",
    "!biom add-metadata \\\n",
    "-i Nielsen2014_OTU_table/feature-table.biom \\\n",
    "-o Nielsen2014_OTU_table/feature-table-metaD.biom \\\n",
    "-m Nielsen2014_OTU_table/Nielsen2014-metadata2.txt\n",
    "\n",
    "# Add taxonomy data\n",
    "!biom add-metadata \\\n",
    "-i Nielsen2014_OTU_table/feature-table-metaD.biom \\\n",
    "-o Nielsen2014_OTU_table/feature-table-metaD-tax.biom \\\n",
    "--observation-metadata-fp Nielsen2014_OTU_table/taxonomy.tsv \\\n",
    "--sc-separated taxonomy \\\n",
    "--observation-header OTUID,taxonomy\n",
    "\n",
    "# Check your work by creating a summary text file - view summary to make sure information was saved to .biom\n",
    "!biom summarize-table \\\n",
    "-i Nielsen2014_OTU_table/feature-table-metaD.biom \\\n",
    "-o Nielsen2014_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "!head -20 Nielsen2014_OTU_table/feature-table-metaD-tax-summary.txt\n",
    "\n",
    "# convert the .biom to json format to work with phyloseq package\n",
    "!biom convert \\\n",
    "-i Nielsen2014_OTU_table/feature-table-metaD-tax.biom \\\n",
    "-o Nielsen2014_OTU_table/feature-table-metaD-tax_json.biom \\\n",
    "--table-type=\"OTU table\" \\\n",
    "--to-json"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
